---
title: "Biological Databases: Human Protein Atlas"
output:
  learnr::tutorial:
    css: css/cmoor.css
    progressive: false
    version: 2.0
subtitle: |
  ![](images/cmoor_logo_notext.png){width=60%}
  
  [Get Help](https://help.c-moor.org/){target="_blank"}
runtime: shiny_prerendered

# The fields below will be used to create the first page of the tutorial.  You can use markdown.
# For summary, all new lines must be indented to the same level as the first line.
# The image must be surrounded with quotes
summary: |
  This tutorial will introduce you to the [Human Protein Atlas](https://www.proteinatlas.org/). The Human Protein Atlas is a website that organizes research results about human proteins.
image: "![Human Protein Atlas logo](images/HPA_logo.png)"
image_caption: ""
author:
  - "[Katherine Cox](https://c-moor.github.io/portfolio/coxkatherine/)"
learning_goals:
  - "Use the Human Protein Atlas to look up proteins by name and find out where they are localized."
time: "15 min"
prerequisites: "None"
---

```{r setup, include=FALSE}
library(learnr)
knitr::opts_chunk$set(echo = FALSE)
```


<!---
Don't edit the Welcome page, it will be filled in automatically using the information from the YAML header
Edit the rest of the document as you like
There are some suggested sections to provide a standard order across our tutorials, but they may not all be needed/appropriate for all tutorials.
Section 1. Content 1 has example quizes and exercises
-->

## Welcome {.splashpage}

### `r rmarkdown::metadata$title`

<div class="splashpage-container">
  <figure class="splashpage-image">
  `r rmarkdown::metadata$image`{width=100%}
  <figcaption class="caption">`r rmarkdown::metadata$image_caption`</figcaption>
  </figure>

  `r rmarkdown::metadata$summary`
  

</div>


#### Learning Goals

```{r}
# Extract learning goals from YAML and add HTML tags to make an ordered list
learningGoals <- rmarkdown::metadata$learning_goals
learningGoals <- paste("<li>", learningGoals, "</li>", sep="", collapse="")

```

<ol>
`r learningGoals`
</ol>

#### Authors:

```{r}
# Extract authors from YAML and add HTML tags to make a list
authorList <- rmarkdown::metadata$author
authorList <- paste("<li>", authorList, "</li>", sep="", collapse="")

```

<ul>
`r authorList`
</ul>


```{r}
# Extract the tutorial version from the YAML data and store it so we can print it using inline r code below.  This can't be done directly inline because the code for extracting the YAML data uses backticks
tv <- rmarkdown::metadata$output$`learnr::tutorial`$version
```

#### Version: `r tv`

## Introduction

### What you need to know before you start:

- The form and function of genes
- The form and function of proteins
- The Central Dogma: How transcription and translation read a DNA template and synthesize a polypeptide from   those instructions
- Gene expression: How variation in gene expression creates specialized cells

<!-- #Image of Central Dogma  -->
![](images/Central_Dogma_NIH_2025.jpg){width=100%}

<p class=caption> **The Central Dogma**: The process of reading a DNA template and synthesizing a polypeptide from those instructions. Image sourced from The NIH National Human Genome Research Institute (February 11th 2025). Available from: https://www.genome.gov/genetics-glossary/Central-Dogma License: [CC BY 3.0] </p>


### The Human Protein Atlas: What is it?

The Human Protein Atlas is a website that organizes research results about human proteins.  It contains many types of information, including what the proteins do, where they are active, and what types of diseases they are involved in.  We can use it to find out what has been discovered about genes we’re interested in.

Remember: Not all genes code for proteins; genes can have other types of information and instructions.  The Human Protein Atlas is focused specifically on proteins and the genes that code for them (“protein-coding genes”).

### The Human Protein Atlas: What can I do with it?

![](images/HPA_Home_Head_2025.png){width=70%}
<p class=caption> **The Human Protein Atlas Homepage**. Image sourced from The Human Protein Atlas (www.proteinatlas.org) (February 11th 2025).</p>

You will primarily interact with the Human Protein Atlas by typing the names of genes or proteins of interest into the search bar. 


![](images/HPA_Home_Middle_2025.png){width=70%}
<p class=caption> **The Human Protein Atlas Homepage: Resources**. Scroll down on the homepage of the Human Protein Atlas to find these resources. We will explore them a little later in this module. Image sourced from The Human Protein Atlas (www.proteinatlas.org) (February 11th 2025).</p>

The boxes below the search bar ("Tissue Atlas", "Cell Atlas", etc.) provide explanations about the types of information available in the Human Protein Atlas. These pages can also be accessed through the "Resources" tab at the top of the page.

Feel free to explore the website as you wish. When you're ready, in the next section we'll try using the search bar on the homepage to learn more about a protein.

## Human Protein Atlas: I have a protein I'm interested in.

We have a protein in mind! Great!

Every protein in the Human Protein Atlas has its own page with a large collection of information about the protein. 

**Here is where the goals of the how-to can go**

### Step 1: Use the search bar to find a protein

![](images/HPA_Search_Box_2025.png){width=100%}

Proteins have a number of **aliases** - different names that all refer to the same protein. In this example, we searched for Insulin, but we would also get similar results if we searched for the other aliases for insulin such as INS, IDDM1, or IDDM2.

### Step 2: Examine the search results page

![](images/HPA_Search_Results_2025.png){width=100%}

Here is our results page. In the top left side of the results, we can see that our search brought back 635 genes and their related proteins. Notice also how even though we asked for insulin, insulin is actually the third result.

We can also get a quick overview of each protein by hovering our mouse over each resource. In this example, we see that within all of HPA's single-cell RNA-seq data insulin had the highest expression in pancreatic cells. That makes a lot of sense, since insulin is produced in the pancreas!

Let's click on "Insulin" in the 2nd column to go to insulin's protein page.

### Step 3: Explore the protein's page

![](images/HPA_Insulin_Protein_Summary_2025.png){width=100%}
Here we are in the summary tab on the insulin page. Here we can view a quick overview of what the protein does, where it is expressed, and where in the body the protein can be found. 

### Step 4: Examine the gene information page

While we're on the summary tab, let's check out the gene information page.

![](images/HPA_Insulin_Gene_Information_2025.png){width=100%}
This page has another summary of what the protein does in the body and has helpful links to other databases (highlighted in the red box). You can use these other databases to get more information on the protein. For example, you can find resources for insulin orthologs in other species using  Ensembl, Entrez gene (NCBI), and UniProt.

![](images/HPA_Insulin_Tissue_2025.png){width=100%}
Insulin is located specifically in the pancreas.

### Step 5: Learn more about what tissues in the body the protein is found

![](images/HPA_Insulin_Tissue_2025.png){width=100%}

Now let's check out the ["Tissue" tab](https://www.proteinatlas.org/ENSG00000254647-INS/tissue). At the top of this tab under "Human Protein Atlas Information", we can get an overview of which tissues in the body insulin is found. As before, we can see that insulin is primarily found in the pancreas.

![](images/HPA_Insulin_Tissue_Expression_2025.png){width=100%}

Scrolling down further, we arrive at a section titled "RNA AND PROTEIN EXPRESSION SUMMARY". Here we can see the expression data for both the RNA that codes for the protein and the protein itself. We can also see where in the body those tissues are. We know that insulin is made in the pancreas, but in this diagram, we can see other tissues represented in light red. Indeed, there seems to be some minimal expression of insulin in other parts of the body, but the pancreas - shown in dark red - is where the vast majority of insulin is expressed.

The other tabs, such as "Brain", "Single cell" and "Subcell" all look at expression data relating to those specific parts of the body or those specific data types.

### Try it yourself! Human Protein Atlas: Look up a protein

Phenylketonuria (PKU) is a severe disease where individuals are unable to metabolize the amino acid phenylalanine.  If untreated, children with PKU develop seizures and irreversible intellectual disabilities.  It can be controlled by maintaining a strict diet, and individuals can go on to live a normal life.  Babies are routinely screened for PKU shortly after birth, so that they can begin treatment immediately.

PKU is caused by deficiency in phenylalanine hydroxylase - the protein that processes phenylalanine.

Go to the Human Protein Atlas and enter "Phenylalanine hydroxylase" in the search bar to find out more about this protein.

```{r pku-quiz}
quiz(caption = "Investigate phenylalanine hydroxylase",
  question("What is the name of this gene?",
    answer("PKU"),
    answer("PAH", correct = TRUE),
    answer("PHE"),
    answer("Phenylalanine"),
    allow_retry = TRUE,
    random_answer_order = TRUE
  ),
  question("What tissues is this gene expressed in?  HINT: Choose 3",
    answer("Stomach"),
    answer("Liver", correct = TRUE),
    answer("Spleen"),
    answer("Gallbladder", correct = TRUE),
    answer("Kidney", correct = TRUE),
    answer("Small intestine"),
    allow_retry = TRUE,
    random_answer_order = TRUE
  ),
  question("Where in the cell is it localized?  HINT: Choose 2",
    answer("Nucleus"),
    answer("Cytoplasm"),
    answer("Extracellular matrix"),
    answer("Endoplasmic reticulum", correct = TRUE),
    answer("Mitochondria"),
    answer("Vesicles", correct = TRUE),
    allow_retry = TRUE,
    random_answer_order = TRUE
  )
)
```

## Summary

The Human Protein Atlas is a useful source of information when researching human genes.  Some key features include:

- A convenient summary of what is known about the gene's function
- Information on disease associations
- Nice graphical depicitions of where the protein is active
