<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Valeriya Gaysinskaya, John Hopkins University" />
<meta name="author" content="Sayumi York, Notre Dame of Maryland University" />
<meta name="author" content="Frederick Tan, John Hopkins University" />


<meta name="progressive" content="false" />
<meta name="allow-skip" content="false" />
<meta name="learnr-version-prerender" content="0.11.5" />

<title>Analyzing 16S data in Phyloseq</title>

<!-- header-includes START -->
<!-- HEAD_CONTENT -->
<!-- header-includes END -->
<!-- HEAD_CONTENT -->

<!-- highlightjs -->
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>


<!-- taken from https://github.com/rstudio/rmarkdown/blob/de8a9c38618903627ca509f5401d50a0876079f7/inst/rmd/h/default.html#L293-L343 -->
<!-- tabsets -->
<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- end tabsets -->

<link rel="stylesheet" href="css/cmoor.css" type="text/css" />

</head>

<body>
<a class='sr-only sr-only-focusable visually-hidden-focusable' href='#learnr-tutorial-content'>Skip to Tutorial Content</a>



<div class="pageContent band">
<main class="bandContent page">

<article class="topics" id="learnr-tutorial-content">

<!---
Don't edit the Welcome page, it will be filled in automatically using the information from the YAML header
Edit the rest of the document as you like
There are some suggested sections to provide a standard order across our tutorials, but they may not all be needed/appropriate for all tutorials.
Section 1. Content 1 has example quizes and exercises
-->
<div id="section-welcome" class="section level2 splashpage">
<h2 class="splashpage">Welcome</h2>
<div id="section-analyzing-16s-data-in-phyloseq" class="section level3">
<h3>Analyzing 16S data in Phyloseq</h3>
<div class="splashpage-container">
<figure class="splashpage-image">
<div class="float">
<img src="images/phyloseq_logo.png" style="width:100.0%"
alt="The phyloseq logo (2025)" />
<div class="figcaption">The phyloseq logo (2025)</div>
</div>
<figcaption class="caption">
</figcaption>
</figure>
<p>This tutorial will introduce you to more advanced functions in
phyloseq and DESeq2 that will allow us to perform more complex analyses
including alpha diversity, multidimensional scaled plots, and
differential abundance. These analyses would be incredibly complex and
difficult to do manually and showcase the convenience and power of R and
the associated packages.</p>
</div>
<div id="section-learning-goals" class="section level4">
<h4>Learning Goals</h4>
<ol>
<li>
Use phyloseq to analyze and visualize 16S microbiome data.
</li>
<li>
Use DESeq2 in phyloseq to perform differential abundance analysis.
</li>
</ol>
</div>
<div id="section-authors" class="section level4">
<h4>Authors:</h4>
<ul>
<li>
Valeriya Gaysinskaya, John Hopkins University
</li>
<li>
Sayumi York, Notre Dame of Maryland University
</li>
<li>
Frederick Tan, John Hopkins University
</li>
</ul>
</div>
<div id="section-version-2" class="section level4">
<h4>Version: 2</h4>
</div>
</div>
</div>
<div id="section-phyloseq" class="section level2">
<h2>Phyloseq</h2>
<p>In this module we’ll do some more advanced analysis on our 16S data.
Up until this point we have used R as a more efficient, more powerful
tool compared to working in Google Sheets or Microsoft Excel. The
following analyses would be exceptionally to complete without a
computational tool like R.</p>
<p>This module will use the following packages:</p>
<ul>
<li><a
href="https://joey711.github.io/phyloseq/"><strong>phyloseq</strong></a>:
A library that specializes in 16S microbial community data</li>
<li><a
href="https://ggplot2.tidyverse.org/"><strong>ggplot2</strong></a>: A
library that is part of the tidyverse group of packages that will allow
us to create great figures</li>
<li><a
href="https://github.com/thelovelab/DESeq2"><strong>DESeq2</strong></a>:
A library made for performing differential expression analysis; we will
be co-opting this function for a differential abundance analysis.</li>
</ul>
<p>This module assumes that you have previously completed our “Exploring
16S data with phyloseq” module. We will be using the same data.
Likewise, this tutorial assumes you have some previous exposure to
working with R objects.</p>
</div>
<div id="section-alpha-diversity" class="section level2">
<h2>Alpha diversity</h2>
<!-- Uses raw count data -->
<ul>
<li><strong>Objective</strong>: Compare the biodiversity (richness and
evenness) between samples/groups</li>
<li><strong>Best for questions like</strong>: How variable or stable is
the microbial community over time or space?</li>
<li><strong>Example question</strong>: Does changing the diet reduce
biodiversity in the human gut? Are male microbiomes more diverse than
female microbiomes?</li>
<li><strong>Data format</strong>: Because we want to include rare
species in our analysis, we will use our raw data, which is in
counts.</li>
</ul>
<p>We can measure the biodiversity by taking a measure of each sample’s
alpha diversity (diversity within a sample) and comparing these
measurements. We will use <strong>Simpson’s index</strong> to calculate
alpha diversity here. Simpson’s alpha diversity measure based on the
number of unique ASVs and their proportion of them within sample, taking
into account which species are more dominant and which are more rare,
with an emphasis on species evenness.</p>
<p><strong><em>Note that with Simpson’s index, a HIGHER score on the
scale (0-1) means a LOWER alpha diversity.</em></strong></p>
<img src="images/Diversity_Figure.png" style="width:70.0%" />
<p class="caption">
<strong>Species richness vs species evenness</strong>: Although the
number of species between the two samples is the same, sample 2 has a
higher species evenness, where each species is represented more equally.
Bacteria art from NIAID Visual &amp; Medical Arts (42, 57, &amp; 179).
Figure by Sayumi York (March 27, 2025).
</p>
<p>We will do this using the plot_richness() function of phyloseq. As
you can see, we can again customize the plot based on different metadata
fields. If we don’t want these extra fields, we can just delete them or
add a # in front of them to comment them out. We will use non-normalized
data in counts (miso_counts) as recommended by <a
href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003531">McMurdie
&amp; Holmes (2014)</a> for our alpha diversity and differential
abundance analyses.</p>
<p>If you get a warning that contains the message: The data you have
provided does not have any singletons, you may ignore this and
continue.</p>
<div class="tutorial-exercise" data-label="plot_richness_example"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>plot_richness(miso_counts, x=&quot;subject&quot;, 
              color=&quot;timepointgroup&quot;, 
              shape=&quot;timepointgroup&quot;, 
              title =&quot;fill in the blank&quot;,
              measures=&quot;Simpson&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Naturally, we can also subset our dataset through either
subset_samples or subset_taxa and make new plots.</p>
<div class="tutorial-exercise" data-label="plot_richness_subset_example"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>my_subset &lt;- subset_samples(miso_counts, gender == &quot;F&quot;)

plot_richness(my_subset, x=&quot;timepoint&quot;, 
              color=&quot;phase&quot;, 
              shape=&quot;timepointgroup&quot;, 
              title =&quot;fill in the blank&quot;,
              measures=&quot;Simpson&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="panel-heading tutorial-quiz-title">Using plot_richness to plot alpha diversity</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="alpha_diversity_quiz-1" class="tutorial-question panel-body">
<div id="alpha_diversity_quiz-1-answer_container" class="shiny-html-output"></div>
<div id="alpha_diversity_quiz-1-message_container" class="shiny-html-output"></div>
<div id="alpha_diversity_quiz-1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="alpha_diversity_quiz-2" class="tutorial-question panel-body">
<div id="alpha_diversity_quiz-2-answer_container" class="shiny-html-output"></div>
<div id="alpha_diversity_quiz-2-message_container" class="shiny-html-output"></div>
<div id="alpha_diversity_quiz-2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-multidimensional-analysis" class="section level2">
<h2>Multidimensional analysis</h2>
<div id="section-what-is-a-multidimensional-analysis"
class="section level3">
<h3>What is a multidimensional analysis?</h3>
<img src="images/1D_Graph.png" style="width:70.0%" />
<p class="caption">
<strong>A 1D graph</strong>: We have one dimension (age) to compare
samples. Figure by Sayumi York (March 27, 2025).
</p>
<p>What does a one-dimensional graph look like? It’s likely you use one
every day although you may not think of it as a graph. Take for example,
a ruler or the volume control on your computer. A one-dimensional graph
simply graphs a measure of something with just one variable (ex. # of
centimeters or volume measurement).</p>
<img src="images/2D_Graph.png" style="width:70.0%" />
<p class="caption">
<strong>A 2D graph</strong>: We have two dimensions (age and height) to
compare samples. Figure by Sayumi York (March 27, 2025).
</p>
<p>A two-dimensional graph is probably the graph you’re most familiar
with. A two-dimensional graph has an x-axis and a y-axis and it draws a
relationship between the measurements on each axis to describe their
relationship. For example, <em>y = mx + b</em> describes a linear
relationship between a given x variable and its respective y
variable.</p>
<img src="images/3D_Graph.png" style="width:70.0%" />
<p class="caption">
<strong>A 3D graph</strong>: We have three dimensions (age, height, and
weight) to compare samples. The size of the data points correlates to
their z-axis. Figure by Sayumi York (March 27, 2025).
</p>
<p>How about a three-dimensional graph? A three-dimensional graph would
have a z-axis, and would add depth. Data points would be plotted in a
space, like a cube sticking out from your screen (here we are imagining
the larger data points are closer to us).</p>
<p>We hit some problems once we have a four-dimensional graph. We live
in a 3-dimensional world (3D), so we would need to plot the graph on a
fourth axis like time, shape, or color.</p>
<p>As you can imagine as we add on a fifth or sixth and beyond dimension
things become even more complicated to graph and then interpret. But
some experiments have more than one factor, and in genomics we often
have many; one for each variable in our metadata and one for each ASV
depending on what question we’re asking.</p>
<p>We can tackle this challenge with a PCoA.</p>
</div>
</div>
<div id="section-pcoa" class="section level2">
<h2>PCoA</h2>
<ul>
<li><strong>Objective</strong>: Comparing how similar samples are to one
another</li>
<li><strong>Best for questions like</strong>: How much variability is
there within a given condition? What samples are most similar to each
other?</li>
<li><strong>Example question</strong>: Are male and female samples
grouped together? Do samples group together by a given metabolite? How
similar are samples within each timepointgroup?</li>
<li><strong>Data format</strong>: We will use normalized counts
(proportions) to be able to make comparisons between samples.</li>
</ul>
<p>A Principal coordinates analysis (PCoA) is a multidimensional
analysis that shrinks down your dimensions into groups called principal
coordinates. Each principle coordinate (PC) is a collection of variables
and their relationships, and the components are ordered by the amount of
variation within the data set they explain. A PCoA then plots the data
along two axes, each representing a PC and thus takes a multidimensional
analysis and makes it so the results are visible in a 2D space!</p>
<p>Let’s see it in action!</p>
<hr />
<div id="section-step-1-ordinate" class="section level4">
<h4>Step 1: Ordinate</h4>
<p>The first step is to create a similarity matrix between all samples
through the ordinate function. Every sample is compared to every other
using the Bray-Curtis distance (a statistical measure); we have stored
the data in a new object called miso.pcoa.</p>
<p>We can look at the ordination through head(miso.pcoa) to see that
something has been calculated, but the results are not that easy to
understand at this point.</p>
<div class="tutorial-exercise" data-label="ordinate_PCoA"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>miso.pcoa &lt;- ordinate(miso, method=&quot;PCoA&quot;, distance=&quot;bray&quot;)
head(miso.pcoa)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-step-2-plot-pcoa" class="section level4">
<h4>Step 2: Plot PCoA</h4>
<p>Now that we have our ordination, we use our original phyloseq object
(miso) along with the ordination object (miso.pcoa) we created in the
previous code block to create the graph.</p>
<p>Notice how we can ask R to color or change the shape of the data
points based on a metadata column. And of course, we can’t forget a
title!</p>
<div class="tutorial-exercise" data-label="plot_PCoA"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>plot_ordination(miso, miso.pcoa, 
        color = &quot;subject&quot;, 
        shape = &quot;phase&quot;, 
        title=&quot;My PCoA&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>The samples that are closer together are more similar in composition
than samples far away from each other.</p>
<p>Each principal coordiate is a collection of relationships between the
different variables in your dataset. Axis 1 represents principal
coordinate 1, which explains 16.7% of the variance in the dataset.
Likewise, Axis 2 represents principal coordinate 2, which explains 11.9%
of variance in the dataset. These percentages are rather low. Generally,
if the percentage is high that means there are only a handful of factors
that are really driving the relationship between samples.</p>
<p>A PCoA is a complex analysis. It’s okay if you still don’t understand
the principal coordinates themselves. Just focus on how close the
samples are to each other.</p>
</div>
<div id="section-subsetting-a-pcoa" class="section level4">
<h4>Subsetting a PCoA</h4>
<p>If we just change the field for color or shape, the PCoA plot itself
won’t change - only the appearance of the data point changes. If we want
to do an PCoA with a given subset of samples, we need to create that
subset and repeat the ordination too. Be careful with this, since R
won’t give you an error if you use the same ordination with a different
phyloseq object so long as R notices the sample names match!</p>
<p>The following code plots an PCoA to compare the composition of all
ASVs with the phylum Firmicutes.</p>
<p>Just like before, you can subset your phyloseq object (miso) with
subset_taxa() and subset_samples() to create variations of your
PCoA.</p>
<div class="tutorial-exercise" data-label="subset_PCoA"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#Subset a group of interest
miso_subset &lt;- subset_taxa(miso, Phylum == &quot;Firmicutes&quot;)

#Ordinate based on your subset
miso.pcoa.subset &lt;- ordinate(miso_subset, method=&quot;PCoA&quot;, distance=&quot;bray&quot;)

#Plot the new subsetted ordination with the subsetted phyloseq object
plot_ordination(miso_subset, miso.pcoa.subset, 
        color = &quot;subject&quot;, 
        shape = &quot;phase&quot;, 
        title=&quot;My subsetted (Firmicutes) PCoA&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<!-- Eigenvalues Tutorial?-->
<!-- Eigenvectors Tutorial? -->
<div class="panel-heading tutorial-quiz-title">Creating and interpreting a PCoA plot</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="pcoa_quiz-1" class="tutorial-question panel-body">
<div id="pcoa_quiz-1-answer_container" class="shiny-html-output"></div>
<div id="pcoa_quiz-1-message_container" class="shiny-html-output"></div>
<div id="pcoa_quiz-1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="pcoa_quiz-2" class="tutorial-question panel-body">
<div id="pcoa_quiz-2-answer_container" class="shiny-html-output"></div>
<div id="pcoa_quiz-2-message_container" class="shiny-html-output"></div>
<div id="pcoa_quiz-2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="pcoa_quiz-3" class="tutorial-question panel-body">
<div id="pcoa_quiz-3-answer_container" class="shiny-html-output"></div>
<div id="pcoa_quiz-3-message_container" class="shiny-html-output"></div>
<div id="pcoa_quiz-3-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
</div>
<div id="section-differential-abundance" class="section level2">
<h2>Differential abundance</h2>
<ul>
<li><strong>Objective</strong>: Compare the change in
abundance/proportion of ASVs between groups (conditions)</li>
<li><strong>Best for questions like</strong>: What ASVs showed the most
change between groups (conditions)?</li>
<li><strong>Example question</strong>: What ASVs changed the most
between timepoint 1 and 2? What trends are there in the change in
abundance for a given taxonomic group?</li>
<li><strong>Data format</strong>: DESeq2 will normlalize the data for
us, so we will use our raw count data</li>
</ul>
<p>Differential abundance shows us the difference in abundance of ASVs
between two conditions. We will be using the library DESeq2. You can see
the application of DESeq2 for differential expression in our RNA-seq
modules where was also talk about how to interpret the log fold
change.</p>
<p>For this analysis, we will be using count data; DESeq2 will take into
consideration the differences in total number of reads between samples
for us. In this section we will show you the different parts of the code
one by one and then combine them at the end.</p>
<hr />
<div
id="section-step-1-convert-the-phyloseq-object-to-a-deseq2-object-define-an-experimental-design"
class="section level3">
<h3>Step 1: Convert the phyloseq object to a DESeq2 object define an
experimental design</h3>
<p>The function phyloseq_to_deseq2 will take care of all the conversion
of the data for us. What is important is that we tell DESeq2 what do
with with the data.</p>
<p>The design field is where we put our factor of interest.</p>
<p>In this example, we are comparing samples between the groups of the
variable timepointgroup (BD, HD, and WO). We will then call
my_miso_DESeq2 to check that it is a DESeqDataSet object.</p>
<div class="tutorial-exercise" data-label="phyloseq_to_deseq2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)
my_miso_DESeq2</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-step-2-tell-deseq2-which-groups-you-want-to-compare"
class="section level3">
<h3>Step 2: Tell DESeq2 which groups you want to compare</h3>
<p>DESeq2 will only give you the results for one comparision at a time.
For example, in the variable timepointgroup, we have to make a total of
three comparisions if we wanted to make every possible comparision.</p>
<ul>
<li>BD vs HD</li>
<li>HD vs WO</li>
<li>WO vs BD</li>
</ul>
<p>We’ll just choose the last one for now. my_comparision should be a
list of strings identical to what we put in (“timepointgroup”, “WO”,
“BD”)</p>
<div class="tutorial-exercise" data-label="create_my_comparision"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>my_comparision &lt;-c(&quot;timepointgroup&quot;, &quot;WO&quot;, &quot;BD&quot;)
my_comparision</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-step-3-run-the-differential-abundance-analysis"
class="section level3">
<h3>Step 3: Run the differential abundance analysis</h3>
<p>We have created the function Differential_Abundance() for you to help
skip some of the more complicated steps. As written, this function will
show you all the ASVs that have a differential abundance with a p-value
of 0.05 (5%) or below. If you require a different p-value, you can
change the number.</p>
<p>We store the results in a new object called Significant_DEseq2_ASVs.
To make sure it’s working we call Significant_DEseq2_ASVs to see a table
of significantly differential abundance ASVs based on our chosen
p-value.</p>
<div class="tutorial-exercise" data-label="run_differential_abundance"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>Significant_DEseq2_ASVs&lt;-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)
Significant_DEseq2_ASVs</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div
id="section-step-4-view-the-asvs-with-significant-differential-abundance"
class="section level3">
<h3>Step 4: View the ASVs with significant differential abundance</h3>
<p>We include calling Significant_DEseq2_ASVs as step 4. While we won’t
check the output of steps 1 and 2 every time we run the code, we
definitely want the table each time we run the analysis.</p>
<div class="tutorial-exercise" data-label="significant_DEseq2_genes"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>Significant_DEseq2_ASVs</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-step-5-plot-our-results-in-a-ggplot"
class="section level3">
<h3>Step 5: Plot our results in a ggplot</h3>
<p>Change the value in x = to select what goes on the x-axis and change
the value in color = to change how those values are labeled. For
instance, we might plot a phylum on the x-axis and then color our data
points by class.</p>
<p>Note that this plot only uses the ASVs with a significantly different
abundance between our groups. So if only see a few data points, the plot
is working you just don’t have that many ASVs with significantly
different abundance between the groups in the comparison you chose.</p>
<div class="tutorial-exercise" data-label="plot_deseq2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>ggplot(Significant_DEseq2_ASVs, aes(x=Class, y=log2FoldChange, color=Class)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-the-complete-differential-abundance-code"
class="section level3">
<h3>The complete differential abundance code</h3>
<p>Let’s put it all together. Try run the analysis below.</p>
<div class="tutorial-exercise" data-label="DESeq2_example"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design 
my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)

# STEP 2: Select the groups to compare
my_comparision &lt;-c(&quot;timepointgroup&quot;, &quot;WO&quot;, &quot;BD&quot;)

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs&lt;-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle(&quot;my title here&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>How do we interpret the results?</p>
<ul>
<li><strong>log2FoldChange</strong>: The latter of the two conditions in
our chosen variable is what DESeq2 considers the baseline. For example,
the code: <em>my_comparision &lt;-c(“timepointgroup”, “WO”, “BD”)</em>,
sets the baseline as BD. Thus, any positive number represents a greater
abundance in the former (“WO”) condition compared to the latter (“BD”).
The larger the magnitude of the number, the greater the difference in
abundance.</li>
<li><strong>p-value</strong>: The base p-value, without corrections for
multiple comparisons.</li>
<li><strong>p-adj</strong>: The p-value after correcting for multiple
comparisions. You should use p-adj when reporting your results.</li>
</ul>
<p>In this example, ASV245 has a negative log2FoldChange. Since our
baseline condition is BD, that means there is a lower abundance of
ASV245 in the WO condition. Likewise, ASV1114 has a positive
log2FoldChange, so ASV1114 has an increased abundance in the WO
condition. Finally, the magnitude of ASV245 is greater than the
magnitude of ASV1114. Thus, ASV245 has a larger difference in abundance
between BD and WO conditions than ASV1114 does.</p>
<div class="panel-heading tutorial-quiz-title">Using DESeq2 to examine differential abundance - WO vs BD</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz-1" class="tutorial-question panel-body">
<div id="differential_abundance_quiz-1-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz-1-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz-1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz-2" class="tutorial-question panel-body">
<div id="differential_abundance_quiz-2-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz-2-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz-2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz-3" class="tutorial-question panel-body">
<div id="differential_abundance_quiz-3-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz-3-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz-3-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel-heading tutorial-quiz-title"></div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz_part2-1" class="tutorial-question panel-body">
<div id="differential_abundance_quiz_part2-1-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_part2-1-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_part2-1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<p>To change the comparision, if we are using the same factor, we only
need to change the code in Step 2. Change the code below and swap out
“WO” for “HD” so that we can compare the “HD” to the “BD” (baseline)
group.</p>
<div class="tutorial-exercise" data-label="DESeq2_example_comparison"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design 
my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)

# STEP 2: Select the groups to compare
my_comparision &lt;-c(&quot;timepointgroup&quot;, &quot;WO&quot;, &quot;BD&quot;)

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs&lt;-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle(&quot;my title here&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="panel-heading tutorial-quiz-title">Using DESeq2 to examine differential abundance - HD vs BD</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz_comparison-1" class="tutorial-question panel-body">
<div id="differential_abundance_quiz_comparison-1-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_comparison-1-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_comparison-1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz_comparison-2" class="tutorial-question panel-body">
<div id="differential_abundance_quiz_comparison-2-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_comparison-2-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_comparison-2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<p>If we want to change the factor we are looking at (example: from
timepointgroup to subject) we will have to change the design code in
Step 1 as well. In step 1 change timepointgroup to subject.</p>
<p>Then of course, step 2 needs to match. In step 2 change
timepointgroup to subject.</p>
<p>Finally, we need to choose two variables within the factor subject to
compare. Let’s choose S02 and subject S03. In step 2 replace “WO” with
“S02”. Then replace “BD” with “S03”.</p>
<div class="tutorial-exercise" data-label="DESeq2_example_design"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design 
my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)

# STEP 2: Select the groups to compare
my_comparision &lt;-c(&quot;timepointgroup&quot;, &quot;WO&quot;, &quot;BD&quot;)

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs&lt;-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle(&quot;my title here&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="panel-heading tutorial-quiz-title">Using DESeq2 to examine differential abundance</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz_design-1" class="tutorial-question panel-body">
<div id="differential_abundance_quiz_design-1-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_design-1-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_design-1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<div class="panel panel-default tutorial-question-container">
<div data-label="differential_abundance_quiz_design-2" class="tutorial-question panel-body">
<div id="differential_abundance_quiz_design-2-answer_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_design-2-message_container" class="shiny-html-output"></div>
<div id="differential_abundance_quiz_design-2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
</div>
<div id="section-template-code" class="section level2">
<h2>Template code</h2>
<div id="section-pcoa-1" class="section level3">
<h3>PCoA</h3>
<div class="tutorial-exercise" data-label="Try_it_PCoA_end"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#Subset a group of interest
miso_subset &lt;- subset_taxa(miso, Phylum == &quot;Firmicutes&quot;)

#Ordinate based on your subset
miso.pcoa.subset &lt;- ordinate(miso, method=&quot;PCoA&quot;, distance=&quot;bray&quot;)

#Plot the new subsetted ordination with the subsetted phyloseq object
plot_ordination(miso_subset, miso.pcoa.subset, 
        color = &quot;subject&quot;, 
        shape = &quot;phase&quot;, 
        title=&quot;My subsetted (Firmicutes) PCoA&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-alpha-diversity-1" class="section level3">
<h3>Alpha Diversity</h3>
<div class="tutorial-exercise" data-label="try_it_plot_richness_end"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>my_subset &lt;- subset_samples(miso_counts, gender == &quot;F&quot;)

plot_richness(my_subset, x=&quot;timepoint&quot;, 
              color=&quot;phase&quot;, 
              shape=&quot;timepointgroup&quot;,
              title =&quot;fill in the blank&quot;, 
              measures= &quot;Simpson&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<hr />
</div>
<div id="section-differential-abundance-1" class="section level3">
<h3>Differential abundance</h3>
<div class="tutorial-exercise" data-label="try_it_DESeq2_end"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design 
my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)

# STEP 2: Select the groups to compare
my_comparision &lt;-c(&quot;timepointgroup&quot;, &quot;WO&quot;, &quot;HD&quot;)

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs&lt;-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle(&quot;my title here&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise"
data-label="Try_it_differential_abundance_blank" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-related-r-commands" class="section level2">
<h2>Related R commands</h2>
</div>
<div id="section-advanced" class="section level2">
<h2>Advanced</h2>
<ul>
<li>Add ellipsis CI @ 95%</li>
<li>R formulas in DESeq2 (ex. interaction terms)</li>
</ul>
</div>
<div id="section-summary" class="section level2">
<h2>Summary</h2>
</div>
<div id="section-faq" class="section level2">
<h2>FAQ</h2>
</div>
<div id="section-common-errors" class="section level2">
<h2>Common errors</h2>
<p>No significant genes found No datapoint on the DESeq2 graph - missing
NA? Warning: Removed 3 rows containing missing values or values outside
the scale range (<code>geom_point()</code>).</p>
</div>
<div id="section-works-cited" class="section level2">
<h2>Works cited</h2>
<ul>
<li><p>Aden-Buie G, Schloerke B, Allaire J, Rossell Hayes A (2023).
learnr: Interactive Tutorials for R. <a
href="https://rstudio.github.io/learnr/"
class="uri">https://rstudio.github.io/learnr/</a>, <a
href="https://github.com/rstudio/learnr"
class="uri">https://github.com/rstudio/learnr</a></p></li>
<li><p>Evans, Ciaran, Johanna Hardin, and Daniel M. Stoebel. “Selecting
between-sample RNA-Seq normalization methods from the perspective of
their assumptions.” Briefings in bioinformatics 19.5 (2018):
776-792.</p></li>
<li><p>Guthrie, Leah, et al. “Impact of a 7-day homogeneous diet on
interpersonal variation in human gut microbiomes and metabolomes.” Cell
host &amp; microbe 30.6 (2022): 863-874.</p></li>
<li><p>McMurdie, Paul J., and Susan Holmes. “phyloseq: an R package for
reproducible interactive analysis and graphics of microbiome census
data.” PloS one 8.4 (2013): e61217.</p></li>
<li><p>McMurdie, Paul J., and Susan Holmes. “Waste not, want not: why
rarefying microbiome data is inadmissible.” PLoS computational biology
10.4 (2014): e1003531.</p></li>
<li><p>NIAID Visual &amp; Medical Arts. (10/7/2024). Bacillus Bacteria.
NIAID NIH BIOART Source. bioart.niaid.nih.gov/bioart/42</p></li>
<li><p>NIAID Visual &amp; Medical Arts. (10/7/2024). Borrelia. NIAID NIH
BIOART Source. bioart.niaid.nih.gov/bioart/57</p></li>
<li><p>NIAID Visual &amp; Medical Arts. (10/7/2024). Gram Negative
Bacteria. NIAID NIH BIOART Source.
bioart.niaid.nih.gov/bioart/179</p></li>
<li><p>R Core Team (2024). R: A Language and Environment for Statistical
Computing. R Foundation for Statistical Computing, Vienna, Austria. <a
href="https://www.R-project.org/"
class="uri">https://www.R-project.org/</a>.</p></li>
<li><p>Stoudt, Sara, Anthony D. Scotina, and Karsten Luebke. “Supporting
Statistics and Data Science Education with learnr.” Technology
Innovations in Statistics Education 14.1 (2022).</p></li>
<li><p>Wickham, Hadley. “ggplot2.” Wiley interdisciplinary reviews:
computational statistics 3.2 (2011): 180-185.</p></li>
</ul>
<!-- ADDITIONAL UNUSED CODE -->
<!-- #DA Barplot code (requires miso_counts_unpivot table) ########################-->
<!-- miso_counts <- readRDS("miso_counts_unpivot.RDS") -->
<!-- ASVs_for_subset <- row.names(Significant_DEseq2_ASVs) -->
<!-- unpivot_keep_ASVs <- miso_unpivot[miso_unpivot$ASV %in% Keep_ASVs, ] -->
<!-- ggplot(unpivot_keep_ASVs, aes(fill=subject, y=count, x=ASV)) + -->
<!--     geom_bar(position="dodge", stat="identity")+ -->
<!--   ggtitle("WO vs HD") -->
<!-- #Heatmap ###################### -->
<!-- #prune_taxa to subset DA ASVs only -->
<!-- Keep_ASVs<-row.names(Significant_DEseq2_ASVs) -->
<!-- miso_counts_DA <- prune_taxa(Keep_ASVs, miso_counts) -->
<!-- #Remove rows with no ASV counts (prevents creation of Bray-Curtis matrix) -->
<!-- miso_counts_DA <- prune_samples(sample_sums(miso_counts_DA) > 0, miso_counts_DA) -->
<!-- #Different heatmaps -->
<!-- plot_heatmap(miso_counts_DA, "PCoA", "bray", title="WO vs HD", sample.label="subject", taxa.label=NULL) -->
<!-- plot_heatmap(miso_counts_DA, "PCoA", "bray", "Phylum", title="WO vs HD") -->
<!-- # Change axes orders -->
<!-- sampleOrder = unique(sample_names(miso_counts_DA)) -->
<!-- taxaOrder = unique(taxa_names(miso_counts_DA)) -->
<!-- plot_heatmap(miso_counts_DA, "PCoA", "bray", title="WO vs HD", sample.label="", sample.order = sampleOrder, taxa.order=taxaOrder) -->
<!-- plot_heatmap(miso_counts_DA, "PCoA", "bray", sample.label="timepoint", "Phylum") -->
<!-- plot_heatmap(miso_counts_DA, method = "PCoA", "bray", taxa.label="Phylum", low="blue", high="red", title="My title here") -->
<!-- # Tax glom heatmap -->
<!-- miso_glom<-tax_glom(miso_counts, taxrank="Phylum", NArm=TRUE, bad_empty=c(NA, "", " ", "\t")) -->
<!-- plot_bar(miso_glom, x="timepointgroup", fill = "Phylum", title = "choose a name for your graph") + -->
<!--   geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")  -->
<!-- PERMANOVA CODE -->
<!-- permanova_results <- adonis2(distance(ps_arch.prop, method="bray", permutations=9999) ~ species*location, data = samdf) -->
<!-- permanova_results -->
<p>
<script type="application/shiny-prerendered" data-context="server-start">
#Load learnr
library(learnr)
knitr::opts_chunk$set(echo = FALSE)
tutorial_options(exercise.completion=TRUE) #Uncomment to enable 

#Load libraries
library("tidyverse")
library("phyloseq")
library("DESeq2")

#Uncomment exercise timelimit when deployed
tutorial_options(exercise.timelimit = 360)

#Load MISO_16S.prop.RDS
#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds
#Normalization by turning counts into proportions

# #Local path
miso <- readRDS("MISO_16S.prop.clean.RDS")
miso_counts <- readRDS("MISO_16S.clean.RDS")

#SciServer paths
#miso <- readRDS( "/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds" )
#miso_counts <- readRDS("/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds")

# <!-- EXERCISE = should we see an interactive code chunk with this code? -->
# <!-- EVAL = should you run this code? -->
# <!-- ECHO = should you print the code? -->
# Significant_DEseq2_ASVs <-0
# sigtab<-0

Differential_Abundance <- function(x, y, z) {
  
 # Accept the input and give it a human friendly name
  my_DESeq2_object <- x
  my_comparision_list <- y
  my_pvalue <- z
  
  
  #Do the DESeq2 analysis
  my_DESeq2_object <- DESeq(my_DESeq2_object, test="Wald", fitType="parametric")
 
  #Change the contrast section to compare different variables and groups.
  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)
  
  #Select only abundances with a p-value of z or below
  alpha = my_pvalue
  sigtab = res[which(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(miso_counts)[rownames(sigtab), ], "matrix"))

  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results
  theme_set(theme_bw())
  scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
  }
  # Phylum order
  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))
  
  # Class order
  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))
  
  # Order order
  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
  
  # Family order
  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
  
  # Genus order
  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
  
  # Species order
  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))

  return(sigtab)
}


Differential_Abundance_Continuous <- function(x, y, z) {
  
 # Accept the input and give it a human friendly name
  my_DESeq2_object <- x
  my_name <- y
  my_pvalue <- z
  
  
  #Do the DESeq2 analysis
  my_DESeq2_object <- DESeq(my_DESeq2_object, test="Wald", fitType="parametric")
 
  #Change the contrast section to compare different variables and groups.
  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)
  
  
  #Select only abundances with a p-value of z or below
  alpha = my_pvalue
  sigtab = res[which(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(miso_counts)[rownames(sigtab), ], "matrix"))

  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results
  theme_set(theme_bw())
  scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
  }
  # Phylum order
  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))
  
  # Class order
  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))
  
  # Order order
  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
  
  # Family order
  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
  
  # Genus order
  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
  
  # Species order
  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))

  return(sigtab)
}

</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::register_http_handlers(session, metadata = NULL)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::prepare_tutorial_state(session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::i18n_observe_tutorial_language(input, session)
</script>


<script type="application/shiny-prerendered" data-context="server">
session$onSessionEnded(function() {
        learnr:::event_trigger(session, "session_stop")
      })
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-plot_richness_example-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-plot_richness_example-code-editor`)), session)
output$`tutorial-exercise-plot_richness_example-output` <- renderUI({
  `tutorial-exercise-plot_richness_example-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "plot_richness_example", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "plot_richness_example", 
    code = "plot_richness(miso_counts, x=\"subject\", \n              color=\"timepointgroup\", \n              shape=\"timepointgroup\", \n          \t  title =\"fill in the blank\",\n              measures=\"Simpson\")\n", 
    opts = list(label = "\"plot_richness_example\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "plot_richness_example", exercise = TRUE, code = c("plot_richness(miso_counts, x=\"subject\", ", 
        "              color=\"timepointgroup\", ", "              shape=\"timepointgroup\", ", 
        "          \t  title =\"fill in the blank\",", "              measures=\"Simpson\")", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "plot_richness_example, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-plot_richness_subset_example-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-plot_richness_subset_example-code-editor`)), session)
output$`tutorial-exercise-plot_richness_subset_example-output` <- renderUI({
  `tutorial-exercise-plot_richness_subset_example-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "plot_richness_subset_example", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "plot_richness_subset_example", 
    code = "\nmy_subset <- subset_samples(miso_counts, gender == \"F\")\n\nplot_richness(my_subset, x=\"timepoint\", \n              color=\"phase\", \n              shape=\"timepointgroup\", \n          \t  title =\"fill in the blank\",\n              measures=\"Simpson\")\n\n", 
    opts = list(label = "\"plot_richness_subset_example\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "plot_richness_subset_example", exercise = TRUE, 
        code = c("", "my_subset <- subset_samples(miso_counts, gender == \"F\")", 
        "", "plot_richness(my_subset, x=\"timepoint\", ", "              color=\"phase\", ", 
        "              shape=\"timepointgroup\", ", "          \t  title =\"fill in the blank\",", 
        "              measures=\"Simpson\")", "", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "plot_richness_subset_example, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "alpha_diversity_quiz-1", 
    question = structure("For subject S02, which timepointgroup has the highest alpha diveristy?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_f3e72c", 
        option = "BD", value = "BD", label = structure("BD", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_5edff68", 
        option = "HD", value = "HD", label = structure("HD", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_1f31721", 
        option = "WO", value = "WO", label = structure("WO", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "alpha_diversity_quiz-1-answer", question = "alpha_diversity_quiz-1"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 1842454629.14204, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "alpha_diversity_quiz-2", 
    question = structure("What subject has the data point for the LOWEST alpha diversity when looking only at ASVs from the phylum Firmicutes? HINT: Don’t forget to change subset_samples() to subset_taxa().", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_9df08cd", 
        option = "S21", value = "S21", label = structure("S21", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_2ef29f", 
        option = "S12", value = "S12", label = structure("S12", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_dd6bd60", 
        option = "S31", value = "S31", label = structure("S31", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_e9918b6", 
        option = "S02", value = "S02", label = structure("S02", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_125ea19", 
        option = "S15", value = "S15", label = structure("S15", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "alpha_diversity_quiz-2-answer", question = "alpha_diversity_quiz-2"), 
    loading = NULL, random_answer_order = TRUE, allow_retry = TRUE, 
    seed = 1228785903.4278, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ordinate_PCoA-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ordinate_PCoA-code-editor`)), session)
output$`tutorial-exercise-ordinate_PCoA-output` <- renderUI({
  `tutorial-exercise-ordinate_PCoA-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ordinate_PCoA", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ordinate_PCoA", 
    code = "miso.pcoa <- ordinate(miso, method=\"PCoA\", distance=\"bray\")\nhead(miso.pcoa)", 
    opts = list(label = "\"ordinate_PCoA\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ordinate_PCoA", exercise = TRUE, code = c("miso.pcoa <- ordinate(miso, method=\"PCoA\", distance=\"bray\")", 
        "head(miso.pcoa)"), out.width.px = 624, out.height.px = 384, 
        params.src = "ordinate_PCoA, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-plot_PCoA-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-plot_PCoA-code-editor`)), session)
output$`tutorial-exercise-plot_PCoA-output` <- renderUI({
  `tutorial-exercise-plot_PCoA-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "plot_PCoA", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = "miso.pcoa <- ordinate(miso, method=\"PCoA\", distance=\"bray\")\nhead(miso.pcoa)", 
    chunks = list(list(label = "ordinate_PCoA", code = "miso.pcoa <- ordinate(miso, method=\"PCoA\", distance=\"bray\")\nhead(miso.pcoa)", 
        opts = list(label = "\"ordinate_PCoA\"", exercise = "TRUE", 
            echo = "FALSE"), engine = "r"), list(label = "plot_PCoA", 
        code = "plot_ordination(miso, miso.pcoa, \n        color = \"subject\", \n        shape = \"phase\", \n        title=\"My PCoA\")", 
        opts = list(label = "\"plot_PCoA\"", exercise = "TRUE", 
            echo = "FALSE", exercise.setup = "\"ordinate_PCoA\""), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "plot_PCoA", exercise = TRUE, exercise.setup = "ordinate_PCoA", 
        code = c("plot_ordination(miso, miso.pcoa, ", "        color = \"subject\", ", 
        "        shape = \"phase\", ", "        title=\"My PCoA\")"
        ), out.width.px = 624, out.height.px = 384, params.src = "plot_PCoA, exercise=TRUE, echo=FALSE, exercise.setup = \"ordinate_PCoA\"", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-subset_PCoA-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-subset_PCoA-code-editor`)), session)
output$`tutorial-exercise-subset_PCoA-output` <- renderUI({
  `tutorial-exercise-subset_PCoA-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "subset_PCoA", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "subset_PCoA", 
    code = "#Subset a group of interest\nmiso_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")\n\n#Ordinate based on your subset\nmiso.pcoa.subset <- ordinate(miso_subset, method=\"PCoA\", distance=\"bray\")\n\n#Plot the new subsetted ordination with the subsetted phyloseq object\nplot_ordination(miso_subset, miso.pcoa.subset, \n        color = \"subject\", \n        shape = \"phase\", \n        title=\"My subsetted (Firmicutes) PCoA\")", 
    opts = list(label = "\"subset_PCoA\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "subset_PCoA", exercise = TRUE, code = c("#Subset a group of interest", 
        "miso_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")", 
        "", "#Ordinate based on your subset", "miso.pcoa.subset <- ordinate(miso_subset, method=\"PCoA\", distance=\"bray\")", 
        "", "#Plot the new subsetted ordination with the subsetted phyloseq object", 
        "plot_ordination(miso_subset, miso.pcoa.subset, ", "        color = \"subject\", ", 
        "        shape = \"phase\", ", "        title=\"My subsetted (Firmicutes) PCoA\")"
        ), out.width.px = 624, out.height.px = 384, params.src = "subset_PCoA, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "pcoa_quiz-1", 
    question = structure("In a PCoA plot, samples that are farther apart…", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_cb3e4c7", 
        option = "Have less in common than samples that are closer together", 
        value = "Have less in common than samples that are closer together", 
        label = structure("Have less in common than samples that are closer together", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_90745c6", 
        option = "Have more extreme compositions", value = "Have more extreme compositions", 
        label = structure("Have more extreme compositions", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_a2f9c4e", 
        option = "Have more ASVs", value = "Have more ASVs", 
        label = structure("Have more ASVs", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "pcoa_quiz-1-answer", question = "pcoa_quiz-1"), 
    loading = NULL, random_answer_order = TRUE, allow_retry = TRUE, 
    seed = 573883583.232765, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "pcoa_quiz-2", 
    question = structure("Looking at the PCoA with all data included, what other subject is subject 28 (purple, top) most similar to?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_6ad4b4e", 
        option = "S09", value = "S09", label = structure("S09", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_3546733", 
        option = "S24", value = "S24", label = structure("S24", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_76aab7b", 
        option = "S18", value = "S18", label = structure("S18", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_7ed01e9", 
        option = "S12", value = "S12", label = structure("S12", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_5d1b380", 
        option = "S32", value = "S32", label = structure("S32", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "pcoa_quiz-2-answer", question = "pcoa_quiz-2"), 
    loading = NULL, random_answer_order = TRUE, allow_retry = TRUE, 
    seed = 1917942728.10689, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "pcoa_quiz-3", 
    question = structure("In a PCoA with only data from ASVs with the class Bacteroidia, what is the percent of variance in the dataset explained by principal coordinate 1? HINT: You will need to change the code subsetting phylum and Firmicutes", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_1a7a81", 
        option = "22.5%", value = "22.5%", label = structure("22.5%", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_33a78d3", 
        option = "12.8%", value = "12.8%", label = structure("12.8%", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_892b5b2", 
        option = "13.1%", value = "13.1%", label = structure("13.1%", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_a72a433", 
        option = "6.2%", value = "6.2%", label = structure("6.2%", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_34a1f75", 
        option = "11.7%", value = "11.7%", label = structure("11.7%", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "pcoa_quiz-3-answer", question = "pcoa_quiz-3"), 
    loading = NULL, random_answer_order = TRUE, allow_retry = TRUE, 
    seed = 884765384.087999, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-phyloseq_to_deseq2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-phyloseq_to_deseq2-code-editor`)), session)
output$`tutorial-exercise-phyloseq_to_deseq2-output` <- renderUI({
  `tutorial-exercise-phyloseq_to_deseq2-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "phyloseq_to_deseq2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "phyloseq_to_deseq2", 
    code = "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\nmy_miso_DESeq2", 
    opts = list(label = "\"phyloseq_to_deseq2\"", exercise = "TRUE"), 
    engine = "r")), code_check = NULL, error_check = NULL, check = NULL, 
    solution = NULL, tests = NULL, options = list(eval = FALSE, 
        echo = TRUE, results = "markup", tidy = FALSE, tidy.opts = NULL, 
        collapse = FALSE, prompt = FALSE, comment = NA, highlight = FALSE, 
        size = "normalsize", background = "#F7F7F7", strip.white = TRUE, 
        cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "phyloseq_to_deseq2", exercise = TRUE, code = c("my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)", 
        "my_miso_DESeq2"), out.width.px = 624, out.height.px = 384, 
        params.src = "phyloseq_to_deseq2, exercise=TRUE", fig.num = 0, 
        exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-create_my_comparision-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-create_my_comparision-code-editor`)), session)
output$`tutorial-exercise-create_my_comparision-output` <- renderUI({
  `tutorial-exercise-create_my_comparision-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "create_my_comparision", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "create_my_comparision", 
    code = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_comparision", 
    opts = list(label = "\"create_my_comparision\"", exercise = "TRUE", 
        eval = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "create_my_comparision", exercise = TRUE, code = c("my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")", 
        "my_comparision"), out.width.px = 624, out.height.px = 384, 
        params.src = "create_my_comparision, exercise=TRUE, eval=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-run_differential_abundance-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-run_differential_abundance-code-editor`)), session)
output$`tutorial-exercise-run_differential_abundance-output` <- renderUI({
  `tutorial-exercise-run_differential_abundance-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "run_differential_abundance", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n", 
    chunks = list(list(label = "prepare-forDESeq2", code = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n", 
        opts = list(label = "\"prepare-forDESeq2\""), engine = "r"), 
        list(label = "run_differential_abundance", code = "Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)\nSignificant_DEseq2_ASVs", 
            opts = list(label = "\"run_differential_abundance\"", 
                exercise = "TRUE", eval = "FALSE", exercise.setup = "\"prepare-forDESeq2\""), 
            engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "run_differential_abundance", exercise = TRUE, 
        exercise.setup = "prepare-forDESeq2", code = c("Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        "Significant_DEseq2_ASVs"), out.width.px = 624, out.height.px = 384, 
        params.src = "run_differential_abundance, exercise=TRUE, eval=FALSE, exercise.setup= \"prepare-forDESeq2\"", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-significant_DEseq2_genes-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-significant_DEseq2_genes-code-editor`)), session)
output$`tutorial-exercise-significant_DEseq2_genes-output` <- renderUI({
  `tutorial-exercise-significant_DEseq2_genes-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "significant_DEseq2_genes", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
    chunks = list(list(label = "prepare-forDESeq2-2", code = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        opts = list(label = "\"prepare-forDESeq2-2\""), engine = "r"), 
        list(label = "significant_DEseq2_genes", code = "Significant_DEseq2_ASVs", 
            opts = list(label = "\"significant_DEseq2_genes\"", 
                exercise = "TRUE", eval = "FALSE", exercise.setup = "\"prepare-forDESeq2-2\""), 
            engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "significant_DEseq2_genes", exercise = TRUE, 
        exercise.setup = "prepare-forDESeq2-2", code = "Significant_DEseq2_ASVs", 
        out.width.px = 624, out.height.px = 384, params.src = "significant_DEseq2_genes, exercise=TRUE, eval=FALSE, exercise.setup=\"prepare-forDESeq2-2\"", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-plot_deseq2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-plot_deseq2-code-editor`)), session)
output$`tutorial-exercise-plot_deseq2-output` <- renderUI({
  `tutorial-exercise-plot_deseq2-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "plot_deseq2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
    chunks = list(list(label = "prepare-forDESeq2-2", code = "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        opts = list(label = "\"prepare-forDESeq2-2\""), engine = "r"), 
        list(label = "plot_deseq2", code = "ggplot(Significant_DEseq2_ASVs, aes(x=Class, y=log2FoldChange, color=Class)) + geom_point(size=3) + \n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))", 
            opts = list(label = "\"plot_deseq2\"", exercise = "TRUE", 
                eval = "FALSE", exercise.setup = "\"prepare-forDESeq2-2\""), 
            engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "plot_deseq2", exercise = TRUE, exercise.setup = "prepare-forDESeq2-2", 
        code = c("ggplot(Significant_DEseq2_ASVs, aes(x=Class, y=log2FoldChange, color=Class)) + geom_point(size=3) + ", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))"
        ), out.width.px = 624, out.height.px = 384, params.src = "plot_deseq2, exercise=TRUE, eval=FALSE, exercise.setup=\"prepare-forDESeq2-2\"", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-DESeq2_example-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-DESeq2_example-code-editor`)), session)
output$`tutorial-exercise-DESeq2_example-output` <- renderUI({
  `tutorial-exercise-DESeq2_example-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "DESeq2_example", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "DESeq2_example", 
    code = "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design \nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n\n# STEP 2: Select the groups to compare\nmy_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+\n  ggtitle(\"my title here\")\n", 
    opts = list(label = "\"DESeq2_example\"", exercise = "TRUE", 
        echo = "TRUE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "DESeq2_example", exercise = TRUE, code = c("# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design ", 
        "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)", 
        "", "# STEP 2: Select the groups to compare", "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+", 
        "  ggtitle(\"my title here\")", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "DESeq2_example, exercise=TRUE, echo=TRUE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz-1", 
    question = structure("How many differentially abundant ASVs were found between the WO and BD conditions?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_2885b0e", 
        option = "7", value = "7", label = structure("7", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_11b2df", 
        option = "10", value = "10", label = structure("10", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_639a623", 
        option = "32", value = "32", label = structure("32", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_b0103f2", 
        option = "11", value = "11", label = structure("11", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_79e902b", 
        option = "9", value = "9", label = structure("9", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz-1-answer", question = "differential_abundance_quiz-1"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 2124984144.01048, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz-2", 
    question = structure("Does ASV 2114 have a greater abundance in WO or BD?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_76679ad", 
        option = "BD", value = "BD", label = structure("BD", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_9fd3b9", 
        option = "WO", value = "WO", label = structure("WO", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz-2-answer", question = "differential_abundance_quiz-2"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 1348797618.87192, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz-3", 
    question = structure("Of all the ASVs found to be differentially abundant between WO and BD, which has the MOST differential abundance between the WO and BD conditions? HINT: Look at Log2FoldChange", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_fd0cd8a", 
        option = "ASV245", value = "ASV245", label = structure("ASV245", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_504d3aa", 
        option = "ASV1248", value = "ASV1248", label = structure("ASV1248", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_b43738c", 
        option = "ASV1693", value = "ASV1693", label = structure("ASV1693", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_5bdd6b2", 
        option = "ASV2074", value = "ASV2074", label = structure("ASV2074", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_e2f49cb", 
        option = "ASV2114", value = "ASV2114", label = structure("ASV2114", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz-3-answer", question = "differential_abundance_quiz-3"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 611746820.215133, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz_part2-1", 
    question = structure("How many phylum have differentially abundant ASVs between WO and BD?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_d86e28f", 
        option = "3", value = "3", label = structure("3", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_d2367c0", 
        option = "1", value = "1", label = structure("1", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_ed38b2a", 
        option = "0", value = "0", label = structure("0", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_5765713", 
        option = "4", value = "4", label = structure("4", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz_part2-1-answer", 
        question = "differential_abundance_quiz_part2-1"), loading = NULL, 
    random_answer_order = TRUE, allow_retry = TRUE, seed = 876587529.091807, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-DESeq2_example_comparison-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-DESeq2_example_comparison-code-editor`)), session)
output$`tutorial-exercise-DESeq2_example_comparison-output` <- renderUI({
  `tutorial-exercise-DESeq2_example_comparison-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "DESeq2_example_comparison", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "DESeq2_example_comparison", 
    code = "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design \nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n\n# STEP 2: Select the groups to compare\nmy_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+\n  ggtitle(\"my title here\")\n", 
    opts = list(label = "\"DESeq2_example_comparison\"", exercise = "TRUE", 
        echo = "TRUE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "DESeq2_example_comparison", exercise = TRUE, 
        code = c("# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design ", 
        "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)", 
        "", "# STEP 2: Select the groups to compare", "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+", 
        "  ggtitle(\"my title here\")", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "DESeq2_example_comparison, exercise=TRUE, echo=TRUE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz_comparison-1", 
    question = structure("How many differentially abundant ASVs are there between HD and BD?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_77a9b1b", 
        option = "6", value = "6", label = structure("6", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_bb11f10", 
        option = "4", value = "4", label = structure("4", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_3658c3c", 
        option = "9", value = "9", label = structure("9", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_ed39471", 
        option = "10", value = "10", label = structure("10", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_96df621", 
        option = "2", value = "2", label = structure("2", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz_comparison-1-answer", 
        question = "differential_abundance_quiz_comparison-1"), 
    loading = NULL, random_answer_order = TRUE, allow_retry = TRUE, 
    seed = 1827691944.14891, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz_comparison-2", 
    question = structure("Does ASV807 a higher abundance in HD or BD?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_9f1d0fe", 
        option = "HD", value = "HD", label = structure("HD", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_f497107", 
        option = "BD", value = "BD", label = structure("BD", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz_comparison-2-answer", 
        question = "differential_abundance_quiz_comparison-2"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 1499400928.30179, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-DESeq2_example_design-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-DESeq2_example_design-code-editor`)), session)
output$`tutorial-exercise-DESeq2_example_design-output` <- renderUI({
  `tutorial-exercise-DESeq2_example_design-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "DESeq2_example_design", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "DESeq2_example_design", 
    code = "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design \nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n\n# STEP 2: Select the groups to compare\nmy_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+\n  ggtitle(\"my title here\")\n", 
    opts = list(label = "\"DESeq2_example_design\"", exercise = "TRUE", 
        echo = "TRUE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "DESeq2_example_design", exercise = TRUE, code = c("# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design ", 
        "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)", 
        "", "# STEP 2: Select the groups to compare", "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+", 
        "  ggtitle(\"my title here\")", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "DESeq2_example_design, exercise=TRUE, echo=TRUE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "differential_abundance_quiz_design-1", 
    question = structure("In our example, which is the baseline condition?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_fb54385", 
        option = "S03", value = "S03", label = structure("S03", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_6397d24", 
        option = "S02", value = "S02", label = structure("S02", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz_design-1-answer", 
        question = "differential_abundance_quiz_design-1"), loading = NULL, 
    random_answer_order = TRUE, allow_retry = TRUE, seed = 465814207.783088, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_checkbox", label = "differential_abundance_quiz_design-2", 
    question = structure("Which of the following Phylum have ASVs that are differentially abundant between the subject S02 and subject S03?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_89c2386", 
        option = "Firmicutes", value = "Firmicutes", label = structure("Firmicutes", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_2049b39", 
        option = "Bacteroidetes", value = "Bacteroidetes", label = structure("Bacteroidetes", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_8b352d6", 
        option = "Proteobacteria", value = "Proteobacteria", 
        label = structure("Proteobacteria", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_9ff9a7e", 
        option = "Actinobacteria", value = "Actinobacteria", 
        label = structure("Actinobacteria", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_8a471f1", 
        option = "Verrucomicrobia", value = "Verrucomicrobia", 
        label = structure("Verrucomicrobia", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_1518ef3", 
        option = "Tenericutes", value = "Tenericutes", label = structure("Tenericutes", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = NULL, type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect. Be sure to select every correct answer.", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "differential_abundance_quiz_design-2-answer", 
        question = "differential_abundance_quiz_design-2"), loading = NULL, 
    random_answer_order = TRUE, allow_retry = TRUE, seed = 14195860.4933895, 
    options = list()), class = c("learnr_checkbox", "tutorial_question"
)), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_end-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_end-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_end-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_end-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_end", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_end", 
    code = "#Subset a group of interest\nmiso_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")\n\n#Ordinate based on your subset\nmiso.pcoa.subset <- ordinate(miso, method=\"PCoA\", distance=\"bray\")\n\n#Plot the new subsetted ordination with the subsetted phyloseq object\nplot_ordination(miso_subset, miso.pcoa.subset, \n        color = \"subject\", \n        shape = \"phase\", \n        title=\"My subsetted (Firmicutes) PCoA\")", 
    opts = list(label = "\"Try_it_PCoA_end\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_end", exercise = TRUE, code = c("#Subset a group of interest", 
        "miso_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")", 
        "", "#Ordinate based on your subset", "miso.pcoa.subset <- ordinate(miso, method=\"PCoA\", distance=\"bray\")", 
        "", "#Plot the new subsetted ordination with the subsetted phyloseq object", 
        "plot_ordination(miso_subset, miso.pcoa.subset, ", "        color = \"subject\", ", 
        "        shape = \"phase\", ", "        title=\"My subsetted (Firmicutes) PCoA\")"
        ), out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_end, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_plot_richness_end-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_plot_richness_end-code-editor`)), session)
output$`tutorial-exercise-try_it_plot_richness_end-output` <- renderUI({
  `tutorial-exercise-try_it_plot_richness_end-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_plot_richness_end", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_plot_richness_end", 
    code = "\nmy_subset <- subset_samples(miso_counts, gender == \"F\")\n\nplot_richness(my_subset, x=\"timepoint\", \n              color=\"phase\", \n              shape=\"timepointgroup\",\n          \t  title =\"fill in the blank\", \n              measures= \"Simpson\")\n", 
    opts = list(label = "\"try_it_plot_richness_end\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_plot_richness_end", exercise = TRUE, 
        code = c("", "my_subset <- subset_samples(miso_counts, gender == \"F\")", 
        "", "plot_richness(my_subset, x=\"timepoint\", ", "              color=\"phase\", ", 
        "              shape=\"timepointgroup\",", "          \t  title =\"fill in the blank\", ", 
        "              measures= \"Simpson\")", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "try_it_plot_richness_end, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_DESeq2_end-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_DESeq2_end-code-editor`)), session)
output$`tutorial-exercise-try_it_DESeq2_end-output` <- renderUI({
  `tutorial-exercise-try_it_DESeq2_end-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_DESeq2_end", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", 
"", "#SciServer paths", "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
"#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
"", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_DESeq2_end", 
    code = "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design \nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n\n# STEP 2: Select the groups to compare\nmy_comparision <-c(\"timepointgroup\", \"WO\", \"HD\")\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+\n  ggtitle(\"my title here\")", 
    opts = list(label = "\"try_it_DESeq2_end\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_DESeq2_end", exercise = TRUE, code = c("# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design ", 
        "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)", 
        "", "# STEP 2: Select the groups to compare", "my_comparision <-c(\"timepointgroup\", \"WO\", \"HD\")", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3) +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+", 
        "  ggtitle(\"my title here\")"), out.width.px = 624, 
        out.height.px = 384, params.src = "try_it_DESeq2_end, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_differential_abundance_blank-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_differential_abundance_blank-code-editor`)), session)
output$`tutorial-exercise-Try_it_differential_abundance_blank-output` <- renderUI({
  `tutorial-exercise-Try_it_differential_abundance_blank-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_differential_abundance_blank", 
    global_setup = structure(c("#Load learnr", "library(learnr)", 
    "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
    "", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
    "library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
    "tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
    "#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
    "#Normalization by turning counts into proportions", "", 
    "# #Local path", "miso <- readRDS(\"MISO_16S.prop.clean.RDS\")", 
    "miso_counts <- readRDS(\"MISO_16S.clean.RDS\")", "", "#SciServer paths", 
    "#miso <- readRDS( \"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds\" )", 
    "#miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.clean.rds\")", 
    "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
    "# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
    "# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
    "  ", " # Accept the input and give it a human friendly name", 
    "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
    "  my_pvalue <- z", "  ", "  ", "  #Do the DESeq2 analysis", 
    "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
    " ", "  #Change the contrast section to compare different variables and groups.", 
    "  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
    "  ", "  #Select only abundances with a p-value of z or below", 
    "  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
    "  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
    "", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
    "  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
    "    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
    "  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
    "  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
    "  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
    "  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
    "  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
    "  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
    "", "  return(sigtab)", "}", "", "", "Differential_Abundance_Continuous <- function(x, y, z) {", 
    "  ", " # Accept the input and give it a human friendly name", 
    "  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
    "  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
    " ", "  #Change the contrast section to compare different variables and groups.", 
    "  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
    "  ", "  ", "  #Select only abundances with a p-value of z or below", 
    "  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
    "  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
    "", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
    "  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
    "    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
    "  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
    "  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
    "  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
    "  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
    "  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
    "  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
    "  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
    "", "  return(sigtab)", "}", ""), chunk_opts = list(label = "setup", 
        include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_differential_abundance_blank", 
        code = "", opts = list(label = "\"Try_it_differential_abundance_blank\"", 
            exercise = "TRUE", echo = "FALSE"), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "analyze-16s-rrna-data-with-phyloseq_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "analyze-16s-rrna-data-with-phyloseq_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_differential_abundance_blank", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_differential_abundance_blank, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
</p>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["header-attrs"]},{"type":"character","attributes":{},"value":["2.29"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pandoc"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["header-attrs.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/cerulean.min.css"]},{"type":"character","attributes":{},"value":["<style>h1 {font-size: 34px;}\n       h1.title {font-size: 38px;}\n       h2 {font-size: 30px;}\n       h3 {font-size: 24px;}\n       h4 {font-size: 18px;}\n       h5 {font-size: 16px;}\n       h6 {font-size: 12px;}\n       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}\n       pre:not([class]) { background-color: white }<\/style>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["pagedtable"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pagedtable-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["js/pagedtable.js"]},{"type":"character","attributes":{},"value":["css/pagedtable.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["textmate.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.11.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["i18n"]},{"type":"character","attributes":{},"value":["21.6.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/i18n"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["i18next.min.js","tutorial-i18n-init.js"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["<script id=\"i18n-cstm-trns\" type=\"application/json\">{\"language\":\"en\",\"resources\":{\"en\":{\"translation\":{\"button\":{\"runcode\":\"Run Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Hint\",\"hint_plural\":\"Hints\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Next Hint\",\"hintprev\":\"Previous Hint\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copy to Clipboard\",\"startover\":\"Start Over\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continue\",\"submitanswer\":\"Submit Answer\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Previous Topic\",\"nexttopic\":\"Next Topic\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Try Again\"},\"text\":{\"startover\":\"Start Over\",\"areyousure\":\"Are you sure you want to start over? (all exercise progress will be reset)\",\"youmustcomplete\":\"You must complete the\",\"exercise\":\"exercise\",\"exercise_plural\":\"exercises\",\"inthissection\":\"in this section before continuing.\",\"code\":\"Code\",\"enginecap\":\"{{engine}} $t(text.code)\",\"quiz\":\"Quiz\",\"blank\":\"blank\",\"blank_plural\":\"blanks\",\"exercisecontainsblank\":\"This exercise contains {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Please replace {{blank}} with valid code.\",\"unparsable\":\"It looks like this might not be valid R code. R cannot determine how to turn your text into a complete command. You may have forgotten to fill in a blank, to remove an underscore, to include a comma between arguments, or to close an opening <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> or <code>{<\\/code> with a matching <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> or <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>It looks like your R code contains specially formatted quotation marks or &quot;curly&quot; quotes (<code>{{character}}<\\/code>) around character strings, making your code invalid. R requires character values to be contained in straight quotation marks (<code>&quot;<\\/code> or <code>'<\\/code>).<\\/p> {{code}} <p>Don't worry, this is a common source of errors when you copy code from another app that applies its own formatting to text. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. Try deleting the special character from your code and retyping it manually.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"and\":\"and\",\"or\":\"or\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"fr\":{\"translation\":{\"button\":{\"runcode\":\"Lancer le Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Indication\",\"hint_plural\":\"Indications\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Indication Suivante\",\"hintprev\":\"Indication Précédente\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copier dans le Presse-papier\",\"startover\":\"Recommencer\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuer\",\"submitanswer\":\"Soumettre\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Chapitre Précédent\",\"nexttopic\":\"Chapitre Suivant\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Réessayer\"},\"text\":{\"startover\":\"Recommencer\",\"areyousure\":\"Êtes-vous certains de vouloir recommencer? (La progression sera remise à zéro)\",\"youmustcomplete\":\"Vous devez d'abord compléter\",\"exercise\":\"l'exercice\",\"exercise_plural\":\"des exercices\",\"inthissection\":\"de cette section avec de continuer.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"et\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"es\":{\"translation\":{\"button\":{\"runcode\":\"Ejecutar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Pista\",\"hint_plural\":\"Pistas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Siguiente pista\",\"hintprev\":\"Pista anterior\",\"solution\":\"Solución\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar al portapapeles\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar respuesta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tema anterior\",\"nexttopic\":\"Tema siguiente\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Volver a intentar\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"¿De verdad quieres empezar de nuevo? (todo el progreso del ejercicio se perderá)\",\"youmustcomplete\":\"Debes completar\",\"exercise\":\"el ejercicio\",\"exercise_plural\":\"los ejercicios\",\"inthissection\":\"en esta sección antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Cuestionario\",\"and\":\"y\",\"or\":\"o\",\"oxfordcomma\":\"\"}}},\"pt\":{\"translation\":{\"button\":{\"runcode\":\"Executar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Dica\",\"hint_plural\":\"Dicas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Próxima dica\",\"hintprev\":\"Dica anterior\",\"solution\":\"Solução\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar para a área de transferência\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar resposta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tópico anterior\",\"nexttopic\":\"Próximo tópico\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tentar novamente\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"Tem certeza que deseja começar novamente? (todo o progresso feito será perdido)\",\"youmustcomplete\":\"Você deve completar\",\"exercise\":\"o exercício\",\"exercise_plural\":\"os exercícios\",\"inthissection\":\"nesta seção antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"e\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"tr\":{\"translation\":{\"button\":{\"runcode\":\"Çalıştırma Kodu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Ipucu\",\"hint_plural\":\"İpuçları\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Sonraki İpucu\",\"hintprev\":\"Önceki İpucu\",\"solution\":\"Çözüm\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Pano'ya Kopyala\",\"startover\":\"Baştan Başlamak\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Devam et\",\"submitanswer\":\"Cevabı onayla\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Önceki Konu\",\"nexttopic\":\"Sonraki Konu\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tekrar Deneyin\"},\"text\":{\"startover\":\"Baştan Başlamak\",\"areyousure\":\"Baştan başlamak istediğinizden emin misiniz? (tüm egzersiz ilerlemesi kaybolacak)\",\"youmustcomplete\":\"Tamamlamalısın\",\"exercise\":\"egzersiz\",\"exercise_plural\":\"egzersizler\",\"inthissection\":\"devam etmeden önce bu bölümde\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Sınav\",\"oxfordcomma\":\"\"}}},\"emo\":{\"translation\":{\"button\":{\"runcode\":\"🏃\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"💡\",\"hint_plural\":\"$t(button.hint)\",\"hinttitle\":\"$t(button.hint)\",\"solution\":\"🎯\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"📋\",\"startover\":\"⏮\",\"startovertitle\":\"Start Over\",\"continue\":\"✅\",\"submitanswer\":\"🆗\",\"submitanswertitle\":\"Submit Answer\",\"previoustopic\":\"⬅\",\"nexttopic\":\"➡\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"🔁\"},\"text\":{\"startover\":\"⏮\",\"areyousure\":\"🤔\",\"youmustcomplete\":\"⚠️ 👉 🧑‍💻\",\"exercise\":\"\",\"exercise_plural\":\"\",\"inthissection\":\"\",\"code\":\"💻\",\"enginecap\":\"$t(text.code) {{engine}}\",\"oxfordcomma\":\"\"}}},\"eu\":{\"translation\":{\"button\":{\"runcode\":\"Kodea egikaritu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Laguntza\",\"hint_plural\":\"Laguntza\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Aurreko laguntza\",\"hintprev\":\"Hurrengo laguntza\",\"solution\":\"Ebazpena\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Arbelean kopiatu\",\"startover\":\"Berrabiarazi\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Jarraitu\",\"submitanswer\":\"Erantzuna bidali\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Aurreko atala\",\"nexttopic\":\"Hurrengo atala\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Berriro saiatu\"},\"text\":{\"startover\":\"Berrabiarazi\",\"areyousure\":\"Berriro hasi nahi duzu? (egindako lana galdu egingo da)\",\"youmustcomplete\":\"Aurrera egin baino lehen atal honetako\",\"exercise\":\"ariketa egin behar duzu.\",\"exercise_plural\":\"ariketak egin behar dituzu.\",\"inthissection\":\"\",\"code\":\"Kodea\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Galdetegia\",\"oxfordcomma\":\"\"}}},\"de\":{\"translation\":{\"button\":{\"runcode\":\"Code ausführen\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Tipp\",\"hint_plural\":\"Tipps\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Nächster Tipp\",\"hintprev\":\"Vorheriger Tipp\",\"solution\":\"Lösung\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"In die Zwischenablage kopieren\",\"startover\":\"Neustart\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Weiter\",\"submitanswer\":\"Antwort einreichen\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Vorheriges Kapitel\",\"nexttopic\":\"Nächstes Kapitel\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Nochmal versuchen\"},\"text\":{\"startover\":\"Neustart\",\"areyousure\":\"Bist du sicher, dass du neustarten willst? (der gesamte Lernfortschritt wird gelöscht)\",\"youmustcomplete\":\"Vervollstädinge\",\"exercise\":\"die Übung\",\"exercise_plural\":\"die Übungen\",\"inthissection\":\"in diesem Kapitel, bevor du fortfährst.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"Lücke\",\"blank_plural\":\"Lücken\",\"pleasereplaceblank\":\"Bitte ersetze {{blank}} mit gültigem Code.\",\"unparsable\":\"Dies scheint kein gültiger R Code zu sein. R kann deinen Text nicht in einen gültigen Befehl übersetzen. Du hast vielleicht vergessen, die Lücke zu füllen, einen Unterstrich zu entfernen, ein Komma zwischen Argumente zu setzen oder ein eröffnendes <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> oder <code>{<\\/code> mit einem zugehörigen <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> oder <code>}<\\/code> zu schließen.\\n\",\"and\":\"und\",\"or\":\"oder\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"ko\":{\"translation\":{\"button\":{\"runcode\":\"코드 실행\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"힌트\",\"hint_plural\":\"힌트들\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"다음 힌트\",\"hintprev\":\"이전 힌트\",\"solution\":\"솔루션\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"클립보드에 복사\",\"startover\":\"재학습\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"다음 학습으로\",\"submitanswer\":\"정답 제출\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"이전 토픽\",\"nexttopic\":\"다음 토픽\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"재시도\"},\"text\":{\"startover\":\"재학습\",\"areyousure\":\"다시 시작 하시겠습니까? (모든 예제의 진행 정보가 재설정됩니다)\",\"youmustcomplete\":\"당신은 완료해야 합니다\",\"exercise\":\"연습문제\",\"exercise_plural\":\"연습문제들\",\"inthissection\":\"이 섹션을 실행하기 전에\",\"code\":\"코드\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"퀴즈\",\"blank\":\"공백\",\"blank_plural\":\"공백들\",\"exercisecontainsblank\":\"이 연습문제에는 {{count}}개의 $t(text.blank)이 포함되어 있습니다.\",\"pleasereplaceblank\":\"{{blank}}를 유효한 코드로 바꾸십시오.\",\"unparsable\":\"이것은 유효한 R 코드가 아닐 수 있습니다. R은 텍스트를 완전한 명령으로 변환하는 방법을 결정할 수 없습니다. 당신은 공백이나 밑줄을 대체하여 채우기, 인수를 컴마로 구분하기, 또는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> , <code>{<\\/code>로 시작하는 구문을 닫는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>, <code>}<\\/code>을 잊었을 수도 있습니다.\\n\",\"and\":\"그리고\",\"or\":\"혹은\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}},\"zh\":{\"translation\":{\"button\":{\"runcode\":\"运行代码\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"提示\",\"hint_plural\":\"提示\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"下一个提示\",\"hintprev\":\"上一个提示\",\"solution\":\"答案\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"复制到剪切板\",\"startover\":\"重新开始\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"继续\",\"submitanswer\":\"提交答案\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"上一专题\",\"nexttopic\":\"下一专题\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"再试一次\"},\"text\":{\"startover\":\"重置\",\"areyousure\":\"你确定要重新开始吗? (所有当前进度将被重置)\",\"youmustcomplete\":\"你必须完成\",\"exercise\":\"练习\",\"exercise_plural\":\"练习\",\"inthissection\":\"在进行本节之前\",\"code\":\"代码\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"测试\",\"blank\":\"空\",\"blank_plural\":\"空\",\"exercisecontainsblank\":\"本练习包含{{count}}个$t(text.blank)\",\"pleasereplaceblank\":\"请在{{blank}}内填写恰当的代码\",\"unparsable\":\"这似乎不是有效的R代码。 R不知道如何将您的文本转换为完整的命令。 您是否忘了填空，忘了删除下划线，忘了在参数之间包含逗号，或者是忘了用<code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>,<code>}<\\/code>来封闭<code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code>。 or <code>{<\\/code>。\\n\",\"unparsablequotes\":\"<p>您的R代码中似乎含有特殊格式的引号，或者弯引号(<code>{{character}}<\\/code>) 在字符串前后，在R中字符串应该被直引号(<code>&quot;<\\/code> 或者 <code>'<\\/code>)包裹。<\\/p> {{code}} <p>别担心，该错误经常在复制粘贴包含格式的代码时遇到， 您可以尝试将该行中的代码替换为以下代码，也许还有其他地方需要修改。<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"and\":\"且\",\"or\":\"或\",\"listcomma\":\",\",\"oxfordcomma\":\",\"}}},\"pl\":{\"translation\":{\"button\":{\"runcode\":\"Uruchom kod\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Podpowiedź\",\"hint_plural\":\"Podpowiedzi\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Następna podpowiedź\",\"hintprev\":\"Poprzednia podpowiedź\",\"solution\":\"Rozwiązanie\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Kopiuj do schowka\",\"startover\":\"Zacznij od początku\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Kontynuuj\",\"submitanswer\":\"Wyślij\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Poprzednia sekcja\",\"nexttopic\":\"Następna sekcja\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Spróbuj ponownie\"},\"text\":{\"startover\":\"Zacznij od początku\",\"areyousure\":\"Czy na pewno chcesz zacząć od początku? (cały postęp w zadaniu zostanie utracony)\",\"youmustcomplete\":\"Musisz ukończyć\",\"exercise\":\"ćwiczenie\",\"exercise_plural\":\"ćwiczenia\",\"inthissection\":\"w tej sekcji przed kontynuowaniem\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"luka\",\"blank_plural\":\"luk(i)\",\"exercisecontainsblank\":\"To ćwiczenie zawiera {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Proszę uzupełnić {{blank}} prawidłowym kodem.\",\"unparsable\":\"Wygląda na to, że może to nie być prawidłowy kod R. R nie jest w stanie przetworzyć Twojego tekstu na polecenie. Mogłeś(-aś) zapomnieć wypełnić luki, usunąć podkreślnik, umieścić przecinka między argumentami, lub zamknąć znak <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> lub <code>{<\\/code> odpowiadającym <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> lub <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>Wygląda na to, że Twój kod zawiera szczególnie sformatowane cudzysłowy lub cudzysłowy typograficzne (<code>{{character}}<\\/code>) przy ciągach znaków, co sprawia, że kod jest niepoprawny. R wymaga cudzysłowów prostych (<code>&quot;<\\/code> albo <code>'<\\/code>).<\\/p> {{code}} <p>Nie martw się, to powszechne źródło błędów, gdy kopiuje się kod z innego programu, który sam formatuje teskt. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Spróbuj usunąć znak specjalny i wpisać do ponownie ręcznie.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"and\":\"i\",\"or\":\"lub\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}}}}<\/script>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-format"]},{"type":"character","attributes":{},"value":["0.11.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmarkdown/templates/tutorial/resources"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-format.js"]},{"type":"character","attributes":{},"value":["tutorial-format.css","rstudio-theme.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["font-awesome"]},{"type":"character","attributes":{},"value":["6.5.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["fontawesome"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/all.min.css","css/v4-shims.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["fontawesome"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.5.3"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootbox"]},{"type":"character","attributes":{},"value":["5.5.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/bootbox"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["bootbox.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["idb-keyvalue"]},{"type":"character","attributes":{},"value":["3.2.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/idb-keyval"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["idb-keyval-iife-compat.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.11.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120]}},"value":[{"type":"character","attributes":{},"value":["abind","ade4","ape","backports","base","Biobase","BiocGenerics","BiocParallel","biomformat","Biostrings","bslib","cachem","checkmate","cli","cluster","codetools","colorspace","commonmark","compiler","crayon","data.table","datasets","DelayedArray","DESeq2","digest","dplyr","ellipsis","evaluate","fastmap","fontawesome","forcats","foreach","generics","GenomeInfoDb","GenomeInfoDbData","GenomicRanges","ggplot2","glue","graphics","grDevices","grid","gtable","hms","htmltools","htmlwidgets","httpuv","httr","igraph","IRanges","iterators","jquerylib","jsonlite","knitr","later","lattice","learnr","lifecycle","locfit","lubridate","magrittr","markdown","MASS","Matrix","MatrixGenerics","matrixStats","methods","mgcv","mime","multtest","munsell","nlme","parallel","permute","phyloseq","pillar","pkgconfig","plyr","promises","purrr","R6","Rcpp","readr","reshape2","rhdf5","rhdf5filters","Rhdf5lib","rlang","rmarkdown","rprojroot","rstudioapi","S4Arrays","S4Vectors","sass","scales","shiny","SparseArray","splines","stats","stats4","stringi","stringr","SummarizedExperiment","survival","tibble","tidyr","tidyselect","tidyverse","timechange","tools","tzdb","UCSC.utils","utils","vctrs","vegan","withr","xfun","xtable","XVector","yaml","zlibbioc"]},{"type":"character","attributes":{},"value":["1.4-8","1.7-23","5.8-1","1.5.0","4.4.2","2.66.0","0.52.0","1.40.0","1.34.0","2.74.1","0.9.0","1.1.0","2.3.2","3.6.4","2.1.6","0.2-20","2.1-1","1.9.2","4.4.2","1.5.3","1.17.0","4.4.2","0.32.0","1.46.0","0.6.37","1.1.4","0.3.2","1.0.3","1.2.0","0.5.3","1.0.0","1.5.2","0.1.3","1.42.3","1.2.13","1.58.0","3.5.1","1.8.0","4.4.2","4.4.2","4.4.2","0.3.6","1.1.3","0.5.8.1","1.6.4","1.6.15","1.4.7","2.1.4","2.40.1","1.0.14","0.1.4","1.9.0","1.49","1.4.1","0.22-6","0.11.5","1.0.4","1.5-9.11","1.9.4","2.0.3","1.13","7.3-61","1.7-1","1.18.1","1.5.0","4.4.2","1.9-1","0.12","2.62.0","0.5.1","3.1-166","4.4.2","0.9-7","1.50.0","1.10.1","2.0.3","1.8.9","1.3.2","1.0.4","2.6.1","1.0.14","2.1.5","1.4.4","2.50.2","1.18.0","1.28.0","1.1.5","2.29","2.0.4","0.17.1","1.6.0","0.44.0","0.4.9","1.3.0","1.10.0","1.6.2","4.4.2","4.4.2","4.4.2","1.8.4","1.5.1","1.36.0","3.7-0","3.2.1","1.3.1","1.2.1","2.0.0","0.3.0","4.4.2","0.4.0","1.2.0","4.4.2","0.6.5","2.6-10","3.0.2","0.51","1.8-4","0.46.0","2.3.10","1.52.0"]}]}]}
</script>
<!--/html_preserve-->
</div>

</article> <!-- topics -->

<div class="topicsContainer">
<div class="topicsPositioner">
<div class="band">
<div class="bandContent topicsListContainer">

<!-- begin doc-metadata -->
<div id="doc-metadata">
<h1 class="title toc-ignore" style="display:none;">Analyzing 16S data in
Phyloseq</h1>
<h2 class="subtitle"><em><p><img src="images/cmoor_logo_notext.png"
style="width:60.0%" /> <a href="https://help.c-moor.org/"
target="_blank">Get Help</a></p></em></h2>
<h4 class="author"><em>Valeriya Gaysinskaya, John Hopkins
University</em></h4>
<h4 class="author"><em>Sayumi York, Notre Dame of Maryland
University</em></h4>
<h4 class="author"><em>Frederick Tan, John Hopkins University</em></h4>
</div>
<!-- end doc-metadata -->

</div> <!-- bandContent.topicsListContainer -->
</div> <!-- band -->
</div> <!-- topicsPositioner -->
</div> <!-- topicsContainer -->


</main> <!-- bandContent page -->
</div> <!-- pageContent band -->



<!-- Build Tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<script>
// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>

</html>
