---
title: "Explore 16S rRNA Data with phyloseq"
output:
  learnr::tutorial:
    css: css/cmoor.css
    progressive: false #set to true to force stop until exercises are completed with 100% score
    version: 2.0
subtitle: |
  ![](images/cmoor_logo_notext.png){width=60%}
  
  [Get Help](https://help.c-moor.org/){target="_blank"}
runtime: shiny_prerendered

# The fields below will be used to create the first page of the tutorial.  You can use markdown.
# For summary, all new lines must be indented to the same level as the first line.
# The image must be surrounded with quotes
summary: |
  This tutorial will introduce you to...
image: #"![Human Protein Atlas logo](images/HPA_logo.png)"
image_caption: ""
author:
  - "Valeriya Gaysinskaya, Johns Hopkins University"
  - "Sayumi York, Notre Dame of Maryland University"
  - "Frederick Tan, Johns Hopkins University"
learning_goals:
  - "Use phyloseq to explore a 16S dataset."
  - "Use tidyverse to visualize your results."
time: "X min"
prerequisites: "None"
---

```{r setup, include=FALSE}
#Load learnr
library(learnr)
knitr::opts_chunk$set(echo = FALSE)
tutorial_options(exercise.completion=TRUE) #Uncomment to enable 


library("tidyverse")
library("phyloseq")

#Uncomment exercise timelimit when deployed
tutorial_options(exercise.timelimit = 180)

#Load MISO_16S.prop.RDS
#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds
#Normalization by turning counts into proportions

#Local path
#miso <- readRDS("MISO_16S.prop.clean.RDS")

#SciServer paths
# miso <- readRDS( "/home/idies/workspace/Storage/vgaysin1/persistent/MISO_16S.prop.rds" )
miso <- readRDS( "/home/idies/workspace/c_moor_data/miso-16s/MISO_16S.prop.clean.rds" )


# <!-- EXERCISE = should we see an interactive code chunk with this code? -->
# <!-- EVAL = should you run this code? -->
# <!-- ECHO = should you print the code? -->

```


<!---
Don't edit the Welcome page, it will be filled in automatically using the information from the YAML header
Edit the rest of the document as you like
There are some suggested sections to provide a standard order across our tutorials, but they may not all be needed/appropriate for all tutorials.
Section 1. Content 1 has example quizes and exercises
-->

## Welcome {.splashpage}

### `r rmarkdown::metadata$title`

<div class="splashpage-container">
  <figure class="splashpage-image">
  `r rmarkdown::metadata$image`{width=100%}
  <figcaption class="caption">`r rmarkdown::metadata$image_caption`</figcaption>
  </figure>

  `r rmarkdown::metadata$summary`
  

</div>


#### Learning Goals

```{r}
# Extract learning goals from YAML and add HTML tags to make an ordered list
learningGoals <- rmarkdown::metadata$learning_goals
learningGoals <- paste("<li>", learningGoals, "</li>", sep="", collapse="")

```

<ol>
`r learningGoals`
</ol>

#### Authors:

```{r}
# Extract authors from YAML and add HTML tags to make a list
authorList <- rmarkdown::metadata$author
authorList <- paste("<li>", authorList, "</li>", sep="", collapse="")

```

<ul>
`r authorList`
</ul>


```{r}
# Extract the tutorial version from the YAML data and store it so we can print it using inline r code below.  This can't be done directly inline because the code for extracting the YAML data uses backticks
tv <- rmarkdown::metadata$output$`learnr::tutorial`$version
```

#### Version: `r tv`

## Phyloseq

In this module, we'll use two R packages to explore our 16S microbiome data:

  - *phyloseq*, a package for microbiome analysis
  - *ggplot2*, a popular graph making tool that is a part of the tidyverse group of packages

While no prior experience in R is needed, we will refer to objects throughout this module. We have a separate module for learning about R objects (Meet R!). For more on ggplot check out the R for Data Science module (r4ds2) or the R Graph Galley (https://r-graph-gallery.com/)

<!-- REMINDER: Link the R graph gallery -->

## Intro concepts

Phyloseq object:
- A phyloseq object is composed of 4 core data classes. They are (1) the OTU abundance table (otu_table), (2) a table of sample data (sample_data), (3) a table of taxonomic descriptors (taxonomyTable); and (4 - Optional) a phylogenetic tree. Together, these datasets create a phyloseq-class object.

OTU or ASV:
- Amplicon sequencing results in Amplicon Sequence Variants (ASVs). ASVs varying from as little as one single nucleotide can be defined as separate ASVs. ASVs can further be grouped ASVs into OTUs (Operational Taxonomic Units) by clustering similar ASVs based on a threshold of similarity (e.g. ASVs with 1% difference in sequence can be grouped into same species.) Phyloseq , can use ASV or OTU counts table, but will refer to both as OTUs, such that you only will have an otu_table, and not an asv_table.

ASV counts normalization:
- An important concept in microbiome data analysis is the concept of data normalization, to enable statistical analyses and comparisons between samples. Therefore, for exploring 16S with phyloseq module we will use normalized ASV counts. The ASV counts were normalized by the total number of reads in each sample to convert the counts to proportion.

Subsetting data in R:
- R has powerful features for accessing data parts via subsetting, which involves selecting portions of information from a bigger set of data. Different data structures (e.g. dataframes, matrices, lists) can have different rules for subsetting. For example, an otu_table is a matrix, and can be subsetted using rules for subsetting a matrix in R, which utilizes brackets [ ]. To subset matrices, you can specify the [row 1, column 2] to access information stored in row 1 and column 2. 


## Exploring a phyloseq object

Let's explore our phyloseq object (miso). We can check by calling the object through its name.

### Demo Code

```{r, echo=TRUE}
miso
```

Here we see that the phyloseq object consists of an otu_table, sample_data and tax_table. Phyloseq knits all these tables together inside the object. 

#### Multiple choice

```{r miso-object_quiz}
quiz(caption = "What's in the phyloseq object?",
    
  question("How many ASVs are in the phyloseq object?",
    answer("1702", correct = TRUE),
    answer("105"),
    answer("32"),
    answer("9"),
    allow_retry = TRUE,
    random_answer_order = TRUE
  ),
  
  question("How many samples are in the phyloseq object?",
    answer("1702"),
    answer("105", correct=TRUE),
    answer("32"),
    answer("9"),
    allow_retry = TRUE,
    random_answer_order = TRUE
  ),
  
  question("How many variables are there in the experiment? (ex. age, phase, timepoint)",
    answer("1702"),
    answer("105"),
    answer("32", correct = TRUE),
    answer("9"),
    allow_retry = TRUE,
    random_answer_order = TRUE
  )
)
```

---

## OTU_table

An otu_table contains ASV counts for all the samples. In an otu_table, columns are different ASVs, and rows are different samples. Remember, that even though we are using ASVs for input, and not OTUs, phyloseq treats them the same, and uses only 'otu' nomenclature.

### Interactive Code

Examine the first few columns and rows with the following code.

```{r otu_table-1, exercise=TRUE, echo=FALSE}
head(colnames(otu_table(miso)))
```

```{r otu_table-2, exercise=TRUE, echo=FALSE}
head(rownames(otu_table(miso)))
```

#### Multiple choice

```{r otu_table_quiz1}
quiz(caption = "What is the otu_table?",
  
question("What are columns in the otu_table?",
    answer("sample names"),
    answer("ASV IDs", correct=TRUE),
    answer("taxonomy ranks"),
    allow_retry = TRUE
  ),

question("What are rows in the otu_table?",
    answer("sample names", correct=TRUE),
    answer("ASV IDs"),
    answer("taxonomy ranks"),
    allow_retry = TRUE
  )
)
```


We can examine a specific sample or ASV in the otu_table by selecting the samples (rows) and ASVs (columns). 

### Interactive Code

Use code below to examine ASV1 counts for sample 1. This is done by specifying sample information (rows) and ASV information (columns) in the brackets. 

```{r otu_table-3, exercise=TRUE, echo=FALSE}
otu_table(miso)[1,"ASV1"]
```

We can see that the sample MISO1-S02-1 has a value of 0.059 for ASV 1. Recall that we are working with normalized data, so our data is in proportions. So really our data are telling us that 5.9% of this sample is ASV 1! Explore different samples and ASVs by changing the numbers in the brackets.

For samples 1 through 5, explore counts for ASV50.

```{r otu_table-4, exercise=TRUE, echo=FALSE}

otu_table(miso)[1:5, "ASV50"]
```

#### Multiple choice

```{r otu_table_quiz2}
quiz(caption = "What's in the otu_table?",
     
  question("What is the normalized count for ASV115 in sample 10",
    answer("there is no ASV0115",),
    answer("0.05988178", ),
    answer("0.005442962", ),
    answer("0.0009637625", correct=TRUE),
    allow_retry = TRUE
  ),
  
  question("For samples 1 through 5, how many samples have zero counts for ASV1000?",
    answer("1",),
    answer("3", ),
    answer("All of them", correct=TRUE),
    allow_retry = TRUE
  )
)
```

---

## Tax_table

The tax_table contains all the information relating to what ASV has what taxonomy assigned to it.

### Demo Code

Examine the column and rows in the tax_table with the following code.

```{r, echo=TRUE}
colnames(tax_table(miso))
head(rownames(tax_table(miso)))
```

#### Multiple choice

```{r tax_table_cols_rows_quiz}
quiz(caption = "What's in the tax_table?",
  question("What are columns in the tax_table?",
    answer("Sample names"),
    answer("ASV IDs"),
    answer("Taxonomy ranks", correct=TRUE),
    allow_retry = TRUE
  ),
  
  question("What are rows in the tax_table?",
    answer("Sample names",),
    answer("ASV IDs", correct=TRUE),
    answer("Taxonomy ranks"),
    allow_retry = TRUE
  )
)
```

### Interactive Code

Use the interactive code block below to try out the commands you learned and retreive specific ASV information.

Check the taxonomy associated with ASV1. Within the brackets, rows and columns are separated by a comma (","). By leaving the column blank, R will return all columns. In this case it will return the taxonomic assignment at all levels for ASV1.

```{r tax_table_col_rows_ASV1, exercise=TRUE, echo=FALSE}
tax_table(miso)["ASV1",]
```
#### Multiple choice

```{r tax_table_ASV2005}
quiz(caption = "Let's find a taxnomic assignment by ASV",
  question("What is the taxonomic Family associated with ASV2005?",
    answer("Bifidobacteriaceae"),
    answer("Lachnospiraceae", correct=TRUE),
    answer("Clostridiales"),
    answer("NA"),
    allow_retry = TRUE
  )
)
```

---

## Sample_data

The sample_data contains all the metadata for each sample. 

### Demo Code

Using the head() command lets look only at the first handful of columns and rows, but if we remove head we can also see the entirety of each. 

Note, that because sample data is a data frame class (or table) - as opposed to matrix class for example - sample data can be called using a data.frame() command and display the sample data as a nice table.

```{r, echo=TRUE}
#Examine sample_data (metadata) table
head(data.frame(sample_data(miso)))
```

### Interactive Code

Now, use the code block below to view a handful of column names and rownames

```{r sample-data1, exercise=TRUE, echo=FALSE}
head(colnames(sample_data(miso)))
head(rownames(sample_data(miso)))
```

### Multiple Choice

```{r overview_sample_data_quiz}
quiz(caption = "Exploring the sample_table",
  question("What are columns in the sample_table?",
    answer("Samples"),
    answer("Sample metadata", correct=TRUE),
    answer("Taxonomy ranks"),
    allow_retry = TRUE
  ),
  
  question("What are rows in the sample_table?",
    answer("Samples", correct=TRUE),
    answer("Sample metadata"),
    answer("Taxonomy ranks"),
    allow_retry = TRUE
  ),
  
  question("Check all that are variables in the sample_data",
    answer("age", correct=TRUE),
    answer("subject", correct=TRUE),
    answer("gender", correct=TRUE),
    answer("timepointgroup", correct=TRUE),
    answer("id", correct=TRUE),
    allow_retry = TRUE
  )
)
```

### Interactive Code

Let's try looking at different variables for a given column of metadata.

```{r unique_metadata, exercise=TRUE, echo=FALSE}
unique(data.frame(sample_data(miso)[, "subject"]))
unique(data.frame(sample_data(miso)[, "timepointgroup"]))

```

#### Multiple choice

```{r metadata_sample_table_quiz}
quiz(caption="Exploring metadata in sample_table",

  question("How many subjects are in the dataset? HINT: Use the arrow on the top right side of the data frame table to access more columns",
    answer("5"),
    answer("21", correct=TRUE),
    answer("35"),
    answer("105"),
    answer("32"),
    allow_retry = TRUE
  ),
  
  question("What are the timepointgroups in the dataset?",
    answer("1-5"),
    answer("BD, HD, WO", correct=TRUE),
    answer("miso and non-miso"),
    allow_retry = TRUE
  )
)
```


## Visualizing taxonomy

Let's try plotting our phyloseq object to get an idea of everything that's inside it.
The code below plots the taxonomy profile of individuals (subjects) based on Phylum.

### Demo Code

```{r, echo=TRUE}
  plot_bar(miso, "subject", fill = "Phylum") + 
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack") 
```

Notice that the y-axis currently has "Abundance" on a scale of 1-5. Why is this the case? Remember that we are working with proportions, so each sample should add up to 1, which represents 100% of a sample. For each subject we sampled them at 5 different time points. Thus, we have a total of "5" for all time points included.

#### Multiple choice

```{r plot_bar_phylum_quiz}
question("How many different Phyla are in the data?",
  answer("14"),
  answer("13", correct=TRUE),
  answer("5"),
  allow_retry = TRUE
)
```

---

### Interactive Code

The demo code above plotted the taxonomy profile of individuals based on Phylum. Use the code below to play around with the plot_bar() function. 

The demo plot did not have a plot title. Use code block below to give your plot a title by substituting "Choose a name for your graph" with a title name appropriate for the plot (e.g. Phylum by individual).

```{r plot_bar_phylum_title, exercise=TRUE, echo=FALSE}
plot_bar(miso, x="subject", fill = "Phylum", title = "Choose a name for your graph") +
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")
```

Try substituting phylum with another taxonomic rank of interest, like class by substituting "Phylum" with "Class" in the code block below. Make sure you replace all instances of phylum, and pay attention to the case (example: notice that subject is in all lowercase, but Phylum uses a capital P).

```{r plot_bar_subject_Class, exercise=TRUE, exercise.eval=FALSE}
plot_bar(miso, x= "subject", fill = "Phylum", title = "choose a name for your graph") +
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")
```

Try plotting a different taxonomy based on another variable such as timepointgroup, phase, age and gender, by substituting "subject".

```{r plot_bar_subject_Phylum, exercise=TRUE, exercise.eval=FALSE}
plot_bar(miso, x= "subject", fill = "Phylum", title = "choose a name for your graph") +
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")
```

#### Multiple choice

```{r plot_bar_tax_quiz}
quiz(caption = "Changing the phyloseq graph by taxonomic level",
  question("What are the 2 most common bacterial classes in the dataset?",
    answer("Clostridia", correct = TRUE),
    answer("Bacteroidia", correct = TRUE),
    answer("Mollicutes"),
    answer("Synergistia"),
    allow_retry = TRUE,
    random_answer_order = TRUE
  )
)
```

```{r plot_bar_metadata_quiz}
quiz(caption = "Changing the phyloseq graph by metadata",
  question("What timepoint has the lowest proportion of Verrucomicrobia?",
    answer("1"),
    answer("2"),
    answer("3"),
    answer("4"),
    answer("5", correct=TRUE),
    allow_retry = TRUE,
    random_answer_order = FALSE
  )
)
```

---

## Using subset_taxa()

### Interactive Code

What happens when we try to plot our data at the level of order? Try it out with the following code.

```{r plot_bar_metadata_change_order, exercise=TRUE, exercise.eval=FALSE}
plot_bar(miso, x= "subject", fill = "Order", title = "choose a name for your graph") +
  geom_bar(aes(color = Order, fill = Order), stat = "identity", position = "stack")
```


There are way more orders than phyla. As we get more and more groups in our graph, it becomes harder to read.

Subset_taxa() allows you to select a group of data based on a taxonomic rank. This allows us to discriminate between taxa on a finer scale.

The code below subsets only ASVs with the phylum Firmicutes, and plots the associated order for all these ASVs by subject. First, we create a new phyloseq object called my_subset, which then stores only our Firmicutes ASVs. Then we plot these ASVs by their taxonomic order.

In the code block below, subset taxa by Phylum first, then, plot all Order associated with Phylum Firmicutes.

```{r r_subset_example, exercise=TRUE, echo=FALSE}
my_subset <- subset_taxa(miso, Phylum == "Firmicutes")

plot_bar(my_subset, x="subject", fill = "Order", title = "Orders of Firmicutes ASVs by subject") + 
  geom_bar(aes(color = Order, fill = Order), stat = "identity", position = "stack")

```

That looks much better! There are only 7 orders represented within ASVs of the phyla Firmicutes so we can see them clearly in our graph.

The command below will retrieve a list of all the different Phyla you could subset. Of course, if we change "Phylum" to another taxonomic rank we will see all the different groups of that rank.

```{r examine_unique_taxonomic_group, exercise=TRUE, echo=FALSE}
unique(tax_table(miso)[ , "Phylum"])

```

Modify the code below to subset class Bacteroidia (instead of Firmicutes) and plot the different orders of the class Bacteroidia for the 21 individuals (subjects). Be careful - there a few things to modify in the code below to get the right plot! And don't forget the title!

```{r subsetting_bacteroidia, exercise=TRUE, exercise.eval=FALSE}
my_subset <- subset_taxa(miso, Phylum == "Firmicutes")

plot_bar(my_subset, x="subject", fill = "Class", title = "My title") +
  geom_bar(aes(color = Class, fill = Class), stat = "identity", position = "stack")

```

#### Multiple choice

```{r subsetting_quiz}
quiz(caption = "Changing the phyloseq graph by metadata",
  question("How many different classes are represented in the entire dataset?",
    answer("13"),
    answer("14"),
    answer("22", correct=TRUE),
    allow_retry = TRUE
  ),

  question("What are the 2 most abundant orders in the class Gammaproteobacteria?",
    answer("Pseudomonadales"),
    answer("Pasteurellales"),
    answer("Aeromonadales"),
    answer("Enterobacteriales", correct=TRUE),
    answer("Betaproteobacteriales", correct=TRUE),
    allow_retry = TRUE
  )
)
```


```{r question-problem-summary}
question("What was the goal of these exercises?",
  answer("To see how much you already know"),
  answer("To give you a taste of some useful commands", correct=TRUE),
  answer("To daze and confuse you"),
  allow_retry = TRUE
)
```

## Try it out!

Use the following code blocks and example code to explore the data.

```{r try_it_out_2, exercise=TRUE, echo=FALSE}

```

```{r try_it_out_3, exercise=TRUE, echo=FALSE}

```

```{r try_it_out_4, exercise=TRUE, echo=FALSE}

```

```{r try_it_out_5, exercise=TRUE, echo=FALSE}

```


## Phyloseq-related R commands

### Exploring data with base R commands
- **head**: A base R function that gives you the first part of a called object
  - head(my_object) 
  - head(colnames(otu_table(miso)))
  - "Look into the object called miso and find the otu_table, then look at the column names   and give me the head (first part) of the column names"
  
- **tail**: A base R function that gives you the last part of a called object
  - tail(my_object)
  
- **colnames()**: A base R function that gives you the column names of a given data frame
  - colnames(my_data_frame)
  
- **rownames()**: A base R function that gives you the row names of a given data frame
  - rownames(my_data_frame)
  
- **unique()**: A base R function that lets you filter only the unique values of a given data frame

- **ls.str()**: A base R function that will list out all the objects available to you in your environment. In this module, each code block is its own environment.

### Phyloseq commands

- **otu_table()**: Used to reference the otu_table of a phyloseq object

- **tax_table()**: Used to reference the tax_table of a phyloseq object

- **sample_data()**: Used to reference the sample_data of a phyloseq object 
  - Note: Requires data.frame() to work

- **subset_taxa()**: Used to subset a data frame and keep only the chosen taxa

- **plot_bar()**: Returns a ggplot2 object that can be modified further with ggplot commands using + to add more options. 

### ggplot commands

- **facet_wrap** (Advanced): Ask ggplot to facet by a given field. This will break the graph down into smaller graphs focused on each subject.

```{r facet_example, exercise=TRUE, echo=FALSE}
plot_bar(miso, x="timepoint", fill = "Phylum", title = "choose a name for your graph") +
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack") +
  facet_wrap(~subject)
```

## Summary

In this module we have explored a 16S dataset and used phyloseq to create informative barplots about our data. Using these barplots we can make some observations about the data and come up with new questions or draw conclusions! 

## FAQ

## Common errors

**When subsetting data**:
- **1) Error in dimnames(x) <- dn : length of 'dimnames' [1] not equal to array extent**
Your subset does not contain any data. Make sure your taxonomic level (example: phylum, class, order) matches the name of your taxonomic group (Alphaproteobacteria, Bacilli, Clostridiales). You may have a mismatch (example: asking phyloseq to subset by the phylum Clostridiales, which is an order)




