---
title: "RNA-seq Project Work"
output:
  learnr::tutorial:
    css: css/cmoor.css
    progressive: false #set to true to force stop until exercises are completed with 100% score
    version: 2.0
subtitle: |
  ![](images/cmoor_logo_notext.png){width=60%}
  [Get Help](https://help.c-moor.org/){target="_blank"}
runtime: shiny_prerendered

# The fields below will be used to create the first page of the tutorial.  You can use markdown.
# For summary, all new lines must be indented to the same level as the first line.
# The image must be surrounded with quotes
summary: |
  This module is meant to provide students with a working space for their RNA-seq projects. In addition to the midgut dataset from the main differential expression module, we have a number of other datasets you can explore here as well as code chunks and starting off points to help get you started!
image: ""
image_caption: ""
author:
  - "Sayumi York, Notre Dame of Maryland University"
  - "Gauri Paul, Clovis Community College"
learning_goals:
  - "Familiarize the datasets available to you"
  - "Select a dataset that you'd like to work on for your research project"
  - "Begin your exploratory analysis of your chosen dataset to refine your research questions"
time: ""
prerequisites: "None"
---

```{r setup, include=FALSE}
#Load learnr
library(learnr)
knitr::opts_chunk$set(echo = FALSE)
tutorial_options(exercise.completion=TRUE) #Uncomment to enable 

#Uncomment exercise timelimit when deployed
tutorial_options(exercise.timelimit = 360)

#Load libraries
library("DESeq2")
library("clusterProfiler")
library("org.Dm.eg.db")
library("tidyverse")
library("org.Mm.eg.db")
library("MarianesMidgutData")

#####################################
#Define functions
####################################

# Generic functions

formatDESeq2Results <- function( x ) {
  df <- as.data.frame(x)
  df <- data.frame(rownames(df), df)
  colnames(df) <- c("GeneID", colnames(df)[-1])
  rownames(df) <- c()
  return(df)
}

# Fly functions

runClusterProfiler <- function (x) {
  ids <- bitr( x$GeneID, "ENSEMBL", "ENTREZID", "org.Dm.eg.db" )
  kegg <- enrichKEGG(ids$ENTREZID, "dme", keyType="ncbi-geneid")
  kegg@result$Description <- sub( " - Drosophila melanogaster \\(fruit fly\\)", "", kegg@result$Description )
  return(kegg)
}

getClusterProfilerGenes <- function (x, i) {
  data.frame( x ) %>%
    filter( str_detect( Description, i ) ) %>%
    pull( geneID ) %>%
    strsplit( "/" ) %>%
    unlist() %>%
    bitr( fromType="ENTREZID", toType="SYMBOL", OrgDb="org.Dm.eg.db") %>%
    pull( SYMBOL )
}

plotAcrossRegions <- function( x ) {
  df <- data.frame(counts(midgut)[x,], midgut_tsv$condition)
  colnames(df) <- c("counts", "region")
  df <- df[10:30,]
  df$region <- fct_relevel( factor( df$region ), "a1", "a2_3", "Cu", "LFCFe", "Fe", "p1", "p2_4" )
  ggplot( df ) +
    geom_bar( aes( region, counts ), stat="identity" ) +    
    ggtitle(x)+
    theme_bw() + 
    theme( axis.text.x = element_text( angle=90, vjust=0.5 ) )
}

#Mouse functions
runClusterProfilerMouse <- function (x) {
  ids <- bitr( x$GeneID, "ENSEMBL", "ENTREZID", "org.Mm.eg.db" )
  kegg <- enrichKEGG(ids$ENTREZID, "mmu", keyType="ncbi-geneid")
  kegg@result$Description <- sub( " - Mus musculus \\(house mouse\\)", "", kegg@result$Description )
  return(kegg)
}

getClusterProfilerGenesMouse <- function (x, i) {
  data.frame( x ) %>%
    filter( str_detect( Description, i ) ) %>%
    pull( geneID ) %>%
    strsplit( "/" ) %>%
    unlist() %>%
    bitr( fromType="ENTREZID", toType="SYMBOL", OrgDb="org.Mm.eg.db") %>%
    pull( SYMBOL )
}

plotAcrossRegionsMouse <- function( x ) {
  df <- data.frame(counts(midgut)[x,], midgut_tsv$condition)
  colnames(df) <- c("counts", "region")
  df <- df[10:30,]
  df$region <- fct_relevel( factor( df$region ), "a1", "a2_3", "Cu", "LFCFe", "Fe", "p1", "p2_4" )
  ggplot( df ) +
    geom_bar( aes( region, counts ), stat="identity" ) +
    theme( axis.text.x = element_text( angle=90, vjust=0.5 ) )
}



#####################################
#Local paths
####################################

#####################################
#Sciserver paths
####################################

#####################################
#Packages
####################################


```


<!---
Don't edit the Welcome page, it will be filled in automatically using the information from the YAML header
Edit the rest of the document as you like
There are some suggested sections to provide a standard order across our tutorials, but they may not all be needed/appropriate for all tutorials.
Section 1. Content 1 has example quizes and exercises
-->

## Welcome {.splashpage}

### `r rmarkdown::metadata$title`

<div class="splashpage-container">
  <figure class="splashpage-image">
  `r rmarkdown::metadata$image`{width=100%}
  <figcaption class="caption">`r rmarkdown::metadata$image_caption`</figcaption>
  </figure>

  `r rmarkdown::metadata$summary`
  

</div>


#### Learning Goals

```{r}
# Extract learning goals from YAML and add HTML tags to make an ordered list
learningGoals <- rmarkdown::metadata$learning_goals
learningGoals <- paste("<li>", learningGoals, "</li>", sep="", collapse="")

```

<ol>
`r learningGoals`
</ol>

#### Authors:

```{r}
# Extract authors from YAML and add HTML tags to make a list
authorList <- rmarkdown::metadata$author
authorList <- paste("<li>", authorList, "</li>", sep="", collapse="")

```

<ul>
`r authorList`
</ul>


```{r}
# Extract the tutorial version from the YAML data and store it so we can print it using inline r code below.  This can't be done directly inline because the code for extracting the YAML data uses backticks
tv <- rmarkdown::metadata$output$`learnr::tutorial`$version
```

#### Version: `r tv`

## Research project brainstorming

Coming up with a good research project idea can be an exciting but daunting challenge. <mark>**We recommend reviewing the available datasets in this document and choose one to focus on, then trying one of the following approaches.**</mark> If you're working with a group, considering assigning every member be responsible for a particular approach and have everyone bring 1 or 2 ideas to your next meeting to vote on.

### Approach 1 - Start with a gene of interest

- Suggested for students that come with a biological phenomenon of interest
- Involves looking for differentially expressed genes involved in that interest

**1. Identify a biological interest of yours.**: Do you have a disease state or condition you are interested in? How about a particular organ, organ system, cell type, or organelle? You might think about things you've heard on the news or in social media relating to human health. 

**2. Find genes that are involved in your chosen interest**: Look up research papers through sites like PubMed and Google Scholar to look up studies that identify genes of interest for your particular topic. Skim through at least 4-5 different papers that contain some type of gene name or a process (like a metabolic process). Don't delve too deep or get stuck in the details. Just start creating a list of genes.

**3. Confirm that the gene exists in both humans and Drosophila**: If you started with a gene found in humans, look it up on FlyBase to get an "FBgn" ID. If you started with a fly gene, you will have to get the "FBgn" ID on FlyBase and search for an ortholog in Human Protein Atlas. Note that some genes may have slightly different names in each species. 


![](images/good-candidate.png){width=100%}

![](images/bad-candidates.png){width=100%}

**4. Use plotAcrossRegions to see if there is an interesting story within your list of genes**: Jump over to the plotAcrossRegions section and run the code on your chosen genes using their FBgn ID. Genes that are likely to be differentially expressed and serve a specialized function in the midgut show an uneven expression across regions. Pay attention to the y-axis. Genes that are unevenly expressed but are found in low counts across all regions are not likely to be key genes.

**5. Confirm that the gene is significantly differentially expressed between regions using the table generated by the DESeq2 analysis**: Compare the regions with the highest and lowest expression (counts). In the generated table, check that the *padj* value is less than 0.05. 

### Approach 2 - Select a category of differentially expressed genes from the ClusterProfiler dot plot

- Suggested for students that aren't sure what they're interested in
- Start with a group of genes identified as differentially expressed
- Find an interesting biological story to tell within those genes

**1. Run a ClusterProfiler analysis on two regions.**: Jump to the ClusterProfiler analysis section and choose two sections of the midgut to run the analysis on. From within your dotplot results, choose a catagory with a with low
*padj* value and high count value. 

**2. Run getClusterProfilerGenes on that category to get a list of differentially expressed genes within the category**: Remember to run getClusterProfilerGenes in the same code chunk as the rest of the ClusterProfiler analysis. 

**3. Pick a gene and follow steps 3-5 from Approach 1**: Repeat for all genes to make an interesting poster based on your chosen metabolic process.


### Approach 3 - Select highly differentially expressed genes from the DESeq2 results table

This approach is similar to approach 2.

- Suggested for students that aren't sure what they're interested in
- Start with a table of significantly differentially expressed genes
- Find an interesting biological story to tell within those genes

**1. Run the DESeq2 analysis and get a results table**: Choose any two regions of your choice to run through the DESeq2 analysis.

**2. Filter genes with significant *padj* value**: Use the filter function to get a list of genes that are significantly (<0.05 or <0.01 or even <0.001) expressed in the two regions of your choice. Remember to run the filter function within the same code block as the DESeq2 code.

**3. Pick a gene and follow steps 3-5 from Approach 1**: Choose 2-3 from this process that you think have a specialized role in the midgut.

**4. Use databases and read preivously published research to figure out what function your genes are involved in**: Look for a story that is presentable. What is special about this gene? What function does it have? What interactions does it have with other genes? What does being differential expressed signify for this gene? What metabolic process is this gene a part of? Are there any studies done on your gene? What diseases or conditions is the gene linked to?

**5. Use databases to find related genes**: These genes could be involved the in the same biological process; they do not have to be differentially expressed to be able to tell a good story.

## Meet the midgut dataset

The data presented here are from the 2013 study by Marianes and Spradling, **Physiological and stem cell compartmentalization within the Drosophila midgut**. We refer to this study as the midgut study. To best make use of this data we recommend you complete the following modules, some of which use these data as an example for related analyses:

- Model Organisms: *Drosophila melanogaster*
- Biological Databases: FlyBase
- Introduction to RNA-seq data
- Differential expression with DESeq2

### Why should we study the Drosophila midgut?

![](images/sadaqat.png){width=100%}

<p class=caption> **A figure comparing the human intestine and fly intestine by Sadaqat et al. 2021**: Different colored sections represent sections that serve a similar function between the two species.</p>

*Drosophila melanogaster*, the fruit fly, is one of the most popular model organisms in research. *Drosophila* and humans have many **orthologs** - genes that come from the same ancestor - that make them useful for foundational research. Scientists start with cheap models like the fly and build a body of evidence before moving on to more complex organisms including vertebrates like mice and humans themselves. The image above shows how the human intestine is similar to the fly intestine (note that not all structures/functions are shared in both).

### Different parts of the midgut we can compare

Just like the human intestine, the *Drosophila* midgut can be seen as different specialized sections, each of which serves a particular role in digestion. Understanding what genes are expressed in each section of the *Drosophila* midgut can give us a clue to what the human ortholog might be doing.

> "…certain digestive processes—such as the breakdown or uptake of particular nutrients—are known to occur only in a specific portion of the intestine. For example, in fruit flies, a region in the middle of the intestine is acidified, and may act like an extra stomach. And in both fruit flies and mammals, iron is taken up mostly in the area of the gut just after the stomach."

> Marianes & Spradling, (2013)

![](images/midgut_objects.png){width=100%}

We have subsetted the data to make research easier for you. The above image shows the general functions of each section of the midgut. The bolded text (example: a1) is what you will use in your code when referring to that specific section.

### Some research ideas to get you started

You will need to identify genes of interest, formulate a research question around them, and check that they are in the midgut dataset.

- Look up gastrointestinal conditions such as diabetes, irritable bowel syndrome, and Celiac's disease. What genes and/or pathways are involved in these disorders?
- We now know the gut is inextricably linked to our brains, immune system, and circadian rhythms. What genes are involved in these connecting systems? What genes are used to communicate between these systems?
- The leaky gut hypothesis is centered on the idea that conditions like Celiac's disease may be caused by irregular gut intestine permeability. What genes are involved in maintaining the gut's defenses and are they the same across all gut regions? 
- What genes are involved in gut inflammation and how might they impact other genes in shared pathways?

### Differential expression template (Drosophila midgut dataset)

```{r deseq2-table-one-pager, exercise=TRUE, echo=FALSE}
temp <- results(midgut, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
R1_vs_R2
```

```{r try_it_out_deseq2-table-a, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-table-b, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-table-c, exercise=TRUE, echo=FALSE}

```

### ClusterProfiler the for Drosophila midgut dataset

```{r clusterprofiler_drosophila-onepager, exercise=TRUE, echo=FALSE}
# Run differential expression analysis
temp <- results(midgut, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
sig_genes <- filter(R1_vs_R2, padj <= 0.01)

# Run ClusterProfiler analysis
R1_vs_R2_clusters <- runClusterProfiler(sig_genes)

# Look at the dotplot to see the groups of genes
dim(R1_vs_R2_clusters)
dotplot(R1_vs_R2_clusters, showCategory=34, title="YOUR TITLE HERE", font.size=10, label_format = 50)

# Get the gene symbols for a group of genes
getClusterProfilerGenes(R1_vs_R2_clusters, "YOUR CATEGORY HERE")
```

```{r try_it_out_clusterprofiler_drosophila-a, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_drosophila-b, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_drosophila-c, exercise=TRUE, echo=FALSE}

```

### plotAcrossRegions the for Drosophila midgut dataset

- FlyBase IDs take the format FBgn#######

```{r plotAcrossRegions_drosophila-one-pager, exercise=TRUE, echo=FALSE}
plotAcrossRegions("YOUR FLYBASE ID HERE")
```


```{r plotAcrossRegions_drosophila-try-it-a, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_drosophila-try-it-b, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_drosophila-try-it-c, exercise=TRUE, echo=FALSE}

```


## Meet the gutbrain dataset

:::{.attention_box_pause}
:::{.attention_header_box_pause}
:::{.attention_title}
#### **The gutbrain dataset is still in development**
:::
:::
::: {.attention_content}
This dataset is currently undergoing pilot testing at Clovis Community College (Fall 2025). We do not have as extensive materials to support some resources such as the Mouse Genome Informatics (MGI) database yet. Regardless, interested students are welcome to test these data. We look forward to your research posters!
:::
:::

<br>

These data are from the 2019 study by Sharon et al. titled **Human Gut Microbiota from Autism Spectrum Disorder Promote Behavioral Symptoms in Mice**. We will refer to these data as the gutbrain dataset, as the original study explores the gut-brain axis - the link between the gut microbiome and the central nervous system. To best make use of this data we recommend you complete the following modules: 

- Model Organisms: *Mus musculus* (In development)
- Biological Databases: Mouse Genome Informatics (In development)
- Introduction to RNA-seq data
- Differential expression with DESeq2

### The gut-brain axis and Autism spectrum disorder

Over the last couple of decades, considerable research effort has been invested into exploring the **gut-brain axis**. It is now widely accepted that the gastrointestinal tract has a bidirectional relationship with the central nervous system, wherein each provides molecular feedback to each other. Consequently, many scientists are interested how the gut microbiome (the microbes that live in your GI tract) could impact human health, including chronic health conditions and neurodiverse development such as autism spectrum disorder (ASD).

Autism is a developmental disorder characterized by atypical social behaviors and experiences as well as repetitive behaviors. The exact number of people with autism in the US is unknown, some estimate that 1 in approximately 60 individuals are affected. 

### The gutbrain study design


![](images/sharon-study-design.png){width=100%}

Sharon et al. took gut microbiome (fecal) samples from children diagnosed with ASD and neurotypical children (controls) and implanted them into germ-free mice (mice without a microbiome). This parental generation of mice bred amongst themselves and produced offspring that were randomly selected for either behavior tests or brain tissue sampling.

The behavioral tests showed that offspring generation mice whose parents had a microbiome transplant from a child with ASD had more behavior characteristic of ASD, such as decreased movement and communication compared to the offspring of control-implanted mice.  

Brain tissue samples included two areas of the brain responsible for social behavior, the **Prefrontal cortex (PFC)** and the **Striatum (STR)**. In addition to gene expression from RNA-seq data, the researchers also explored alternative splicing and metabolomics analyses. 


![](images/sharon-comparisons.png){width=100%}
Although there are more comparisons than the above 4 listed, we recommend your analyses focus on one of these 4, as they are the most straightforward comparisons.

### Working with mouse functions and resources

Our standard RNA-seq miniCURE focuses on the midgut dataset and uses *Drosophila* based resources, such as FlyBase. Since the gutbrain dataset is based on mouse samples we have databases and functions specifically for mice that you should use instead:


![](images/mgi_format.png){width=100%}
- [**Mouse genome informatics (MGI)**](https://www.informatics.jax.org/) is the database you should use instead of FlyBase. MGI uses the Ensembl Gene Model ID format, where mouse genes take the format ENSMUG###########. Of course, since they're in different formats, you will get no results if you search for a fly gene in MGI or a mouse gene in FlyBase. 

- **Use the specialized functions getClusterProfilerGenesMouse, plotAcrossRegionsMouse, and runClusterProfilerMouse**: These functions should work seamlessly with mouse data with no changes in how the code looks to you. 

### Some research ideas to get you started

You will need to identify genes of interest, formulate a research question around them, and check that they are in the gutbrain dataset.

- People with ASD often have neurological comorbidities (additional medical conditions), such as OCD, ADHD, anxiety, migranes, and epilepsy. What genes are linked to these conditions
- Some people with ASD have a greatly expanded memory, especially for a topic of interest. Other individuals have a reduced working memory compared to neurotypical people. What genes are involved in memory? Are there different pathways involved in long-term and short-term memory? How about genes relating to recognizing specific kinds of information, such as language, numbers, or faces? 
- Phineas Gage was involved in an accident that damaged his frontal lobe, which contains the prefrontal cortex. Following this incident, his personality underwent a dramatic change, suggesting that the frontal lobe regulates one's personality, executive decision making, and interactions with others. What genes are involved in executive function and focus, and how might they impact individuals with ASD and neurotypical individuals?
- Phineas Gage's case also demonstrated for us how brain regions are specialized. What genes are only expressed in the striatum or prefrontal cortex? What genes and/or pathways are involved in the two regions communicating with each other?

### Differential expression template (Mouse gutbrain dataset)

```{r deseq2-tablemouse-one-pager, exercise=TRUE, echo=FALSE}
temp <- results(gutbrain, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
R1_vs_R2
```

```{r try_it_out_deseq2-tablemouse-a, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-tablemouse-b, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-tablemouse-c, exercise=TRUE, echo=FALSE}

```


### ClusterProfiler for the mouse gutbrain dataset

```{r clusterprofiler_mouse-one-pager, exercise=TRUE, echo=FALSE}
# Run differential expression analysis
temp <- results(gutbrain, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
sig_genes <- filter(R1_vs_R2, padj <= 0.01)

# Run ClusterProfiler analysis
R1_vs_R2_clusters <- runClusterProfilerMouse(sig_genes)

# Look at the dotplot to see the groups of genes
dim(R1_vs_R2_clusters)
dotplot(R1_vs_R2_clusters, showCategory=34, title="YOUR TITLE HERE", font.size=10, label_format = 50)

# Get the gene symbols for a group of genes
getClusterProfilerGenesMouse(R1_vs_R2_clusters, "YOUR CATEGORY HERE")
```

```{r try_it_out_clusterprofiler_mouse-a, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_mouse-b, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_mouse-c, exercise=TRUE, echo=FALSE}

```


### plotAcrossRegionsMouse the for mouse gutbrain dataset

- MGI IDs take the format ENSMUG########### 

```{r plotAcrossRegions_mouse-one-pager, exercise=TRUE, echo=FALSE}
plotAcrossRegionsMouse("YOUR MGI ID HERE")
```

```{r plotAcrossRegions_mouse-try-it-a, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_mouse-try-it-b, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_mouse-try-it-c, exercise=TRUE, echo=FALSE}

```



## DESeq2: Differential expression analysis

- **Objective**: Identify what genes are significantly differentially expressed between different conditions
- **Example question**: What genes are expressed in the anterior vs posterior regions of the gut? What genes are expressed in muscle cells vs. neurons? 
- **Why is this important**: It can help us figure out what genes are important in different places of the body and/or the function of the gene/body part. 

Not every gene is made in every cell in the body. Different cells have different genes “turned on” to create specialized shapes and functions. When we say a gene is on, we mean that the cell is expressing that gene – creating the protein the gene codes for. Higher gene expression means that more of that protein is being made in that cell/body part/condition compared to some other cell/body part/condition. When one gene is highly expressed in one condition compared to others, this suggests that the gene serves an important, perhaps specialized function, in this condition.

### REGION1 and REGION2

![](images/deseq2-regions.png){width=100%}

- **REGION1**: Put the name of one of the groups you're interested in here.
- **REGION2**: Put the other group you're interested in here. This region is the baseline condition.


### Interpreting results

- **baseMean**: The mean expression level for this gene across all samples in reads
- **log2FoldChange**: How much higher this gene is expressed in the first region (REGION1) than in the second region (REGION2) in a log2 scale. A positive number means the gene is higher in REGION1 than in REGION2, and a negative number means the gene is lower in REGION1 than in REGION2
- ***padj***: (If this variable doesn’t show up on your screen, click the little arrow at the top right of the table, next to the column names. This will let you scroll through all the columns if they don’t fit on your screen). This is the “adjusted p-value” for the gene after statistical corrections are made for multiple comparisons. Use this, not pvalue when reporting your results.


### Differential expression template (Drosophila midgut dataset)

```{r deseq2-table, exercise=TRUE, echo=FALSE}
temp <- results(midgut, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
R1_vs_R2
```

```{r try_it_out_deseq2-table-1, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-table-2, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-table-3, exercise=TRUE, echo=FALSE}

```
### Differential expression template (Mouse gutbrain dataset)

```{r deseq2-tablemouse, exercise=TRUE, echo=FALSE}
temp <- results(gutbrain, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
R1_vs_R2
```

```{r try_it_out_deseq2-table-4, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-table-5, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_deseq2-table-6, exercise=TRUE, echo=FALSE}

```


## ClusterProfiler: Differential expression analysis

- **Objective(s)**: **1)** Get information on what the biological importance of the differentially expressed genes (what do they do?), **2)** identify what biological pathways have the highest count of differentially expressed genes, and **3)** get the name of genes within these pathways.

- **Example question**: What biological pathways have the most number of differentially expressed genes?

- **Why is this important**: We can use this analysis to get genes of interest and also get a rough idea of how many related genes are also differentially expressed.

### REGION 1, REGION2, and a category

- **REGION1**: Put the name of one of the groups you're interested in here.
- **REGION2**: Put the other group you're interested in here. This region is the baseline condition.
- **YOUR CATEGORY HERE**: Choose a category from the dotplot on the Y-axis. Make sure your spelling is correct. This input is also case sensitive. If you do not include this line of code or provide a category the analysis will not give you the gene names.

### Interpreting results

- **p.adjust:** A significance score assigned to the overall pathway that tells us to what degree the pathway is differentially expressed. A lower number (more red in the dotplot) represents a higher significance score. In this case, our dotplot only shows significantly differentially expressed genes (padj<0.05). However, some genes are still more significantly differentially expressed than others.
- **count:** The number of differentially expressed genes in a given pathway.
- **GeneRatio:** The proportion of genes in a given pathway that are differentially expressed, given as a decimal. Imagine a scenario where you have 5 differentially expressed genes in a pathway of 10 genes. A whopping 50% of the genes in the pathway are differentially expressed! This could signify a deep link between this biological pathway and your two conditions. If you instead have 5 differentially expressed genes in a pathway of 100, that means only 5% of genes within the pathway are different between the two conditions. While this does not discredit these individual genes, it implies that overall the pathway is not strongly affected between the two conditions.
- **List of genes**: If a category from the y-axis is swapped out for YOUR CATEGORY HERE, a list of differentially expressed genes in the chosen category/pathway will be listed.

### ClusterProfiler the for Drosophila midgut dataset

```{r clusterprofiler_drosophila, exercise=TRUE, echo=FALSE}
# Run differential expression analysis
temp <- results(midgut, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
sig_genes <- filter(R1_vs_R2, padj <= 0.01)

# Run ClusterProfiler analysis
R1_vs_R2_clusters <- runClusterProfiler(sig_genes)

# Look at the dotplot to see the groups of genes
dim(R1_vs_R2_clusters)
dotplot(R1_vs_R2_clusters, showCategory=34, title="YOUR TITLE HERE", font.size=10, label_format = 50)

# Get the gene symbols for a group of genes
getClusterProfilerGenes(R1_vs_R2_clusters, "YOUR CATEGORY HERE")
```

```{r try_it_out_clusterprofiler_drosophila-1, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_drosophila-2, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_drosophila-3, exercise=TRUE, echo=FALSE}

```


### ClusterProfiler for the mouse gutbrain dataset

```{r clusterprofiler_mouse, exercise=TRUE, echo=FALSE}
# Run differential expression analysis
temp <- results(gutbrain, contrast = c("condition", "REGION1", "REGION2") )
R1_vs_R2 <- formatDESeq2Results(temp)
sig_genes <- filter(R1_vs_R2, padj <= 0.01)

# Run ClusterProfiler analysis
R1_vs_R2_clusters <- runClusterProfilerMouse(sig_genes)

# Look at the dotplot to see the groups of genes
dim(R1_vs_R2_clusters)
dotplot(R1_vs_R2_clusters, showCategory=34, title="YOUR TITLE HERE", font.size=10, label_format = 50)

# Get the gene symbols for a group of genes
getClusterProfilerGenesMouse(R1_vs_R2_clusters, "YOUR CATEGORY HERE")
```

```{r try_it_out_clusterprofiler_mouse-4, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_mouse-5, exercise=TRUE, echo=FALSE}

```
```{r try_it_out_clusterprofiler_mouse-6, exercise=TRUE, echo=FALSE}

```

## plotAcrossRegions

- **Objective**: Create a graph that shows us the gene expression for a single gene across all conditions (regions)

- **Example questions**: In what region of the midgut is this gene expressed? Is this gene expressed in both the striatum and prefrontal cortex? In both ASD and control individuals?

- **Why is this important**: This helps us profile what genes are specialized for a given condition based on their level of gene expression across all conditions. Some genes may be used in all conditions and have a less specialized function. Genes that exclusively expressed in only some (or one) condition at high levels likely play a specialized function in that condition.

### plotAcrossRegions the for Drosophila midgut dataset

- FlyBase IDs take the format FBgn#######

```{r plotAcrossRegions_drosophila, exercise=TRUE, echo=FALSE}
plotAcrossRegions("YOUR FLYBASE ID HERE")
```


```{r plotAcrossRegions_drosophila-try-it-1, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_drosophila-try-it-2, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_drosophila-try-it-3, exercise=TRUE, echo=FALSE}

```


### plotAcrossRegionsMouse the for mouse gutbrain dataset

- MGI IDs take the format ENSMUG########### 

```{r plotAcrossRegions_mouse, exercise=TRUE, echo=FALSE}
plotAcrossRegionsMouse("YOUR MGI ID HERE")
```

```{r plotAcrossRegions_mouse-try-it-1, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_mouse-try-it-2, exercise=TRUE, echo=FALSE}

```
```{r plotAcrossRegions_mouse-try-it-3, exercise=TRUE, echo=FALSE}

```


## Works cited

- Sadaqat, Zoha, Shivam Kaushik, and Pinky Kain. "Gut Feeding the Brain: Drosophila Gut an Animal Model for Medicine to Understand Mechanisms Mediating Food Preferences." Preclinical Animal Modeling in Medicine. IntechOpen, 2021.

- Love, Michael I., Wolfgang Huber, and Simon Anders. "Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2." Genome biology 15 (2014): 1-21.


- Aden-Buie G, Schloerke B, Allaire J, Rossell Hayes A (2023). learnr: Interactive Tutorials for R. <https://rstudio.github.io/learnr/>, <https://github.com/rstudio/learnr>

- Sharon, Gil, et al. "Human gut microbiota from autism spectrum disorder promote behavioral symptoms in mice." Cell 177.6 (2019): 1600-1618.

- R Core Team (2024). R: A Language and Environment for Statistical Computing. R Foundation for Statistical Computing, Vienna, Austria.
  <https://www.R-project.org/>.
  
- Stoudt, Sara, Anthony D. Scotina, and Karsten Luebke. "Supporting Statistics and Data Science Education with learnr." Technology Innovations in Statistics Education 14.1 (2022).

- Wickham, Hadley. "ggplot2." Wiley interdisciplinary reviews: computational statistics 3.2 (2011): 180-185.














