<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Sayumi York, Notre Dame of Maryland University" />


<meta name="progressive" content="false" />
<meta name="allow-skip" content="false" />
<meta name="learnr-version-prerender" content="0.11.5" />

<title>16S rRNA Human Gut Microbiome Project Work</title>

<!-- header-includes START -->
<!-- HEAD_CONTENT -->
<!-- header-includes END -->
<!-- HEAD_CONTENT -->

<!-- highlightjs -->
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>


<!-- taken from https://github.com/rstudio/rmarkdown/blob/de8a9c38618903627ca509f5401d50a0876079f7/inst/rmd/h/default.html#L293-L343 -->
<!-- tabsets -->
<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- end tabsets -->

<link rel="stylesheet" href="css/cmoor.css" type="text/css" />

</head>

<body>
<a class='sr-only sr-only-focusable visually-hidden-focusable' href='#learnr-tutorial-content'>Skip to Tutorial Content</a>



<div class="pageContent band">
<main class="bandContent page">

<article class="topics" id="learnr-tutorial-content">

<!---
Don't edit the Welcome page, it will be filled in automatically using the information from the YAML header
Edit the rest of the document as you like
There are some suggested sections to provide a standard order across our tutorials, but they may not all be needed/appropriate for all tutorials.
Section 1. Content 1 has example quizes and exercises
-->
<div id="section-welcome" class="section level2 splashpage">
<h2 class="splashpage">Welcome</h2>
<div id="section-s-rrna-human-gut-microbiome-project-work"
class="section level3">
<h3>16S rRNA Human Gut Microbiome Project Work</h3>
<div class="splashpage-container">
<figure class="splashpage-image">
{width=100%}
<figcaption class="caption">
</figcaption>
</figure>
<p>This module is meant to provide students with a working space for
their 16S rRNA human gut microbiome projects. In addition to the MISO
dataset from the other tutorials, we have a number of other datasets you
can explore here as well as code chunks and starting off points to help
get you started!</p>
</div>
<div id="section-learning-goals" class="section level4">
<h4>Learning Goals</h4>
<ol>
<li>
Familiarize the datasets available to you
</li>
<li>
Select a dataset that you’d like to work on for your research project
</li>
<li>
Begin your exploratory analysis of your chosen dataset to refine your
research questions
</li>
</ol>
</div>
<div id="section-authors" class="section level4">
<h4>Authors:</h4>
<ul>
<li>
Sayumi York, Notre Dame of Maryland University
</li>
</ul>
</div>
<div id="section-version-2" class="section level4">
<h4>Version: 2</h4>
</div>
</div>
</div>
<div id="section-meet-miso" class="section level2">
<h2>Meet MISO</h2>
<p>These data are from <em>Impact of a 7-day homogeneous diet on
interpersonal variation in human gut microbiomes and metabolomes</em> by
Guthrie et al. (2022).</p>
<p>It’s difficult to capture the mechanisms and effects of diet on the
gut microbiome given the sheer number of variables involved. The
microbiome individuality and stability over time (MISO) study aimed to
explore the connection between diet, microbiome, and metabolites by
looking at the effect of feeding a standardized diet to people over 7
days.</p>
<div id="section-miso-study-diet-and-sampling-schedule"
class="section level3">
<h3>MISO study diet and sampling schedule</h3>
<img src="images/miso_experimental_design.png" style="width:100.0%" />
<p class="caption">
<strong>Experimental design for the MISO study by Guthrie et
al. (2022)</strong>: Figure edited by Sayumi York (July 11, 2025).
</p>
<p>The figure above shows the study design for the MISO study. You do
not need to memorize all these details; feel free to refer to this page
throughout the project:</p>
<ul>
<li>Participants eat their usual, <strong>baseline diet (BD) for 14
days</strong></li>
<li>Participants all eat the same diet, the <strong>homogenized diet
(HD), for 7 days</strong></li>
<li>Participants return to their usual diet during the <strong>washout
(WO) period for 7 days</strong></li>
</ul>
<p>The study lasts a total of 28 days. Samples from the blood, stool,
and urine, our metabolite and 16S rRNA data are taken at 5 different
timepoints:</p>
<ul>
<li><strong>Timepoint 1:</strong> Day 0 (the start of the study)</li>
<li><strong>Timepoint 2:</strong> Day 13</li>
<li><strong>Timepoint 3:</strong> Day 17</li>
<li><strong>Timepoint 4:</strong> Day 21</li>
<li><strong>Timepoint 5:</strong> Day 28</li>
</ul>
</div>
<div id="section-miso-study-variables-and-factors"
class="section level3">
<h3>MISO study variables and factors</h3>
<img src="images/miso_variables.png" style="width:100.0%" />
<p class="caption">
<strong>Metadata variables for the MISO study by Guthrie et
al. (2022)</strong>: Figure edited by Sayumi York (July 11, 2025).
</p>
<p>We have a number of variables we can use in our analysis. Notice that
the variables for the study and subjects are in lowercase. This is also
how you will access these variables in R.</p>
</div>
<div id="section-miso-study-design-variables" class="section level3">
<h3>MISO study design variables</h3>
<!-- How to align table using this format? -->
<table>
<colgroup>
<col width="20%" />
<col width="58%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">timepoint</td>
<td>The 5 samplings that occur on days 0, 13, 17, 21, and 28 coded as
timepoints 1 through 5</td>
<td align="right">1, 2, 3, 4, 5</td>
</tr>
<tr class="even">
<td align="left">timepoint group</td>
<td>The diet the subject was on during the sampling</td>
<td align="right">BD, HD, WO</td>
</tr>
<tr class="odd">
<td align="left">phase</td>
<td>Whether the subject was on their usual diet or the homogenized diet
during sampling</td>
<td align="right">non-miso, miso</td>
</tr>
</tbody>
</table>
</div>
<div id="section-miso-subject-variables" class="section level3">
<h3>MISO subject variables</h3>
<table>
<colgroup>
<col width="18%" />
<col width="36%" />
<col width="45%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">subject</td>
<td>A unique ID given to each subject (participant)</td>
<td align="right">S## (ex. S02 is subject 2. Note that while there are a
total of 21 subjects in the study, but their subject numbers are not 1
through 20</td>
</tr>
<tr class="even">
<td align="left">sample</td>
<td>A unique ID given to every sample taken during the study that
includes the subject and timepoint of the sample</td>
<td align="right">MISO-Subject##-Sample# (ex. MISO1-S02-1 is the sample
from subject 2 at timepoint 1)</td>
</tr>
<tr class="odd">
<td align="left">gender</td>
<td>The gender of the subject</td>
<td align="right">M, F. All subjects were cisgender</td>
</tr>
<tr class="even">
<td align="left">age</td>
<td>The age in years of the subject</td>
<td align="right">A continuous variable from 23 to 75 years old</td>
</tr>
<tr class="odd">
<td align="left">bsa</td>
<td>Body surface area; a measure of body size</td>
<td align="right">A continuous variable from 1.6 to 2.8</td>
</tr>
</tbody>
</table>
<div id="section-miso-metabolite-variables" class="section level4">
<h4>MISO metabolite variables</h4>
<table>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Creatinine</td>
<td>Creatinine</td>
<td align="right">A continuous variable from 1072 to 3971</td>
</tr>
<tr class="even">
<td align="left">PCS</td>
<td><em>p</em>-cresol sulfate</td>
<td align="right">A continuous variable from 2 to 95</td>
</tr>
<tr class="odd">
<td align="left">IS</td>
<td>Indoxyl sulfate</td>
<td align="right">A continuous variable from 3 to 58</td>
</tr>
<tr class="even">
<td align="left">HIPP</td>
<td>Hippuric acid</td>
<td align="right">A continuous variable from 16 to 1119</td>
</tr>
<tr class="odd">
<td align="left">PAG</td>
<td>Phenylacetylglutamine</td>
<td align="right">A continuous variable from 16 to 318</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="section-miso-phyloseq-objects" class="section level3">
<h3>MISO phyloseq objects</h3>
<div class="tutorial-exercise" data-label="miso_objects"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#A phyloseq object containing the normalized (proportions) data
miso

#A phyloseq object containing the data as raw counts
miso_counts</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="miso_block1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="miso_block2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="miso_block3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="miso_block4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="miso_block5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="miso_block6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-meet-fefifo" class="section level2">
<h2>Meet FeFiFo</h2>
<p>These data are from <strong>Gut-microbiota-targeted diets modulate
human immune status</strong> by Wastyk et al. (2021). We call this the
FeFiFo study because the experimental design includes <em>fe</em>rmented
and <em>fi</em>ber-rich <em>fo</em>od treatment groups!</p>
<p>This study looked at the effect of either high fiber or fermented
foods. Rather than controlling diet entirely through standardized
portions, participants were allowed to select what foods they ate so
long as these foods were high in fiber or were fermented. Additional
metadata was collected from participants including ethnicity,
relationship status, employment status, and education level.</p>
<div id="section-fefifo-study-diet-and-sampling-schedule"
class="section level3">
<h3>FeFiFo study diet and sampling schedule</h3>
<img src="images/fefifo_experimental_design.png" style="width:100.0%" />
<p class="caption">
<strong>Experimental design for the FeFiFo study by Wastyk et
al. (2021)</strong>: Figure edited by Sayumi York (July 11, 2025).
</p>
<p>The figure above shows the study design for the FeFiFo study. You do
not need to memorize all these details; feel free to refer to this page
throughout the project:</p>
<ul>
<li>Participants eat their usual, <strong>baseline diet for 3
weeks</strong></li>
<li>Participants slowly increase their intake of intervention (fiber or
fermented, not both!) foods, in the <strong>ramp for 4
weeks</strong></li>
<li>Participants maintain their intervention during a
<strong>maintenance period of 6 weeks</strong></li>
<li>Participants can choose to eat what they want, including any
intervention foods during the <strong>choice period of 4
weeks</strong></li>
</ul>
<p>The study lasts a total of 17 weeks. Samples from the gut microbiota
(among other blood samples) at <strong>9 different timepoints</strong>
across all phases of the experiment. Unlike the MISO study, these
timepoints are not assigned to distinct phases as some of them were
taken during the transition periods between phases.</p>
</div>
<div id="section-fefifo-study-design-variables" class="section level3">
<h3>FeFiFo study design variables</h3>
<img src="images/fefifo_variables.png" style="width:100.0%" />
<p class="caption">
<strong>Metadata variables for the FeFiFo study by Guthrie et
al. (2022)</strong>: Figure edited by Sayumi York (July 11, 2025)
</p>
<p>We have a number of variables we can use in our analysis. Notice that
the variables for the study and subjects are in lowercase. This is also
how you will access these variables in R.</p>
</div>
<div id="section-fefifo-study-design-variables-1"
class="section level3">
<h3>FeFiFo study design variables</h3>
<table>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">timepoint</td>
<td>The 9 samplings that occur throughout the experiment</td>
<td align="right">1-9</td>
</tr>
<tr class="even">
<td align="left">diet</td>
<td>The diet the subject was on during the sampling</td>
<td align="right">Fiber, Fermented</td>
</tr>
</tbody>
</table>
</div>
<div id="section-fefifo-subject-variables" class="section level3">
<h3>FeFiFo subject variables</h3>
<table>
<colgroup>
<col width="20%" />
<col width="58%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">subject</td>
<td>a unique id given to the participant</td>
<td align="right">A number between 8001 and 8041</td>
</tr>
<tr class="even">
<td align="left">age</td>
<td>The age in years of the subject</td>
<td align="right">26.1-74.5</td>
</tr>
<tr class="odd">
<td align="left">gender</td>
<td>The gender of the subject</td>
<td align="right">Male, Female. All subjects were cisgender</td>
</tr>
<tr class="even">
<td align="left">ethnicity</td>
<td>The ethnicity of the subject</td>
<td align="right">Asian, White.</td>
</tr>
<tr class="odd">
<td align="left">relationship</td>
<td>The relationship status of the subject</td>
<td align="right">Single/Never married, Married/Partnered, Divorced</td>
</tr>
<tr class="even">
<td align="left">education</td>
<td>The education of the subject</td>
<td align="right">Some college, College graduate ,Some post-graduate
school, Post-graduate degree</td>
</tr>
<tr class="odd">
<td align="left">employment</td>
<td>The employment status of the subject</td>
<td align="right">Unemployed, Working part-time, Working full-time,
Retired</td>
</tr>
<tr class="even">
<td align="left">height</td>
<td>The height of the subject (meters)</td>
<td align="right">1.52-1.87</td>
</tr>
<tr class="odd">
<td align="left">weight</td>
<td>The height of the subject (kg)</td>
<td align="right">44.3-100.6</td>
</tr>
<tr class="even">
<td align="left">bmi</td>
<td>The bmi of the subject</td>
<td align="right">17.71-31.68</td>
</tr>
<tr class="odd">
<td align="left">waist</td>
<td>The waist measurement of the subject (cm)</td>
<td align="right">27.56-46.85</td>
</tr>
</tbody>
</table>
</div>
<div id="section-some-research-ideas-to-get-you-started"
class="section level3">
<h3>Some research ideas to get you started</h3>
<ul>
<li>How variable is the change of microbiome composition and alpha
diversity across individuals? Do all individuals change the same amount
or are there some individuals much more sensitive to their chosen food
group?</li>
<li>What are the most abundant microbes in each treatment (fiber,
fermented)? Are they similar taxa or different?</li>
<li>Is there any relationship between ethnicity, education, employment,
relationship status and microbiome composition and/or alpha
diversity?</li>
<li>Are the subject’s microbiomes at a given timepoint similar to each
other? Similar to other samples from the same subject no matter the
timepoint?</li>
</ul>
</div>
<div id="section-fefifo-phyloseq-objects" class="section level3">
<h3>FeFiFo phyloseq objects</h3>
<div class="tutorial-exercise" data-label="fefifo_objects"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#A phyloseq object containing the normalized (proportions) data
fefifo

#A phyloseq object containing the data as raw counts
fefifo_counts</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="fefifo_block1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="fefifo_block2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="fefifo_block3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="fefifo_block4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="fefifo_block5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="fefifo_block6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-meet-the-alzheimers-study" class="section level2">
<h2>Meet the Alzheimer’s study</h2>
<p>These data are from <strong>Gut microbiota is altered in patients
with Alzheimer’s disease</strong> by Zhuang et al. (2018). For
convenience’s sake we will refer to it as our AD study.</p>
<p>This study compares the gut microbiomes of subjects with a clinical
diagnosis of Alzheimer’s disease with healthy age and sex matched
individuals. While we can’t control for all health conditions, the study
attempted to limit subjects to participants who match all the
following:</p>
<ol style="list-style-type: decimal">
<li>Have no family history of dementia</li>
<li>Have no other diagnosis of a neurodegenerative disease</li>
<li>Have no other severe conditions such as cancer</li>
<li>Have no chronic mental illness</li>
<li>Have no history of antibiotic use within the last 6 months</li>
<li>Have no gastrointestinal disease (ex. IBS)</li>
</ol>
<p>The study design was very simple. Fecal samples were taken from
subjects - no timeline or intervention included!</p>
<div id="section-ad-study-variables" class="section level3">
<h3>AD study variables</h3>
<p><img src="images/ad_variables.png" style="width:100.0%" /></p>
<p class="caption">
<strong>Metadata variables for the AD study by Zhuang et
al. (2018)</strong>: Figure edited by Sayumi York (July 11, 2025)
</p>
<p>We have a number of variables we can use in our analysis. Notice that
the variables for the study and subjects are in lowercase. This is also
how you will access these variables in R.</p>
<table>
<colgroup>
<col width="20%" />
<col width="58%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">sample_id</td>
<td>a unique id given to the participant</td>
<td align="right">Begins with either AD or CON (Control)</td>
</tr>
<tr class="even">
<td align="left">age</td>
<td>The age in years of the subject</td>
<td align="right">55-85</td>
</tr>
<tr class="odd">
<td align="left">sex</td>
<td>The gender of the subject</td>
<td align="right">male, female</td>
</tr>
<tr class="even">
<td align="left">group</td>
<td>Whether the subject was in the AD group or control group</td>
<td align="right">AD, control.</td>
</tr>
</tbody>
</table>
</div>
<div id="section-some-research-ideas-to-get-you-started-1"
class="section level3">
<h3>Some research ideas to get you started</h3>
<ul>
<li>How similar are microbiomes between healthy and AD individuals in
composition and/or alpha diversity?</li>
<li>Looking at other published research studies, are the microbial taxa
associated with AD also found in this study?</li>
<li>Select a microbial taxa of interest (perhaps from the idea above)
and explore it at increasing levels of resolution up to the level of
species. Looking at other published research studies, what functions are
these taxa associated with? Do they have any relation to AD or other
neurodegenerative diseases?</li>
</ul>
</div>
<div id="section-ad-objects" class="section level3">
<h3>AD objects</h3>
<div class="tutorial-exercise" data-label="ad_objects"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#A phyloseq object containing the normalized (proportions) data
ad

#A phyloseq object containing the data as raw counts
ad_counts</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="ad_block1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="ad_block2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="ad_block3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="ad_block4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="ad_block5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="ad_block6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-meet-the-oc-study" class="section level2">
<h2>Meet the OC study</h2>
<p>These data are from <strong>Influence of menstrual cycle and oral
contraception on taxonomic composition and gas production in the gut
microbiome</strong> by Terrazas et al. (2025). For convenience’s sake we
will refer to it as our OC study.</p>
<p>This study compared the microbiomes of women who were on OC and those
who were not (a history of at least 3 months without taking an OC).</p>
<div id="section-oc-study-variables" class="section level3">
<h3>OC study variables</h3>
<p><img src="images/oc_variables.png" style="width:100.0%" /></p>
<p class="caption">
<strong>Metadata variables for the OC study by Terrazas et
al. (2025)</strong>: Figure edited by Sayumi York (July 11, 2025)
</p>
<table>
<colgroup>
<col width="20%" />
<col width="58%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Variable</th>
<th>What is it?</th>
<th align="right">Factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">sample_id</td>
<td>a unique id given to each sample</td>
<td align="right">Fomatted with subject treatment timepoint (ex.701C2 is
subject 701, who is a control, this is the 2nd timepoint sampled</td>
</tr>
<tr class="even">
<td align="left">subject</td>
<td>a unique id given to the participant</td>
<td align="right">A number between 701 and 990</td>
</tr>
<tr class="odd">
<td align="left">treatment</td>
<td>Whether the subject is on OC or is a control</td>
<td align="right">Control, User</td>
</tr>
<tr class="even">
<td align="left">diet</td>
<td>Whether the subject’s diet contained prunes</td>
<td align="right">Prune, No Prune</td>
</tr>
<tr class="odd">
<td align="left">day</td>
<td>Days into the study the sample was collected</td>
<td align="right">Day 1, Day 21</td>
</tr>
<tr class="even">
<td align="left">age</td>
<td>The age in years of the subject</td>
<td align="right">18-24</td>
</tr>
<tr class="odd">
<td align="left">ethnicity</td>
<td>The ethnicity of the subject</td>
<td align="right">Asian, White, Hispanic, Black, NA</td>
</tr>
<tr class="even">
<td align="left">height</td>
<td>The height of the subject (cm)</td>
<td align="right">149.86-177.8</td>
</tr>
<tr class="odd">
<td align="left">weight</td>
<td>The height of the subject (kg)</td>
<td align="right">44.6-96.9</td>
</tr>
<tr class="even">
<td align="left">bmi</td>
<td>The bmi of the subject</td>
<td align="right">17.86-32.20</td>
</tr>
<tr class="odd">
<td align="left">ageofmenarche</td>
<td>The age of the subject at menarche (first menstrual cycle)</td>
<td align="right">10-15</td>
</tr>
<tr class="even">
<td align="left">avgnumdaysbtwperiods</td>
<td>The average number of days between periods for the subject</td>
<td align="right">20-120</td>
</tr>
<tr class="odd">
<td align="left">OCE2Dosage</td>
<td>The dosage of the oral contracaptive (oestrogen, in mcg)</td>
<td align="right">20, 30, 35</td>
</tr>
<tr class="even">
<td align="left">agefirstOCuse</td>
<td>Age when starting OC</td>
<td align="right">14-24.5</td>
</tr>
<tr class="odd">
<td align="left">totalyearsonOC</td>
<td>Total years on OC</td>
<td align="right">0.3-9</td>
</tr>
<tr class="even">
<td align="left">BMperday</td>
<td>Number of bowel movements per day</td>
<td align="right">0.57-4</td>
</tr>
<tr class="odd">
<td align="left">BMperwk</td>
<td>Number of bowel movements per week</td>
<td align="right">4-28</td>
</tr>
</tbody>
</table>
</div>
<div id="section-some-research-ideas-to-get-you-started-2"
class="section level3">
<h3>Some research ideas to get you started</h3>
<ul>
<li>Is there any relationship between ethnicity, diet, OCE2dosage, or
bowel movements and microbiome composition and/or alpha diversity?</li>
<li>Create a multidimensional analysis: how do samples cluster? By diet?
Age? Subject? Dosage? Treatment?</li>
<li>How do the microbial compositions of different treatments at
increasing levels of resolution (choosing the most abundant group each
time)?</li>
</ul>
</div>
<div id="section-oc-objects" class="section level3">
<h3>OC objects</h3>
<div class="tutorial-exercise" data-label="oc_objects"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#A phyloseq object containing the normalized (proportions) data
oc

#A phyloseq object containing the data as raw counts
oc_counts

#A phyloseq object containing the normalized (proportions) data without the outlier sample (948N2)
oc_no948N2

#A phyloseq object containing the data as raw counts without the outlier sample (948N2)
oc_no948N2_counts</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="oc_block1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="oc_block2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="oc_block3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="oc_block4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="oc_block5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="oc_block6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-bar-plots" class="section level2">
<h2>Bar plots</h2>
<ul>
<li><strong>Objective</strong>: Vizualize the microbial composition of
samples</li>
<li><strong>Best for questions like</strong>: What microbial taxa are
dominant across samples? What drives differences in microbial
composition between conditions?</li>
<li><strong>Example question</strong>: How does the microbial
composition differ between patients on their typical diet and a high
fiber diet?</li>
<li><strong>Data format</strong>: To make it easy to understand and
visualize the data we will use data that has been normalized by
proportions.</li>
</ul>
<div id="section-bar-plot-code-template" class="section level3">
<h3>Bar plot code template</h3>
<div class="tutorial-exercise" data-label="barplot" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>#Subsetting based on a group
my_subset &lt;- subset_taxa(miso, Phylum == &quot;Firmicutes&quot;)

#Plotting using the new subset
plot_bar(my_subset, x= &quot;subject&quot;, fill = &quot;Class&quot;, title = &quot;My title&quot;) +
  geom_bar(aes(color = Class, fill = Class), stat = &quot;identity&quot;, position = &quot;stack&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="try_it_out_barplot1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="try_it_out_barplot2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="try_it_out_barplot3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="try_it_out_barplot4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="try_it_out_barplot5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="try_it_out_barplot6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-alpha-diversity" class="section level2">
<h2>Alpha diversity</h2>
<!-- Uses raw count data -->
<ul>
<li><strong>Objective</strong>: Compare the biodiversity (richness and
evenness) between samples/groups</li>
<li><strong>Best for questions like</strong>: How variable or stable is
the microbial community over time or space?</li>
<li><strong>Example question</strong>: Does changing the diet reduce
biodiversity in the human gut? Are male microbiomes more diverse than
female microbiomes?</li>
<li><strong>Data format</strong>: Because we want to include rare
species in our analysis, we will use our raw data, which is in
counts.</li>
</ul>
<div id="section-alpha-diversity-code-template" class="section level3">
<h3>Alpha diversity code template</h3>
<div class="tutorial-exercise" data-label="try_it_plot_richness_end"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>my_subset &lt;- subset_samples(miso_counts, gender == &quot;F&quot;)

plot_richness(my_subset, x=&quot;timepoint&quot;, 
              color=&quot;phase&quot;, 
              shape=&quot;timepointgroup&quot;,
              title =&quot;fill in the blank&quot;, 
              measures= &quot;Simpson&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_plot_richness_blank6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-multidimensional-analysis" class="section level2">
<h2>Multidimensional analysis</h2>
<div id="section-pcoa" class="section level3">
<h3>PCoA</h3>
<ul>
<li><strong>Objective</strong>: Comparing how similar samples are to one
another</li>
<li><strong>Best for questions like</strong>: How much variability is
there within a given condition? What samples are most similar to each
other?</li>
<li><strong>Example question</strong>: Are male and female samples
grouped together? Do samples group together by a given metabolite? How
similar are samples within each timepointgroup?</li>
<li><strong>Data format</strong>: We will use normalized counts
(proportions) to be able to make comparisons between samples.</li>
</ul>
</div>
<div id="section-how-to-interpret" class="section level3">
<h3>How to interpret</h3>
<p>Samples that cluster together have more in common than samples that
are farther apart.</p>
</div>
<div id="section-multidimensional-analysis-code-template"
class="section level3">
<h3>Multidimensional analysis code template</h3>
<div class="tutorial-exercise" data-label="Try_it_PCoA_end"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>#Subset a group of interest
miso_subset &lt;- subset_taxa(miso, Phylum == &quot;Firmicutes&quot;)

#Ordinate based on your subset
miso.pcoa.subset &lt;- ordinate(miso, method=&quot;PCoA&quot;, distance=&quot;bray&quot;)

#Plot the new subsetted ordination with the subsetted phyloseq object
plot_ordination(miso_subset, miso.pcoa.subset, 
        color = &quot;subject&quot;, 
        shape = &quot;phase&quot;, 
        title=&quot;My subsetted (Firmicutes) PCoA&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_PCoA_blank6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-differential-abundance" class="section level2">
<h2>Differential abundance</h2>
<ul>
<li><strong>Objective</strong>: Compare the change in abundance of ASVs
between groups (conditions)</li>
<li><strong>Best for questions like</strong>: What ASVs showed the most
change in abundance between groups (conditions)?</li>
<li><strong>Example question</strong>: What ASVs changed the most
between timepoint 1 and 2? What trends are there in the change in
abundance for a given taxonomic group?</li>
<li><strong>Data format</strong>: DESeq2 will normlalize the data for us
in the way that best suits it, so we will use our raw count data</li>
</ul>
<div id="section-how-do-we-interpret-the-results"
class="section level3">
<h3>How do we interpret the results?</h3>
<ul>
<li><strong>log2FoldChange</strong>: The latter of the two conditions in
our chosen variable is what DESeq2 considers the baseline. For example,
the code: <em>my_comparision &lt;-c(“timepointgroup”, “WO”, “BD”)</em>,
sets the baseline as BD. Thus, any positive number represents a greater
abundance in the former (“WO”) condition compared to the latter -
baseline (“BD”). The larger the magnitude of the number, the greater the
difference in abundance.</li>
<li><strong>p-value</strong>: The base p-value, without corrections for
multiple comparisons.</li>
<li><strong>p-adj</strong>: The p-value after correcting for multiple
comparisions. You should use p-adj when reporting your results.</li>
</ul>
<p><br></p>
<div class="attention_box_pause">
<div class="attention_header_box_pause">
<div id="section-differential-abundance-functions-special-cases"
class="section level4 attention_title">
<h4><strong>Differential abundance functions: special
cases</strong></h4>
</div>
</div>
<div class="attention_content">
<p>We have two special cases for our differential expression
analysis.</p>
<ol style="list-style-type: decimal">
<li><p><strong>Differential abundance concerned with a continuous
variables</strong> such as age, height, and metabolites: We will use the
function Differential_Abundance_Continuous() instead of
Differential_Abundance().</p></li>
<li><p><strong>Differential abundance with 0-inflated data</strong> that
gives us the error: <em>Error in
estimateSizeFactorsForMatrix(counts(object), locfunc = locfunc, : every
gene contains at least one zero, cannot compute log geometric
means</em></p></li>
</ol>
<p>Please refer to the code chunks below with examples for those
specific functions. Note that they have some slight differences from the
original function!</p>
</div>
</div>
</div>
<div id="section-differential-abundance-code-template"
class="section level3">
<h3>Differential abundance code template</h3>
<div class="tutorial-exercise" data-label="DESeq2_example"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design 
my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)

# STEP 2: Select the groups to compare
my_comparision &lt;-c(&quot;timepointgroup&quot;, &quot;WO&quot;, &quot;BD&quot;)

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs&lt;-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3, position = &quot;jitter&quot;) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle(&quot;my title here&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_blank1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_blank2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_blank3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_blank4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_blank5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_blank6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
<div
id="section-differential-abundance-for-continuous-data-code-template"
class="section level3">
<h3>Differential abundance for continuous data code template</h3>
<div id="section-whats-different-with-continuous-data"
class="section level4">
<h4>What’s different with continuous data?</h4>
<ul>
<li><strong>Interpreting the log2FoldChange</strong>: In this case, the
log2FoldChange is the difference in abundance for every unit increase of
the design factor. In the example below, log2FoldChange represents the
amount of change in the ASV’s abundance for every year the subject grows
older (since age is measured in years in the dataset).</li>
<li><strong>Changes in the code</strong>: We do not need to specify
groups for comparison. Notice how in step 2 we simply provide the
continuous variable we are interested in.</li>
</ul>
<div class="tutorial-exercise" data-label="Try_it_DA_continuous"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design
my_miso_DESeq2 &lt;- phyloseq_to_deseq2(miso_counts, design = ~ age)

# STEP 2: Select the groups to compare
name_for_comparision &lt;-&quot;age&quot;

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs&lt;-Differential_Abundance_Continuous(my_miso_DESeq2, name_for_comparision, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class, shape = Order)) + geom_point(size=6) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_Continuous_blank1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_Continuous_blank2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_Continuous_blank3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_Continuous_blank4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_Continuous_blank5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_Continuous_blank6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div
id="section-differential-abundance-for-0-inflated-data-code-template"
class="section level3">
<h3>Differential abundance for 0-inflated data code template</h3>
<div id="section-whats-different-with-0-inflated-data"
class="section level4">
<h4>What’s different with 0-inflated data?</h4>
<ul>
<li><strong>What’s going on underneath?</strong>: We have too many ASVs
with 0 counts in our dataset. DESeq2 was originally created for RNA-seq
data, where this phenomenon is much rarer than in microbiome data. We
will use a slightly different statistical approach in this function to
get around this problem.</li>
<li><strong>Changes in the code</strong>: In step 3 we will also provide
our count data in addition to our DESeq2 object, comparision groups, and
p-value cutoff.</li>
</ul>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design 
my_ad_DESeq2 &lt;- phyloseq_to_deseq2(ad_counts, design = ~group)

# STEP 2: Select the groups to compare
my_comparison &lt;- c(&quot;group&quot;, &quot;AD&quot;, &quot;control&quot;)

# STEP 3: Run the differential abundance analysis at the chosen p-value
Significant_DEseq2_ASVs &lt;- Differential_Abundance_AltGeo(ad_counts, my_ad_DESeq2, my_comparison, 0.05)

# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups
Significant_DEseq2_ASVs

# STEP 5: Plot the results with your chosen x axis and legend
ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3, position = &quot;jitter&quot;) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle(&quot;my title here&quot;)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf_blank1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf_blank2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf_blank3"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf_blank4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf_blank5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise" data-label="Try_it_DA_zero-inf_blank6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
</div>
<div id="section-common-errors" class="section level2">
<h2>Common errors</h2>
<p>No significant genes found No datapoint on the DESeq2 graph - missing
NA? Warning: Removed 3 rows containing missing values or values outside
the scale range (<code>geom_point()</code>).</p>
</div>
<div id="section-works-cited" class="section level2">
<h2>Works cited</h2>
<ul>
<li><p>Aden-Buie G, Schloerke B, Allaire J, Rossell Hayes A (2023).
learnr: Interactive Tutorials for R. <a
href="https://rstudio.github.io/learnr/"
class="uri">https://rstudio.github.io/learnr/</a>, <a
href="https://github.com/rstudio/learnr"
class="uri">https://github.com/rstudio/learnr</a></p></li>
<li><p>Evans, Ciaran, Johanna Hardin, and Daniel M. Stoebel. “Selecting
between-sample RNA-Seq normalization methods from the perspective of
their assumptions.” Briefings in bioinformatics 19.5 (2018):
776-792.</p></li>
<li><p>Guthrie, Leah, et al. “Impact of a 7-day homogeneous diet on
interpersonal variation in human gut microbiomes and metabolomes.” Cell
host &amp; microbe 30.6 (2022): 863-874.</p></li>
<li><p>McMurdie, Paul J., and Susan Holmes. “phyloseq: an R package for
reproducible interactive analysis and graphics of microbiome census
data.” PloS one 8.4 (2013): e61217.</p></li>
<li><p>McMurdie, Paul J., and Susan Holmes. “Waste not, want not: why
rarefying microbiome data is inadmissible.” PLoS computational biology
10.4 (2014): e1003531.</p></li>
<li><p>R Core Team (2024). R: A Language and Environment for Statistical
Computing. R Foundation for Statistical Computing, Vienna, Austria. <a
href="https://www.R-project.org/"
class="uri">https://www.R-project.org/</a>.</p></li>
<li><p>Stoudt, Sara, Anthony D. Scotina, and Karsten Luebke. “Supporting
Statistics and Data Science Education with learnr.” Technology
Innovations in Statistics Education 14.1 (2022).</p></li>
<li><p>Wickham, Hadley. “ggplot2.” Wiley interdisciplinary reviews:
computational statistics 3.2 (2011): 180-185.</p></li>
</ul>
<p>
<script type="application/shiny-prerendered" data-context="server-start">
#Load learnr
library(learnr)
knitr::opts_chunk$set(echo = FALSE)
tutorial_options(exercise.completion=TRUE) #Uncomment to enable 

#Load libraries
library("tidyverse")
library("phyloseq")
library("DESeq2")

#Uncomment exercise timelimit when deployed
tutorial_options(exercise.timelimit = 360)

#Load MISO_16S.prop.RDS
#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds
#Normalization by turning counts into proportions

# #Local path


miso <- readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS")
miso_counts <- readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS")

fefifo <- readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS")
fefifo_counts <- readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS")


ad <- readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS")
ad_counts <- readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS")

oc<-readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS")
oc_counts<-readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS")
oc_no948N2_counts<-readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS")
oc_no948N2<-readRDS("C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS")





#
# #########################
# #SciServer paths#
# ########################
# # #Alzheimer's disease
# ad <- readRDS("/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS")
# ad_counts<- readRDS("/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS")
# #
# # #FeFiFo
# fefifo <- readRDS("/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS")
# fefifo_counts<- readRDS("/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS")
# #
# # #MISO
# miso <- readRDS( "/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds" )
# miso_counts <- readRDS("/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds")
# #
# # #Oral contraceptives (and prunes)
# oc <- readRDS( "/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS" )
# oc_counts <- readRDS( "/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS" )
# oc_no948N2 <- readRDS( "/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS" )
# oc_no948N2_counts <- readRDS( "/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS" )
# #


# <!-- EXERCISE = should we see an interactive code chunk with this code? -->
# <!-- EVAL = should you run this code? -->
# <!-- ECHO = should you print the code? -->
# Significant_DEseq2_ASVs <-0
# sigtab<-0

Differential_Abundance <- function(x, y, z) {
  
 # Accept the input and give it a human friendly name
  my_DESeq2_object <- x
  my_comparision_list <- y
  my_pvalue <- z
  
  
  #Do the DESeq2 analysis
  my_DESeq2_object <- DESeq(my_DESeq2_object, test="Wald", fitType="parametric")
 
  #Change the contrast section to compare different variables and groups.
  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)
  
  #Select only abundances with a p-value of z or below
  alpha = my_pvalue
  sigtab = res[which(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(miso_counts)[rownames(sigtab), ], "matrix"))

  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results
  theme_set(theme_bw())
  scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
  }
  # Phylum order
  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))
  
  # Class order
  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))
  
  # Order order
  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
  
  # Family order
  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
  
  # Genus order
  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
  
  # Species order
  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))

  return(sigtab)
}

# For using a continuous variable during comparisons
# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable

Differential_Abundance_Continuous <- function(x, y, z) {
  
 # Accept the input and give it a human friendly name
  my_DESeq2_object <- x
  my_name <- y
  my_pvalue <- z
  
  
  #Do the DESeq2 analysis
  my_DESeq2_object <- DESeq(my_DESeq2_object, test="Wald", fitType="parametric")
 
  #Change the contrast section to compare different variables and groups.
  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)
  
  
  #Select only abundances with a p-value of z or below
  alpha = my_pvalue
  sigtab = res[which(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(miso_counts)[rownames(sigtab), ], "matrix"))

  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results
  theme_set(theme_bw())
  scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
  }
  # Phylum order
  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))
  
  # Class order
  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))
  
  # Order order
  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
  
  # Family order
  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
  
  # Genus order
  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
  
  # Species order
  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))

  return(sigtab)
}

# For 0-inflated data that gives the error:
# every gene contains at least one zero, cannot compute log geometric means

Differential_Abundance_AltGeo <- function(w, x, y, z) {
  
  # Accept the input and give it a human friendly name
  my_counts <- w
  my_DESeq2_object <- x
  my_comparision_list <- y
  my_pvalue <- z
  
  
  
  #Do the DESeq2 analysis
  
  gm_mean = function(x, na.rm = TRUE){
    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
  }
  
  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)
  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)
  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = "parametric")
  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)
  
  #Change the contrast section to compare different variables and groups.
  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)
  
  #Select only abundances with a p-value of z or below
  alpha = my_pvalue
  sigtab = res[which(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(my_counts)[rownames(sigtab), ], "matrix"))
  
  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results
  theme_set(theme_bw())
  scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
  }
  # Phylum order
  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))
  
  # Class order
  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))
  
  # Order order
  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
  
  # Family order
  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
  
  # Genus order
  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
  
  # Species order
  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
  x = sort(x, TRUE)
  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))
  
  return(sigtab)
}


</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::register_http_handlers(session, metadata = NULL)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::prepare_tutorial_state(session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::i18n_observe_tutorial_language(input, session)
</script>


<script type="application/shiny-prerendered" data-context="server">
session$onSessionEnded(function() {
        learnr:::event_trigger(session, "session_stop")
      })
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_objects-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_objects-code-editor`)), session)
output$`tutorial-exercise-miso_objects-output` <- renderUI({
  `tutorial-exercise-miso_objects-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_objects", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_objects", 
    code = "#A phyloseq object containing the normalized (proportions) data\nmiso\n\n#A phyloseq object containing the data as raw counts\nmiso_counts\n", 
    opts = list(label = "\"miso_objects\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_objects", exercise = TRUE, code = c("#A phyloseq object containing the normalized (proportions) data", 
        "miso", "", "#A phyloseq object containing the data as raw counts", 
        "miso_counts", ""), out.width.px = 624, out.height.px = 384, 
        params.src = "miso_objects, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_block1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_block1-code-editor`)), session)
output$`tutorial-exercise-miso_block1-output` <- renderUI({
  `tutorial-exercise-miso_block1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_block1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_block1", 
    code = "\n", opts = list(label = "\"miso_block1\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_block1", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "miso_block1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_block2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_block2-code-editor`)), session)
output$`tutorial-exercise-miso_block2-output` <- renderUI({
  `tutorial-exercise-miso_block2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_block2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_block2", 
    code = "\n", opts = list(label = "\"miso_block2\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_block2", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "miso_block2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_block3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_block3-code-editor`)), session)
output$`tutorial-exercise-miso_block3-output` <- renderUI({
  `tutorial-exercise-miso_block3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_block3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_block3", 
    code = "\n", opts = list(label = "\"miso_block3\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_block3", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "miso_block3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_block4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_block4-code-editor`)), session)
output$`tutorial-exercise-miso_block4-output` <- renderUI({
  `tutorial-exercise-miso_block4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_block4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_block4", 
    code = "\n", opts = list(label = "\"miso_block4\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_block4", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "miso_block4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_block5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_block5-code-editor`)), session)
output$`tutorial-exercise-miso_block5-output` <- renderUI({
  `tutorial-exercise-miso_block5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_block5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_block5", 
    code = "\n", opts = list(label = "\"miso_block5\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_block5", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "miso_block5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-miso_block6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-miso_block6-code-editor`)), session)
output$`tutorial-exercise-miso_block6-output` <- renderUI({
  `tutorial-exercise-miso_block6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "miso_block6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "miso_block6", 
    code = "\n", opts = list(label = "\"miso_block6\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "miso_block6", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "miso_block6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_objects-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_objects-code-editor`)), session)
output$`tutorial-exercise-fefifo_objects-output` <- renderUI({
  `tutorial-exercise-fefifo_objects-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_objects", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_objects", 
    code = "#A phyloseq object containing the normalized (proportions) data\nfefifo\n\n#A phyloseq object containing the data as raw counts\nfefifo_counts\n", 
    opts = list(label = "\"fefifo_objects\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_objects", exercise = TRUE, code = c("#A phyloseq object containing the normalized (proportions) data", 
        "fefifo", "", "#A phyloseq object containing the data as raw counts", 
        "fefifo_counts", ""), out.width.px = 624, out.height.px = 384, 
        params.src = "fefifo_objects, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_block1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_block1-code-editor`)), session)
output$`tutorial-exercise-fefifo_block1-output` <- renderUI({
  `tutorial-exercise-fefifo_block1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_block1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_block1", 
    code = "\n", opts = list(label = "\"fefifo_block1\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_block1", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "fefifo_block1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_block2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_block2-code-editor`)), session)
output$`tutorial-exercise-fefifo_block2-output` <- renderUI({
  `tutorial-exercise-fefifo_block2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_block2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_block2", 
    code = "\n", opts = list(label = "\"fefifo_block2\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_block2", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "fefifo_block2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_block3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_block3-code-editor`)), session)
output$`tutorial-exercise-fefifo_block3-output` <- renderUI({
  `tutorial-exercise-fefifo_block3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_block3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_block3", 
    code = "\n", opts = list(label = "\"fefifo_block3\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_block3", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "fefifo_block3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_block4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_block4-code-editor`)), session)
output$`tutorial-exercise-fefifo_block4-output` <- renderUI({
  `tutorial-exercise-fefifo_block4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_block4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_block4", 
    code = "\n", opts = list(label = "\"fefifo_block4\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_block4", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "fefifo_block4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_block5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_block5-code-editor`)), session)
output$`tutorial-exercise-fefifo_block5-output` <- renderUI({
  `tutorial-exercise-fefifo_block5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_block5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_block5", 
    code = "\n", opts = list(label = "\"fefifo_block5\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_block5", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "fefifo_block5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-fefifo_block6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-fefifo_block6-code-editor`)), session)
output$`tutorial-exercise-fefifo_block6-output` <- renderUI({
  `tutorial-exercise-fefifo_block6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "fefifo_block6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "fefifo_block6", 
    code = "\n", opts = list(label = "\"fefifo_block6\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "fefifo_block6", exercise = TRUE, code = c("", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "fefifo_block6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_objects-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_objects-code-editor`)), session)
output$`tutorial-exercise-ad_objects-output` <- renderUI({
  `tutorial-exercise-ad_objects-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_objects", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_objects", 
    code = "#A phyloseq object containing the normalized (proportions) data\nad\n\n#A phyloseq object containing the data as raw counts\nad_counts\n", 
    opts = list(label = "\"ad_objects\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_objects", exercise = TRUE, code = c("#A phyloseq object containing the normalized (proportions) data", 
        "ad", "", "#A phyloseq object containing the data as raw counts", 
        "ad_counts", ""), out.width.px = 624, out.height.px = 384, 
        params.src = "ad_objects, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_block1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_block1-code-editor`)), session)
output$`tutorial-exercise-ad_block1-output` <- renderUI({
  `tutorial-exercise-ad_block1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_block1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_block1", 
    code = "\n", opts = list(label = "\"ad_block1\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_block1", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "ad_block1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_block2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_block2-code-editor`)), session)
output$`tutorial-exercise-ad_block2-output` <- renderUI({
  `tutorial-exercise-ad_block2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_block2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_block2", 
    code = "\n", opts = list(label = "\"ad_block2\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_block2", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "ad_block2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_block3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_block3-code-editor`)), session)
output$`tutorial-exercise-ad_block3-output` <- renderUI({
  `tutorial-exercise-ad_block3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_block3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_block3", 
    code = "\n", opts = list(label = "\"ad_block3\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_block3", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "ad_block3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_block4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_block4-code-editor`)), session)
output$`tutorial-exercise-ad_block4-output` <- renderUI({
  `tutorial-exercise-ad_block4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_block4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_block4", 
    code = "\n", opts = list(label = "\"ad_block4\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_block4", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "ad_block4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_block5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_block5-code-editor`)), session)
output$`tutorial-exercise-ad_block5-output` <- renderUI({
  `tutorial-exercise-ad_block5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_block5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_block5", 
    code = "\n", opts = list(label = "\"ad_block5\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_block5", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "ad_block5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ad_block6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ad_block6-code-editor`)), session)
output$`tutorial-exercise-ad_block6-output` <- renderUI({
  `tutorial-exercise-ad_block6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ad_block6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "ad_block6", 
    code = "\n", opts = list(label = "\"ad_block6\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "ad_block6", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "ad_block6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_objects-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_objects-code-editor`)), session)
output$`tutorial-exercise-oc_objects-output` <- renderUI({
  `tutorial-exercise-oc_objects-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_objects", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_objects", 
    code = "#A phyloseq object containing the normalized (proportions) data\noc\n\n#A phyloseq object containing the data as raw counts\noc_counts\n\n#A phyloseq object containing the normalized (proportions) data without the outlier sample (948N2)\noc_no948N2\n\n#A phyloseq object containing the data as raw counts without the outlier sample (948N2)\noc_no948N2_counts\n", 
    opts = list(label = "\"oc_objects\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_objects", exercise = TRUE, code = c("#A phyloseq object containing the normalized (proportions) data", 
        "oc", "", "#A phyloseq object containing the data as raw counts", 
        "oc_counts", "", "#A phyloseq object containing the normalized (proportions) data without the outlier sample (948N2)", 
        "oc_no948N2", "", "#A phyloseq object containing the data as raw counts without the outlier sample (948N2)", 
        "oc_no948N2_counts", ""), out.width.px = 624, out.height.px = 384, 
        params.src = "oc_objects, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_block1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_block1-code-editor`)), session)
output$`tutorial-exercise-oc_block1-output` <- renderUI({
  `tutorial-exercise-oc_block1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_block1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_block1", 
    code = "\n", opts = list(label = "\"oc_block1\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_block1", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "oc_block1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_block2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_block2-code-editor`)), session)
output$`tutorial-exercise-oc_block2-output` <- renderUI({
  `tutorial-exercise-oc_block2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_block2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_block2", 
    code = "\n", opts = list(label = "\"oc_block2\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_block2", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "oc_block2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_block3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_block3-code-editor`)), session)
output$`tutorial-exercise-oc_block3-output` <- renderUI({
  `tutorial-exercise-oc_block3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_block3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_block3", 
    code = "\n", opts = list(label = "\"oc_block3\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_block3", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "oc_block3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_block4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_block4-code-editor`)), session)
output$`tutorial-exercise-oc_block4-output` <- renderUI({
  `tutorial-exercise-oc_block4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_block4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_block4", 
    code = "\n", opts = list(label = "\"oc_block4\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_block4", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "oc_block4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_block5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_block5-code-editor`)), session)
output$`tutorial-exercise-oc_block5-output` <- renderUI({
  `tutorial-exercise-oc_block5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_block5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_block5", 
    code = "\n", opts = list(label = "\"oc_block5\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_block5", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "oc_block5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-oc_block6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-oc_block6-code-editor`)), session)
output$`tutorial-exercise-oc_block6-output` <- renderUI({
  `tutorial-exercise-oc_block6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "oc_block6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "oc_block6", 
    code = "\n", opts = list(label = "\"oc_block6\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "oc_block6", exercise = TRUE, code = c("", ""
        ), out.width.px = 624, out.height.px = 384, params.src = "oc_block6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-barplot-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-barplot-code-editor`)), session)
output$`tutorial-exercise-barplot-output` <- renderUI({
  `tutorial-exercise-barplot-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "barplot", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "barplot", 
    code = "#Subsetting based on a group\nmy_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")\n\n#Plotting using the new subset\nplot_bar(my_subset, x= \"subject\", fill = \"Class\", title = \"My title\") +\n  geom_bar(aes(color = Class, fill = Class), stat = \"identity\", position = \"stack\")", 
    opts = list(label = "\"barplot\"", exercise = "TRUE", echo = "FALSE"), 
    engine = "r")), code_check = NULL, error_check = NULL, check = NULL, 
    solution = NULL, tests = NULL, options = list(eval = FALSE, 
        echo = TRUE, results = "markup", tidy = FALSE, tidy.opts = NULL, 
        collapse = FALSE, prompt = FALSE, comment = NA, highlight = FALSE, 
        size = "normalsize", background = "#F7F7F7", strip.white = TRUE, 
        cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "barplot", exercise = TRUE, code = c("#Subsetting based on a group", 
        "my_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")", 
        "", "#Plotting using the new subset", "plot_bar(my_subset, x= \"subject\", fill = \"Class\", title = \"My title\") +", 
        "  geom_bar(aes(color = Class, fill = Class), stat = \"identity\", position = \"stack\")"
        ), out.width.px = 624, out.height.px = 384, params.src = "barplot, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_out_barplot1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_out_barplot1-code-editor`)), session)
output$`tutorial-exercise-try_it_out_barplot1-output` <- renderUI({
  `tutorial-exercise-try_it_out_barplot1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_out_barplot1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_out_barplot1", 
    code = "", opts = list(label = "\"try_it_out_barplot1\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_out_barplot1", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "try_it_out_barplot1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_out_barplot2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_out_barplot2-code-editor`)), session)
output$`tutorial-exercise-try_it_out_barplot2-output` <- renderUI({
  `tutorial-exercise-try_it_out_barplot2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_out_barplot2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_out_barplot2", 
    code = "", opts = list(label = "\"try_it_out_barplot2\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_out_barplot2", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "try_it_out_barplot2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_out_barplot3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_out_barplot3-code-editor`)), session)
output$`tutorial-exercise-try_it_out_barplot3-output` <- renderUI({
  `tutorial-exercise-try_it_out_barplot3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_out_barplot3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_out_barplot3", 
    code = "", opts = list(label = "\"try_it_out_barplot3\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_out_barplot3", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "try_it_out_barplot3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_out_barplot4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_out_barplot4-code-editor`)), session)
output$`tutorial-exercise-try_it_out_barplot4-output` <- renderUI({
  `tutorial-exercise-try_it_out_barplot4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_out_barplot4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_out_barplot4", 
    code = "", opts = list(label = "\"try_it_out_barplot4\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_out_barplot4", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "try_it_out_barplot4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_out_barplot5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_out_barplot5-code-editor`)), session)
output$`tutorial-exercise-try_it_out_barplot5-output` <- renderUI({
  `tutorial-exercise-try_it_out_barplot5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_out_barplot5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_out_barplot5", 
    code = "", opts = list(label = "\"try_it_out_barplot5\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_out_barplot5", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "try_it_out_barplot5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_out_barplot6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_out_barplot6-code-editor`)), session)
output$`tutorial-exercise-try_it_out_barplot6-output` <- renderUI({
  `tutorial-exercise-try_it_out_barplot6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_out_barplot6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_out_barplot6", 
    code = "", opts = list(label = "\"try_it_out_barplot6\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_out_barplot6", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "try_it_out_barplot6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-try_it_plot_richness_end-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-try_it_plot_richness_end-code-editor`)), session)
output$`tutorial-exercise-try_it_plot_richness_end-output` <- renderUI({
  `tutorial-exercise-try_it_plot_richness_end-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "try_it_plot_richness_end", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "try_it_plot_richness_end", 
    code = "\nmy_subset <- subset_samples(miso_counts, gender == \"F\")\n\nplot_richness(my_subset, x=\"timepoint\", \n              color=\"phase\", \n              shape=\"timepointgroup\",\n          \t  title =\"fill in the blank\", \n              measures= \"Simpson\")\n", 
    opts = list(label = "\"try_it_plot_richness_end\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "try_it_plot_richness_end", exercise = TRUE, 
        code = c("", "my_subset <- subset_samples(miso_counts, gender == \"F\")", 
        "", "plot_richness(my_subset, x=\"timepoint\", ", "              color=\"phase\", ", 
        "              shape=\"timepointgroup\",", "          \t  title =\"fill in the blank\", ", 
        "              measures= \"Simpson\")", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "try_it_plot_richness_end, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank1-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank1-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank1", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank1\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank1", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank2-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank2-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank2", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank2\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank2", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank3-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank3-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank3", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank3\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank3", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank4-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank4-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank4", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank4\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank4", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank5-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank5-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank5", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank5\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank5", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_plot_richness_blank6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_plot_richness_blank6-code-editor`)), session)
output$`tutorial-exercise-Try_it_plot_richness_blank6-output` <- renderUI({
  `tutorial-exercise-Try_it_plot_richness_blank6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_plot_richness_blank6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_plot_richness_blank6", 
    code = "", opts = list(label = "\"Try_it_plot_richness_blank6\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_plot_richness_blank6", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_plot_richness_blank6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_end-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_end-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_end-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_end-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_end", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_end", 
    code = "#Subset a group of interest\nmiso_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")\n\n#Ordinate based on your subset\nmiso.pcoa.subset <- ordinate(miso, method=\"PCoA\", distance=\"bray\")\n\n#Plot the new subsetted ordination with the subsetted phyloseq object\nplot_ordination(miso_subset, miso.pcoa.subset, \n        color = \"subject\", \n        shape = \"phase\", \n        title=\"My subsetted (Firmicutes) PCoA\")", 
    opts = list(label = "\"Try_it_PCoA_end\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_end", exercise = TRUE, code = c("#Subset a group of interest", 
        "miso_subset <- subset_taxa(miso, Phylum == \"Firmicutes\")", 
        "", "#Ordinate based on your subset", "miso.pcoa.subset <- ordinate(miso, method=\"PCoA\", distance=\"bray\")", 
        "", "#Plot the new subsetted ordination with the subsetted phyloseq object", 
        "plot_ordination(miso_subset, miso.pcoa.subset, ", "        color = \"subject\", ", 
        "        shape = \"phase\", ", "        title=\"My subsetted (Firmicutes) PCoA\")"
        ), out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_end, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank1-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank1-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank1", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank1\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank1", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank2-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank2-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank2", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank2\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank2", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank3-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank3-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank3", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank3\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank3", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank4-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank4-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank4", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank4\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank4", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank5-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank5-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank5", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank5\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank5", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_PCoA_blank6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_PCoA_blank6-code-editor`)), session)
output$`tutorial-exercise-Try_it_PCoA_blank6-output` <- renderUI({
  `tutorial-exercise-Try_it_PCoA_blank6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_PCoA_blank6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_PCoA_blank6", 
    code = "", opts = list(label = "\"Try_it_PCoA_blank6\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_PCoA_blank6", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_PCoA_blank6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-DESeq2_example-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-DESeq2_example-code-editor`)), session)
output$`tutorial-exercise-DESeq2_example-output` <- renderUI({
  `tutorial-exercise-DESeq2_example-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "DESeq2_example", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "DESeq2_example", 
    code = "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design \nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)\n\n# STEP 2: Select the groups to compare\nmy_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3, position = \"jitter\") +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+\n  ggtitle(\"my title here\")\n", 
    opts = list(label = "\"DESeq2_example\"", exercise = "TRUE", 
        echo = "TRUE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "DESeq2_example", exercise = TRUE, code = c("# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design ", 
        "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ timepointgroup)", 
        "", "# STEP 2: Select the groups to compare", "my_comparision <-c(\"timepointgroup\", \"WO\", \"BD\")", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs<-Differential_Abundance(my_miso_DESeq2, my_comparision, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3, position = \"jitter\") +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+", 
        "  ggtitle(\"my title here\")", ""), out.width.px = 624, 
        out.height.px = 384, params.src = "DESeq2_example, exercise=TRUE, echo=TRUE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_blank1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_blank1-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_blank1-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_blank1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_blank1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_blank1", 
    code = "", opts = list(label = "\"Try_it_DA_blank1\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_blank1", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_blank1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_blank2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_blank2-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_blank2-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_blank2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_blank2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_blank2", 
    code = "", opts = list(label = "\"Try_it_DA_blank2\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_blank2", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_blank2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_blank3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_blank3-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_blank3-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_blank3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_blank3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_blank3", 
    code = "", opts = list(label = "\"Try_it_DA_blank3\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_blank3", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_blank3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_blank4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_blank4-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_blank4-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_blank4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_blank4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_blank4", 
    code = "", opts = list(label = "\"Try_it_DA_blank4\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_blank4", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_blank4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_blank5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_blank5-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_blank5-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_blank5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_blank5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_blank5", 
    code = "", opts = list(label = "\"Try_it_DA_blank5\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_blank5", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_blank5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_blank6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_blank6-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_blank6-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_blank6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_blank6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_blank6", 
    code = "", opts = list(label = "\"Try_it_DA_blank6\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_blank6", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_blank6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_continuous-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_continuous-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_continuous-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_continuous-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_continuous", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_continuous", 
    code = "\n# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design\nmy_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ age)\n\n# STEP 2: Select the groups to compare\nname_for_comparision <-\"age\"\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs<-Differential_Abundance_Continuous(my_miso_DESeq2, name_for_comparision, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class, shape = Order)) + geom_point(size=6) +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))\n", 
    opts = list(label = "\"Try_it_DA_continuous\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_continuous", exercise = TRUE, code = c("", 
        "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design", 
        "my_miso_DESeq2 <- phyloseq_to_deseq2(miso_counts, design = ~ age)", 
        "", "# STEP 2: Select the groups to compare", "name_for_comparision <-\"age\"", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs<-Differential_Abundance_Continuous(my_miso_DESeq2, name_for_comparision, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class, shape = Order)) + geom_point(size=6) +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))", 
        ""), out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_continuous, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_Continuous_blank1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_Continuous_blank1-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_Continuous_blank1-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_Continuous_blank1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_Continuous_blank1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_Continuous_blank1", 
    code = "", opts = list(label = "\"Try_it_DA_Continuous_blank1\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_Continuous_blank1", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_Continuous_blank1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_Continuous_blank2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_Continuous_blank2-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_Continuous_blank2-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_Continuous_blank2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_Continuous_blank2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_Continuous_blank2", 
    code = "", opts = list(label = "\"Try_it_DA_Continuous_blank2\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_Continuous_blank2", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_Continuous_blank2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_Continuous_blank3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_Continuous_blank3-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_Continuous_blank3-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_Continuous_blank3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_Continuous_blank3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_Continuous_blank3", 
    code = "", opts = list(label = "\"Try_it_DA_Continuous_blank3\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_Continuous_blank3", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_Continuous_blank3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_Continuous_blank4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_Continuous_blank4-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_Continuous_blank4-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_Continuous_blank4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_Continuous_blank4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_Continuous_blank4", 
    code = "", opts = list(label = "\"Try_it_DA_Continuous_blank4\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_Continuous_blank4", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_Continuous_blank4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_Continuous_blank5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_Continuous_blank5-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_Continuous_blank5-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_Continuous_blank5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_Continuous_blank5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_Continuous_blank5", 
    code = "", opts = list(label = "\"Try_it_DA_Continuous_blank5\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_Continuous_blank5", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_Continuous_blank5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_Continuous_blank6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_Continuous_blank6-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_Continuous_blank6-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_Continuous_blank6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_Continuous_blank6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_Continuous_blank6", 
    code = "", opts = list(label = "\"Try_it_DA_Continuous_blank6\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_Continuous_blank6", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_Continuous_blank6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf", 
    code = "# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design \nmy_ad_DESeq2 <- phyloseq_to_deseq2(ad_counts, design = ~group)\n\n# STEP 2: Select the groups to compare\nmy_comparison <- c(\"group\", \"AD\", \"control\")\n\n# STEP 3: Run the differential abundance analysis at the chosen p-value\nSignificant_DEseq2_ASVs <- Differential_Abundance_AltGeo(ad_counts, my_ad_DESeq2, my_comparison, 0.05)\n\n# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups\nSignificant_DEseq2_ASVs\n\n# STEP 5: Plot the results with your chosen x axis and legend\nggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3, position = \"jitter\") +\n  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+\n  ggtitle(\"my title here\")", 
    opts = list(label = "\"Try_it_DA_zero-inf\"", exercise = "TRUE", 
        echo = "FALSE"), engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf", exercise = TRUE, code = c("# STEP 1: Convert the phyloseq object to a DESeq2 object and tell R the experimental design ", 
        "my_ad_DESeq2 <- phyloseq_to_deseq2(ad_counts, design = ~group)", 
        "", "# STEP 2: Select the groups to compare", "my_comparison <- c(\"group\", \"AD\", \"control\")", 
        "", "# STEP 3: Run the differential abundance analysis at the chosen p-value", 
        "Significant_DEseq2_ASVs <- Differential_Abundance_AltGeo(ad_counts, my_ad_DESeq2, my_comparison, 0.05)", 
        "", "# STEP 4: Retrieve the list of ASVs with a significant difference in abundance between the chosen groups", 
        "Significant_DEseq2_ASVs", "", "# STEP 5: Plot the results with your chosen x axis and legend", 
        "ggplot(Significant_DEseq2_ASVs, aes(x = Phylum, y=log2FoldChange, color= Class)) + geom_point(size=3, position = \"jitter\") +", 
        "  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+", 
        "  ggtitle(\"my title here\")"), out.width.px = 624, 
        out.height.px = 384, params.src = "Try_it_DA_zero-inf, exercise=TRUE, echo=FALSE", 
        fig.num = 0, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf_blank1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf_blank1-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf_blank1-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf_blank1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf_blank1", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf_blank1", 
    code = "", opts = list(label = "\"Try_it_DA_zero-inf_blank1\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf_blank1", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_zero-inf_blank1, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf_blank2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf_blank2-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf_blank2-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf_blank2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf_blank2", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf_blank2", 
    code = "", opts = list(label = "\"Try_it_DA_zero-inf_blank2\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf_blank2", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_zero-inf_blank2, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf_blank3-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf_blank3-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf_blank3-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf_blank3-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf_blank3", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf_blank3", 
    code = "", opts = list(label = "\"Try_it_DA_zero-inf_blank3\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf_blank3", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_zero-inf_blank3, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf_blank4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf_blank4-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf_blank4-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf_blank4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf_blank4", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf_blank4", 
    code = "", opts = list(label = "\"Try_it_DA_zero-inf_blank4\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf_blank4", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_zero-inf_blank4, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf_blank5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf_blank5-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf_blank5-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf_blank5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf_blank5", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf_blank5", 
    code = "", opts = list(label = "\"Try_it_DA_zero-inf_blank5\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf_blank5", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_zero-inf_blank5, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-Try_it_DA_zero-inf_blank6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-Try_it_DA_zero-inf_blank6-code-editor`)), session)
output$`tutorial-exercise-Try_it_DA_zero-inf_blank6-output` <- renderUI({
  `tutorial-exercise-Try_it_DA_zero-inf_blank6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "Try_it_DA_zero-inf_blank6", global_setup = structure(c("#Load learnr", 
"library(learnr)", "knitr::opts_chunk$set(echo = FALSE)", "tutorial_options(exercise.completion=TRUE) #Uncomment to enable ", 
"", "#Load libraries", "library(\"tidyverse\")", "library(\"phyloseq\")", 
"library(\"DESeq2\")", "", "#Uncomment exercise timelimit when deployed", 
"tutorial_options(exercise.timelimit = 360)", "", "#Load MISO_16S.prop.RDS", 
"#Normalized data from Leah Guthrie's original phyloseq object, miso2ps.rds", 
"#Normalization by turning counts into proportions", "", "# #Local path", 
"", "", "miso <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.prop.clean.RDS\")", 
"miso_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S MISO data/MISO_16S.clean.RDS\")", 
"", "fefifo <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo.RDS\")", 
"fefifo_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S FeFiFo data/fefifo_counts.RDS\")", 
"", "", "ad <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad.RDS\")", 
"ad_counts <- readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Alzheimers/ad_counts.RDS\")", 
"", "oc<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc.RDS\")", 
"oc_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_counts.RDS\")", 
"oc_no948N2_counts<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2_counts.RDS\")", 
"oc_no948N2<-readRDS(\"C:/Users/syork/Desktop/Sayumi/C-MOOR/C-MOOR_DEVELOPMENT/C-MOOR_16S/CMOOR-16S DATA/16S Oral Contraceptives/oc_no948N2.RDS\")", 
"", "", "", "", "", "#", "# #########################", "# #SciServer paths#", 
"# ########################", "# # #Alzheimer's disease", "# ad <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad.RDS\")", 
"# ad_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/ad/ad_counts.RDS\")", 
"# #", "# # #FeFiFo", "# fefifo <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo.RDS\")", 
"# fefifo_counts<- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/fefifo/fefifo_counts.RDS\")", 
"# #", "# # #MISO", "# miso <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.prop.clean.rds\" )", 
"# miso_counts <- readRDS(\"/home/idies/workspace/c_moor_data/16s-amplicon-data/miso/MISO_16S.clean.rds\")", 
"# #", "# # #Oral contraceptives (and prunes)", "# oc <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc.RDS\" )", 
"# oc_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_counts.RDS\" )", 
"# oc_no948N2 <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2.RDS\" )", 
"# oc_no948N2_counts <- readRDS( \"/home/idies/workspace/c_moor_data/16s-amplicon-data/oc/oc_no948N2_counts.RDS\" )", 
"# #", "", "", "# <!-- EXERCISE = should we see an interactive code chunk with this code? -->", 
"# <!-- EVAL = should you run this code? -->", "# <!-- ECHO = should you print the code? -->", 
"# Significant_DEseq2_ASVs <-0", "# sigtab<-0", "", "Differential_Abundance <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_comparision_list <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For using a continuous variable during comparisons", 
"# log2foldchange is interpreted as the amount of change for each additional unit of the continuous avriable", 
"", "Differential_Abundance_Continuous <- function(x, y, z) {", 
"  ", " # Accept the input and give it a human friendly name", 
"  my_DESeq2_object <- x", "  my_name <- y", "  my_pvalue <- z", 
"  ", "  ", "  #Do the DESeq2 analysis", "  my_DESeq2_object <- DESeq(my_DESeq2_object, test=\"Wald\", fitType=\"parametric\")", 
" ", "  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, name=my_name)", 
"  ", "  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(miso_counts)[rownames(sigtab), ], \"matrix\"))", 
"", "  #Let's look at the OTUs that were significantly different between the two tissues. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"", "  return(sigtab)", "}", "", "# For 0-inflated data that gives the error:", 
"# every gene contains at least one zero, cannot compute log geometric means", 
"", "Differential_Abundance_AltGeo <- function(w, x, y, z) {", 
"  ", "  # Accept the input and give it a human friendly name", 
"  my_counts <- w", "  my_DESeq2_object <- x", "  my_comparision_list <- y", 
"  my_pvalue <- z", "  ", "  ", "  ", "  #Do the DESeq2 analysis", 
"  ", "  gm_mean = function(x, na.rm = TRUE){", "    exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))", 
"  }", "  ", "  geoMeans = apply(counts(my_DESeq2_object), 1, gm_mean)", 
"  my_DESeq2_object = estimateSizeFactors(my_DESeq2_object, geoMeans = geoMeans)", 
"  my_DESeq2_object = estimateDispersions(my_DESeq2_object, fitType = \"parametric\")", 
"  my_DESeq2_object <- nbinomWaldTest(my_DESeq2_object)", "  ", 
"  #Change the contrast section to compare different variables and groups.", 
"  res = results(my_DESeq2_object, cooksCutoff = FALSE, contrast= my_comparision_list)", 
"  ", "  #Select only abundances with a p-value of z or below", 
"  alpha = my_pvalue", "  sigtab = res[which(res$padj < alpha), ]", 
"  sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(my_counts)[rownames(sigtab), ], \"matrix\"))", 
"  ", "  #Let's look at the OTUs that were significantly different between the two groups. The following makes a nice ggplot2 summary of the results", 
"  theme_set(theme_bw())", "  scale_fill_discrete <- function(palname = \"Set1\", ...) {", 
"    scale_fill_brewer(palette = palname, ...)", "  }", "  # Phylum order", 
"  x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))", 
"  ", "  # Class order", "  x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))", 
"  ", "  # Order order", "  x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))", 
"  ", "  # Family order", "  x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))", 
"  ", "  # Genus order", "  x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))", 
"  ", "  # Species order", "  x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))", 
"  x = sort(x, TRUE)", "  sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))", 
"  ", "  return(sigtab)", "}", "", ""), chunk_opts = list(label = "setup", 
    include = FALSE)), setup = NULL, chunks = list(list(label = "Try_it_DA_zero-inf_blank6", 
    code = "", opts = list(label = "\"Try_it_DA_zero-inf_blank6\"", 
        exercise = "TRUE", echo = "FALSE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "16S-amplicon-project-work_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "16S-amplicon-project-work_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.completion = TRUE, exercise.timelimit = 360, 
        label = "Try_it_DA_zero-inf_blank6", exercise = TRUE, 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "Try_it_DA_zero-inf_blank6, exercise=TRUE, echo=FALSE", 
        fig.num = 0L, exercise.df_print = "paged", exercise.checker = "NULL"), 
    engine = "r", version = "4"), class = c("r", "tutorial_exercise"
)))
</script>
</p>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["header-attrs"]},{"type":"character","attributes":{},"value":["2.29"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pandoc"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["header-attrs.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/cerulean.min.css"]},{"type":"character","attributes":{},"value":["<style>h1 {font-size: 34px;}\n       h1.title {font-size: 38px;}\n       h2 {font-size: 30px;}\n       h3 {font-size: 24px;}\n       h4 {font-size: 18px;}\n       h5 {font-size: 16px;}\n       h6 {font-size: 12px;}\n       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}\n       pre:not([class]) { background-color: white }<\/style>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["pagedtable"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pagedtable-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["js/pagedtable.js"]},{"type":"character","attributes":{},"value":["css/pagedtable.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["textmate.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.11.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["i18n"]},{"type":"character","attributes":{},"value":["21.6.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/i18n"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["i18next.min.js","tutorial-i18n-init.js"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["<script id=\"i18n-cstm-trns\" type=\"application/json\">{\"language\":\"en\",\"resources\":{\"en\":{\"translation\":{\"button\":{\"runcode\":\"Run Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Hint\",\"hint_plural\":\"Hints\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Next Hint\",\"hintprev\":\"Previous Hint\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copy to Clipboard\",\"startover\":\"Start Over\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continue\",\"submitanswer\":\"Submit Answer\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Previous Topic\",\"nexttopic\":\"Next Topic\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Try Again\"},\"text\":{\"startover\":\"Start Over\",\"areyousure\":\"Are you sure you want to start over? (all exercise progress will be reset)\",\"youmustcomplete\":\"You must complete the\",\"exercise\":\"exercise\",\"exercise_plural\":\"exercises\",\"inthissection\":\"in this section before continuing.\",\"code\":\"Code\",\"enginecap\":\"{{engine}} $t(text.code)\",\"quiz\":\"Quiz\",\"blank\":\"blank\",\"blank_plural\":\"blanks\",\"exercisecontainsblank\":\"This exercise contains {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Please replace {{blank}} with valid code.\",\"unparsable\":\"It looks like this might not be valid R code. R cannot determine how to turn your text into a complete command. You may have forgotten to fill in a blank, to remove an underscore, to include a comma between arguments, or to close an opening <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> or <code>{<\\/code> with a matching <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> or <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>It looks like your R code contains specially formatted quotation marks or &quot;curly&quot; quotes (<code>{{character}}<\\/code>) around character strings, making your code invalid. R requires character values to be contained in straight quotation marks (<code>&quot;<\\/code> or <code>'<\\/code>).<\\/p> {{code}} <p>Don't worry, this is a common source of errors when you copy code from another app that applies its own formatting to text. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. Try deleting the special character from your code and retyping it manually.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"and\":\"and\",\"or\":\"or\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"fr\":{\"translation\":{\"button\":{\"runcode\":\"Lancer le Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Indication\",\"hint_plural\":\"Indications\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Indication Suivante\",\"hintprev\":\"Indication Précédente\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copier dans le Presse-papier\",\"startover\":\"Recommencer\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuer\",\"submitanswer\":\"Soumettre\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Chapitre Précédent\",\"nexttopic\":\"Chapitre Suivant\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Réessayer\"},\"text\":{\"startover\":\"Recommencer\",\"areyousure\":\"Êtes-vous certains de vouloir recommencer? (La progression sera remise à zéro)\",\"youmustcomplete\":\"Vous devez d'abord compléter\",\"exercise\":\"l'exercice\",\"exercise_plural\":\"des exercices\",\"inthissection\":\"de cette section avec de continuer.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"et\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"es\":{\"translation\":{\"button\":{\"runcode\":\"Ejecutar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Pista\",\"hint_plural\":\"Pistas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Siguiente pista\",\"hintprev\":\"Pista anterior\",\"solution\":\"Solución\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar al portapapeles\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar respuesta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tema anterior\",\"nexttopic\":\"Tema siguiente\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Volver a intentar\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"¿De verdad quieres empezar de nuevo? (todo el progreso del ejercicio se perderá)\",\"youmustcomplete\":\"Debes completar\",\"exercise\":\"el ejercicio\",\"exercise_plural\":\"los ejercicios\",\"inthissection\":\"en esta sección antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Cuestionario\",\"and\":\"y\",\"or\":\"o\",\"oxfordcomma\":\"\"}}},\"pt\":{\"translation\":{\"button\":{\"runcode\":\"Executar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Dica\",\"hint_plural\":\"Dicas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Próxima dica\",\"hintprev\":\"Dica anterior\",\"solution\":\"Solução\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar para a área de transferência\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar resposta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tópico anterior\",\"nexttopic\":\"Próximo tópico\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tentar novamente\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"Tem certeza que deseja começar novamente? (todo o progresso feito será perdido)\",\"youmustcomplete\":\"Você deve completar\",\"exercise\":\"o exercício\",\"exercise_plural\":\"os exercícios\",\"inthissection\":\"nesta seção antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"e\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"tr\":{\"translation\":{\"button\":{\"runcode\":\"Çalıştırma Kodu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Ipucu\",\"hint_plural\":\"İpuçları\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Sonraki İpucu\",\"hintprev\":\"Önceki İpucu\",\"solution\":\"Çözüm\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Pano'ya Kopyala\",\"startover\":\"Baştan Başlamak\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Devam et\",\"submitanswer\":\"Cevabı onayla\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Önceki Konu\",\"nexttopic\":\"Sonraki Konu\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tekrar Deneyin\"},\"text\":{\"startover\":\"Baştan Başlamak\",\"areyousure\":\"Baştan başlamak istediğinizden emin misiniz? (tüm egzersiz ilerlemesi kaybolacak)\",\"youmustcomplete\":\"Tamamlamalısın\",\"exercise\":\"egzersiz\",\"exercise_plural\":\"egzersizler\",\"inthissection\":\"devam etmeden önce bu bölümde\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Sınav\",\"oxfordcomma\":\"\"}}},\"emo\":{\"translation\":{\"button\":{\"runcode\":\"🏃\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"💡\",\"hint_plural\":\"$t(button.hint)\",\"hinttitle\":\"$t(button.hint)\",\"solution\":\"🎯\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"📋\",\"startover\":\"⏮\",\"startovertitle\":\"Start Over\",\"continue\":\"✅\",\"submitanswer\":\"🆗\",\"submitanswertitle\":\"Submit Answer\",\"previoustopic\":\"⬅\",\"nexttopic\":\"➡\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"🔁\"},\"text\":{\"startover\":\"⏮\",\"areyousure\":\"🤔\",\"youmustcomplete\":\"⚠️ 👉 🧑‍💻\",\"exercise\":\"\",\"exercise_plural\":\"\",\"inthissection\":\"\",\"code\":\"💻\",\"enginecap\":\"$t(text.code) {{engine}}\",\"oxfordcomma\":\"\"}}},\"eu\":{\"translation\":{\"button\":{\"runcode\":\"Kodea egikaritu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Laguntza\",\"hint_plural\":\"Laguntza\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Aurreko laguntza\",\"hintprev\":\"Hurrengo laguntza\",\"solution\":\"Ebazpena\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Arbelean kopiatu\",\"startover\":\"Berrabiarazi\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Jarraitu\",\"submitanswer\":\"Erantzuna bidali\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Aurreko atala\",\"nexttopic\":\"Hurrengo atala\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Berriro saiatu\"},\"text\":{\"startover\":\"Berrabiarazi\",\"areyousure\":\"Berriro hasi nahi duzu? (egindako lana galdu egingo da)\",\"youmustcomplete\":\"Aurrera egin baino lehen atal honetako\",\"exercise\":\"ariketa egin behar duzu.\",\"exercise_plural\":\"ariketak egin behar dituzu.\",\"inthissection\":\"\",\"code\":\"Kodea\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Galdetegia\",\"oxfordcomma\":\"\"}}},\"de\":{\"translation\":{\"button\":{\"runcode\":\"Code ausführen\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Tipp\",\"hint_plural\":\"Tipps\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Nächster Tipp\",\"hintprev\":\"Vorheriger Tipp\",\"solution\":\"Lösung\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"In die Zwischenablage kopieren\",\"startover\":\"Neustart\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Weiter\",\"submitanswer\":\"Antwort einreichen\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Vorheriges Kapitel\",\"nexttopic\":\"Nächstes Kapitel\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Nochmal versuchen\"},\"text\":{\"startover\":\"Neustart\",\"areyousure\":\"Bist du sicher, dass du neustarten willst? (der gesamte Lernfortschritt wird gelöscht)\",\"youmustcomplete\":\"Vervollstädinge\",\"exercise\":\"die Übung\",\"exercise_plural\":\"die Übungen\",\"inthissection\":\"in diesem Kapitel, bevor du fortfährst.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"Lücke\",\"blank_plural\":\"Lücken\",\"pleasereplaceblank\":\"Bitte ersetze {{blank}} mit gültigem Code.\",\"unparsable\":\"Dies scheint kein gültiger R Code zu sein. R kann deinen Text nicht in einen gültigen Befehl übersetzen. Du hast vielleicht vergessen, die Lücke zu füllen, einen Unterstrich zu entfernen, ein Komma zwischen Argumente zu setzen oder ein eröffnendes <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> oder <code>{<\\/code> mit einem zugehörigen <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> oder <code>}<\\/code> zu schließen.\\n\",\"and\":\"und\",\"or\":\"oder\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"ko\":{\"translation\":{\"button\":{\"runcode\":\"코드 실행\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"힌트\",\"hint_plural\":\"힌트들\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"다음 힌트\",\"hintprev\":\"이전 힌트\",\"solution\":\"솔루션\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"클립보드에 복사\",\"startover\":\"재학습\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"다음 학습으로\",\"submitanswer\":\"정답 제출\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"이전 토픽\",\"nexttopic\":\"다음 토픽\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"재시도\"},\"text\":{\"startover\":\"재학습\",\"areyousure\":\"다시 시작 하시겠습니까? (모든 예제의 진행 정보가 재설정됩니다)\",\"youmustcomplete\":\"당신은 완료해야 합니다\",\"exercise\":\"연습문제\",\"exercise_plural\":\"연습문제들\",\"inthissection\":\"이 섹션을 실행하기 전에\",\"code\":\"코드\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"퀴즈\",\"blank\":\"공백\",\"blank_plural\":\"공백들\",\"exercisecontainsblank\":\"이 연습문제에는 {{count}}개의 $t(text.blank)이 포함되어 있습니다.\",\"pleasereplaceblank\":\"{{blank}}를 유효한 코드로 바꾸십시오.\",\"unparsable\":\"이것은 유효한 R 코드가 아닐 수 있습니다. R은 텍스트를 완전한 명령으로 변환하는 방법을 결정할 수 없습니다. 당신은 공백이나 밑줄을 대체하여 채우기, 인수를 컴마로 구분하기, 또는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> , <code>{<\\/code>로 시작하는 구문을 닫는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>, <code>}<\\/code>을 잊었을 수도 있습니다.\\n\",\"and\":\"그리고\",\"or\":\"혹은\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}},\"zh\":{\"translation\":{\"button\":{\"runcode\":\"运行代码\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"提示\",\"hint_plural\":\"提示\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"下一个提示\",\"hintprev\":\"上一个提示\",\"solution\":\"答案\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"复制到剪切板\",\"startover\":\"重新开始\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"继续\",\"submitanswer\":\"提交答案\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"上一专题\",\"nexttopic\":\"下一专题\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"再试一次\"},\"text\":{\"startover\":\"重置\",\"areyousure\":\"你确定要重新开始吗? (所有当前进度将被重置)\",\"youmustcomplete\":\"你必须完成\",\"exercise\":\"练习\",\"exercise_plural\":\"练习\",\"inthissection\":\"在进行本节之前\",\"code\":\"代码\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"测试\",\"blank\":\"空\",\"blank_plural\":\"空\",\"exercisecontainsblank\":\"本练习包含{{count}}个$t(text.blank)\",\"pleasereplaceblank\":\"请在{{blank}}内填写恰当的代码\",\"unparsable\":\"这似乎不是有效的R代码。 R不知道如何将您的文本转换为完整的命令。 您是否忘了填空，忘了删除下划线，忘了在参数之间包含逗号，或者是忘了用<code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>,<code>}<\\/code>来封闭<code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code>。 or <code>{<\\/code>。\\n\",\"unparsablequotes\":\"<p>您的R代码中似乎含有特殊格式的引号，或者弯引号(<code>{{character}}<\\/code>) 在字符串前后，在R中字符串应该被直引号(<code>&quot;<\\/code> 或者 <code>'<\\/code>)包裹。<\\/p> {{code}} <p>别担心，该错误经常在复制粘贴包含格式的代码时遇到， 您可以尝试将该行中的代码替换为以下代码，也许还有其他地方需要修改。<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"and\":\"且\",\"or\":\"或\",\"listcomma\":\",\",\"oxfordcomma\":\",\"}}},\"pl\":{\"translation\":{\"button\":{\"runcode\":\"Uruchom kod\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Podpowiedź\",\"hint_plural\":\"Podpowiedzi\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Następna podpowiedź\",\"hintprev\":\"Poprzednia podpowiedź\",\"solution\":\"Rozwiązanie\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Kopiuj do schowka\",\"startover\":\"Zacznij od początku\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Kontynuuj\",\"submitanswer\":\"Wyślij\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Poprzednia sekcja\",\"nexttopic\":\"Następna sekcja\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Spróbuj ponownie\"},\"text\":{\"startover\":\"Zacznij od początku\",\"areyousure\":\"Czy na pewno chcesz zacząć od początku? (cały postęp w zadaniu zostanie utracony)\",\"youmustcomplete\":\"Musisz ukończyć\",\"exercise\":\"ćwiczenie\",\"exercise_plural\":\"ćwiczenia\",\"inthissection\":\"w tej sekcji przed kontynuowaniem\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"luka\",\"blank_plural\":\"luk(i)\",\"exercisecontainsblank\":\"To ćwiczenie zawiera {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Proszę uzupełnić {{blank}} prawidłowym kodem.\",\"unparsable\":\"Wygląda na to, że może to nie być prawidłowy kod R. R nie jest w stanie przetworzyć Twojego tekstu na polecenie. Mogłeś(-aś) zapomnieć wypełnić luki, usunąć podkreślnik, umieścić przecinka między argumentami, lub zamknąć znak <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> lub <code>{<\\/code> odpowiadającym <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> lub <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>Wygląda na to, że Twój kod zawiera szczególnie sformatowane cudzysłowy lub cudzysłowy typograficzne (<code>{{character}}<\\/code>) przy ciągach znaków, co sprawia, że kod jest niepoprawny. R wymaga cudzysłowów prostych (<code>&quot;<\\/code> albo <code>'<\\/code>).<\\/p> {{code}} <p>Nie martw się, to powszechne źródło błędów, gdy kopiuje się kod z innego programu, który sam formatuje teskt. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Spróbuj usunąć znak specjalny i wpisać do ponownie ręcznie.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"and\":\"i\",\"or\":\"lub\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}}}}<\/script>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-format"]},{"type":"character","attributes":{},"value":["0.11.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmarkdown/templates/tutorial/resources"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-format.js"]},{"type":"character","attributes":{},"value":["tutorial-format.css","rstudio-theme.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.29"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["font-awesome"]},{"type":"character","attributes":{},"value":["6.5.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["fontawesome"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/all.min.css","css/v4-shims.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["fontawesome"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.5.3"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootbox"]},{"type":"character","attributes":{},"value":["5.5.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/bootbox"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["bootbox.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["idb-keyvalue"]},{"type":"character","attributes":{},"value":["3.2.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/idb-keyval"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["idb-keyval-iife-compat.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.11.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117]}},"value":[{"type":"character","attributes":{},"value":["abind","ade4","ape","backports","base","Biobase","BiocGenerics","BiocParallel","biomformat","Biostrings","bslib","cachem","checkmate","cli","cluster","codetools","colorspace","compiler","crayon","data.table","datasets","DelayedArray","DESeq2","digest","dplyr","evaluate","fastmap","fontawesome","forcats","foreach","generics","GenomeInfoDb","GenomeInfoDbData","GenomicRanges","ggplot2","glue","graphics","grDevices","grid","gtable","hms","htmltools","htmlwidgets","httpuv","httr","igraph","IRanges","iterators","jquerylib","jsonlite","knitr","later","lattice","learnr","lifecycle","locfit","lubridate","magrittr","MASS","Matrix","MatrixGenerics","matrixStats","methods","mgcv","mime","multtest","munsell","nlme","parallel","permute","phyloseq","pillar","pkgconfig","plyr","promises","purrr","R6","Rcpp","readr","reshape2","rhdf5","rhdf5filters","Rhdf5lib","rlang","rmarkdown","rprojroot","rstudioapi","S4Arrays","S4Vectors","sass","scales","shiny","SparseArray","splines","stats","stats4","stringi","stringr","SummarizedExperiment","survival","tibble","tidyr","tidyselect","tidyverse","timechange","tools","tzdb","UCSC.utils","utils","vctrs","vegan","withr","xfun","xtable","XVector","yaml","zlibbioc"]},{"type":"character","attributes":{},"value":["1.4-8","1.7-23","5.8-1","1.5.0","4.4.2","2.66.0","0.52.0","1.40.0","1.34.0","2.74.1","0.9.0","1.1.0","2.3.2","3.6.4","2.1.6","0.2-20","2.1-1","4.4.2","1.5.3","1.17.0","4.4.2","0.32.0","1.46.0","0.6.37","1.1.4","1.0.3","1.2.0","0.5.3","1.0.0","1.5.2","0.1.3","1.42.3","1.2.13","1.58.0","3.5.2","1.8.0","4.4.2","4.4.2","4.4.2","0.3.6","1.1.3","0.5.8.1","1.6.4","1.6.15","1.4.7","2.1.4","2.40.1","1.0.14","0.1.4","1.9.0","1.49","1.4.1","0.22-6","0.11.5","1.0.4","1.5-9.11","1.9.4","2.0.3","7.3-61","1.7-1","1.18.1","1.5.0","4.4.2","1.9-1","0.12","2.62.0","0.5.1","3.1-166","4.4.2","0.9-7","1.50.0","1.10.2","2.0.3","1.8.9","1.3.2","1.0.4","2.6.1","1.0.14","2.1.5","1.4.4","2.50.2","1.18.0","1.28.0","1.1.5","2.29","2.0.4","0.17.1","1.6.0","0.44.0","0.4.9","1.3.0","1.10.0","1.6.2","4.4.2","4.4.2","4.4.2","1.8.4","1.5.1","1.36.0","3.7-0","3.2.1","1.3.1","1.2.1","2.0.0","0.3.0","4.4.2","0.4.0","1.2.0","4.4.2","0.6.5","2.6-10","3.0.2","0.51","1.8-4","0.46.0","2.3.10","1.52.0"]}]}]}
</script>
<!--/html_preserve-->
</div>

</article> <!-- topics -->

<div class="topicsContainer">
<div class="topicsPositioner">
<div class="band">
<div class="bandContent topicsListContainer">

<!-- begin doc-metadata -->
<div id="doc-metadata">
<h1 class="title toc-ignore" style="display:none;">16S rRNA Human Gut
Microbiome Project Work</h1>
<h2 class="subtitle"><em><p><img src="images/cmoor_logo_notext.png"
style="width:60.0%" /> <a href="https://help.c-moor.org/"
target="_blank">Get Help</a></p></em></h2>
<h4 class="author"><em>Sayumi York, Notre Dame of Maryland
University</em></h4>
</div>
<!-- end doc-metadata -->

</div> <!-- bandContent.topicsListContainer -->
</div> <!-- band -->
</div> <!-- topicsPositioner -->
</div> <!-- topicsContainer -->


</main> <!-- bandContent page -->
</div> <!-- pageContent band -->



<!-- Build Tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<script>
// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>

</html>
